// igniteui-webcomponents/components/progress/linear-progress.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
import { html, nothing } from 'lit';
import { property } from 'lit/decorators.js';
import { asPercent, partNameMap } from '../common/util.js';
import { IgcProgressBaseComponent } from './base.js';
import { styles } from './themes/linear/linear.progress.base.css.js';
import { styles as bootstrap } from './themes/linear/linear.progress.bootstrap.css.js';
import { styles as fluent } from './themes/linear/linear.progress.fluent.css.js';
import { styles as indigo } from './themes/linear/linear.progress.indigo.css.js';
import { styleMap } from 'lit/directives/style-map.js';
import { themes } from '../../theming/theming-decorator.js';
let IgcLinearProgressComponent = class IgcLinearProgressComponent extends IgcProgressBaseComponent {
    constructor() {
        super(...arguments);
        this.striped = false;
        this.labelAlign = 'top-start';
    }
    get wrapperParts() {
        return {
            fill: true,
            striped: this.striped,
            indeterminate: this.indeterminate,
            primary: this.variant === 'primary',
            success: this.variant === 'success',
            danger: this.variant === 'danger',
            warning: this.variant === 'warning',
            info: this.variant === 'info',
        };
    }
    get animInfo() {
        return {
            width: asPercent(this.value, this.max) + '%',
            '--duration': this.animationDuration + 'ms',
        };
    }
    render() {
        return html `
      <div
        part="track"
        role="progressbar"
        aria-valuemin="0"
        aria-valuemax=${this.max}
        aria-valuenow=${this.indeterminate ? nothing : this.value}
      >
        <div
          part="${partNameMap(this.wrapperParts)}"
          style="${styleMap(this.animInfo)}"
        ></div>
        <div part="${partNameMap(this.wrapperParts)} secondary"></div>
      </div>
      ${this.renderDefaultSlot()}
    `;
    }
};
IgcLinearProgressComponent.tagName = 'igc-linear-progress';
IgcLinearProgressComponent.styles = styles;
__decorate([
    property({ type: Boolean, reflect: true })
], IgcLinearProgressComponent.prototype, "striped", void 0);
__decorate([
    property({ attribute: 'label-align', reflect: true })
], IgcLinearProgressComponent.prototype, "labelAlign", void 0);
IgcLinearProgressComponent = __decorate([
    themes({ bootstrap, indigo, fluent })
], IgcLinearProgressComponent);
export default IgcLinearProgressComponent;
