// igniteui-webcomponents/components/nav-drawer/themes/light/indigo/nav-drawer-item.indigo.css.js
import { css } from 'lit';
export const styles = css `/* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */
:host [part~=base] {
  border-radius: clamp(0rem, calc(var(--ig-radius-factor) * 1.5rem), 1.5rem);
  gap: 1.5rem;
  color: var(--text-color, hsla(var(--ig-gray-300), 0.8));
  padding: 0.25rem;
  margin: 0.9375rem 0.5rem;
}
:host [part~=base]:hover, :host [part~=base]:focus-within {
  background: transparent;
  color: var(--hover-text-color, hsla(var(--ig-gray-100), var(--ig-gray-a)));
}
:host [part=content] {
  -webkit-padding-start: 0.5rem;
          padding-inline-start: 0.5rem;
}

:host([active]) [part~=base] {
  background: var(--active-background, hsla(var(--ig-gray-600), 0.8));
  color: hsla(var(--ig-gray-50), var(--ig-gray-a));
}`;
