// igniteui-webcomponents/components/tree/tree.navigation.js
export const NAVIGATION_KEYS = new Set([
    'down',
    'up',
    'left',
    'right',
    'arrowdown',
    'arrowup',
    'arrowleft',
    'arrowright',
    'home',
    'end',
    'space',
    'spacebar',
    ' ',
]);
export class IgcTreeNavigationService {
    constructor(tree, selectionService) {
        this._focusedItem = null;
        this._lastFocusedItem = null;
        this._activeItem = null;
        this._visibleChildren = [];
        this._invisibleChildren = new Set();
        this._disabledChildren = new Set();
        this.tree = tree;
        this.selectionService = selectionService;
    }
    updateVisChild() {
        var _a;
        this._visibleChildren = ((_a = this.tree) === null || _a === void 0 ? void 0 : _a.items)
            ? this.tree.items.filter((i) => !(this._invisibleChildren.has(i) || this._disabledChildren.has(i)))
            : [];
    }
    get focusedItem() {
        return this._focusedItem;
    }
    focusItem(value, shouldFocus = true) {
        var _a;
        if (this._focusedItem === value) {
            return;
        }
        this._lastFocusedItem = this._focusedItem;
        if (this._lastFocusedItem) {
            this._lastFocusedItem.removeAttribute('tabindex');
        }
        this._focusedItem = value;
        if (this._focusedItem !== null && shouldFocus) {
            this._focusedItem.tabIndex = 0;
            this._focusedItem.focus({
                preventScroll: true,
            });
            (_a = this._focusedItem.wrapper) === null || _a === void 0 ? void 0 : _a.scrollIntoView({
                behavior: 'smooth',
                block: 'nearest',
                inline: 'nearest',
            });
        }
    }
    get activeItem() {
        return this._activeItem;
    }
    setActiveItem(value, shouldEmit = true) {
        if (this._activeItem === value) {
            return;
        }
        if (this._activeItem && value) {
            this._activeItem.active = false;
        }
        this._activeItem = value;
        if (this._activeItem) {
            this._activeItem.active = true;
        }
        if (shouldEmit && this._activeItem) {
            this.tree.emitEvent('igcActiveItem', { detail: this._activeItem });
        }
    }
    get visibleChildren() {
        return this._visibleChildren;
    }
    update_disabled_cache(item) {
        if (item.disabled) {
            this._disabledChildren.add(item);
        }
        else {
            this._disabledChildren.delete(item);
        }
        this.updateVisChild();
    }
    delete_item(item) {
        if (this.activeItem === item) {
            this.setActiveItem(null);
        }
        if (this.focusedItem === item) {
            this.focusItem(null, false);
            const firstNotDisableItem = this.tree.items.find((i) => !i.disabled);
            if (firstNotDisableItem) {
                firstNotDisableItem.tabIndex = 0;
                this.focusItem(firstNotDisableItem, false);
            }
        }
    }
    update_visible_cache(item, expanded, shouldUpdateNestedChildren = true, shouldUpdate = true) {
        var _a, _b;
        if (expanded && !this._invisibleChildren.has(item)) {
            (_a = item.getChildren()) === null || _a === void 0 ? void 0 : _a.forEach((child) => {
                this._invisibleChildren.delete(child);
                if (shouldUpdateNestedChildren) {
                    this.update_visible_cache(child, child.expanded, true, false);
                }
            });
        }
        else {
            (_b = item
                .getChildren({ flatten: true })) === null || _b === void 0 ? void 0 : _b.forEach((c) => this._invisibleChildren.add(c));
        }
        if (shouldUpdate) {
            this.updateVisChild();
        }
    }
    setFocusedAndActiveItem(item, isActive = true, shouldFocus = true) {
        if (isActive) {
            this.setActiveItem(item);
        }
        this.focusItem(item, shouldFocus);
    }
    handleKeydown(event) {
        const key = event.key.toLowerCase();
        if (!this.focusedItem) {
            return;
        }
        if (!(NAVIGATION_KEYS.has(key) || key === '*')) {
            if (key === 'enter') {
                this.setActiveItem(this.focusedItem);
            }
            return;
        }
        event.preventDefault();
        this.handleNavigation(event);
    }
    handleNavigation(event) {
        switch (event.key.toLowerCase()) {
            case 'home':
                this.setFocusedAndActiveItem(this.visibleChildren[0]);
                break;
            case 'end':
                this.setFocusedAndActiveItem(this.visibleChildren[this.visibleChildren.length - 1]);
                break;
            case 'arrowleft':
            case 'left':
                if (this.tree.dir === 'rtl') {
                    this.handleArrowRight();
                }
                else {
                    this.handleArrowLeft();
                }
                break;
            case 'arrowright':
            case 'right':
                if (this.tree.dir === 'rtl') {
                    this.handleArrowLeft();
                }
                else {
                    this.handleArrowRight();
                }
                break;
            case 'arrowup':
            case 'up':
                this.handleUpDownArrow(true, event);
                break;
            case 'arrowdown':
            case 'down':
                this.handleUpDownArrow(false, event);
                break;
            case '*':
                this.handleAsterisk();
                break;
            case ' ':
            case 'spacebar':
            case 'space':
                this.handleSpace(event.shiftKey);
                break;
            default:
                return;
        }
    }
    handleArrowLeft() {
        var _a, _b, _c;
        if (((_a = this.focusedItem) === null || _a === void 0 ? void 0 : _a.expanded) && ((_b = this.focusedItem.getChildren()) === null || _b === void 0 ? void 0 : _b.length)) {
            this.setActiveItem(this.focusedItem);
            this.focusedItem.collapseWithEvent();
        }
        else {
            const parentItem = (_c = this.focusedItem) === null || _c === void 0 ? void 0 : _c.parent;
            if (parentItem && !parentItem.disabled) {
                this.setFocusedAndActiveItem(parentItem);
            }
        }
    }
    handleArrowRight() {
        var _a, _b;
        if (((_a = this.focusedItem.getChildren()) === null || _a === void 0 ? void 0 : _a.length) > 0) {
            if (!((_b = this.focusedItem) === null || _b === void 0 ? void 0 : _b.expanded)) {
                this.setActiveItem(this.focusedItem);
                this.focusedItem.expandWithEvent();
            }
            else {
                const firstChild = this.focusedItem
                    .getChildren()
                    .find((item) => !item.disabled);
                if (firstChild) {
                    this.setFocusedAndActiveItem(firstChild);
                }
            }
        }
    }
    handleUpDownArrow(isUp, event) {
        const next = this.getVisibleItem(this.focusedItem, isUp ? -1 : 1);
        if (next === this.focusedItem) {
            return;
        }
        if (event.ctrlKey) {
            this.setFocusedAndActiveItem(next, false);
        }
        else {
            this.setFocusedAndActiveItem(next);
        }
    }
    handleAsterisk() {
        var _a, _b, _c;
        const items = ((_a = this.focusedItem) === null || _a === void 0 ? void 0 : _a.parent)
            ? (_b = this.focusedItem.parent) === null || _b === void 0 ? void 0 : _b.getChildren()
            : (_c = this.tree.items) === null || _c === void 0 ? void 0 : _c.filter((item) => item.level === 0);
        items === null || items === void 0 ? void 0 : items.forEach((item) => {
            if (!item.disabled && !item.expanded && item.hasChildren) {
                item.expandWithEvent();
            }
        });
    }
    handleSpace(shiftKey = false) {
        if (this.tree.selection === 'none') {
            this.setActiveItem(this.focusedItem);
            return;
        }
        this.setActiveItem(this.focusedItem);
        if (shiftKey) {
            this.selectionService.selectMultipleItems(this.focusedItem);
            return;
        }
        if (this.focusedItem.selected) {
            this.selectionService.deselectItem(this.focusedItem);
        }
        else {
            this.selectionService.selectItem(this.focusedItem);
        }
    }
    getVisibleItem(item, dir = 1) {
        const itemIndex = this.visibleChildren.indexOf(item);
        return this.visibleChildren[itemIndex + dir] || item;
    }
}
