// igniteui-webcomponents/components/calendar/themes/bootstrap/days-view.bootstrap.css.js
import { css } from 'lit';
export const styles = css `/* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */
:host([part=days-view]) {
  overflow: hidden;
  position: relative;
  padding: 0 0 0.25rem 0;
  gap: 0.25rem;
}

:host([part=days-view]) + :host([part=days-view]) {
  -webkit-margin-start: 0.5rem;
          margin-inline-start: 0.5rem;
}

[part~=days-row][part~=first] {
  background: hsla(var(--ig-secondary-50), 0.2);
  border-bottom: 0.0625rem solid hsla(var(--ig-gray-300), var(--ig-gray-a));
}

[part~=label] {
  font-style: italic;
  color: hsla(var(--ig-info-300), var(--ig-info-a));
}

[part~=date-inner] {
  border-radius: clamp(0rem, calc(var(--ig-radius-factor) * 0.25rem), 0.25rem);
}

[part~=date-inner][part~=current] {
  -webkit-box-shadow: none;
          box-shadow: none;
  color: hsla(var(--ig-gray-800), var(--ig-gray-a));
  background: hsla(var(--ig-gray-200), var(--ig-gray-a));
  border-color: hsla(var(--ig-gray-200), var(--ig-gray-a));
}
[part~=date-inner][part~=current]:hover {
  -webkit-box-shadow: none;
          box-shadow: none;
  color: hsla(var(--ig-gray-800), var(--ig-gray-a));
  background: hsla(var(--ig-gray-200), var(--ig-gray-a));
  border-color: hsla(var(--ig-gray-200), var(--ig-gray-a));
}

[part~=date-inner][part~=selected][part~=single],
[part~=date-inner][part~=selected][part~=weekend][part~=single],
[part~=date-inner][part~=selected][part~=weekend][part~=current][part~=single],
[part~=date-inner][part~=selected][part~=current][part~=single] {
  background: hsla(var(--ig-primary-500), var(--ig-primary-a));
  border-color: hsla(var(--ig-primary-500), var(--ig-primary-a));
  color: var(--ig-primary-600-contrast);
}
[part~=date-inner][part~=selected][part~=single]:focus, [part~=date-inner][part~=selected][part~=single]:hover,
[part~=date-inner][part~=selected][part~=weekend][part~=single]:focus,
[part~=date-inner][part~=selected][part~=weekend][part~=single]:hover,
[part~=date-inner][part~=selected][part~=weekend][part~=current][part~=single]:focus,
[part~=date-inner][part~=selected][part~=weekend][part~=current][part~=single]:hover,
[part~=date-inner][part~=selected][part~=current][part~=single]:focus,
[part~=date-inner][part~=selected][part~=current][part~=single]:hover {
  background: hsla(var(--ig-primary-500), var(--ig-primary-a));
  border-color: hsla(var(--ig-primary-500), var(--ig-primary-a));
  color: var(--ig-primary-600-contrast);
}

[part~=date-inner][part~=first][part~=last][part~=selected],
[part~=date-inner][part~=first][part~=selected][part~=range],
[part~=date-inner][part~=first][part~=selected][part~=range][part~=preview],
[part~=date-inner][part~=last][part~=selected][part~=range],
[part~=date-inner][part~=last][part~=selected][part~=range][part~=preview],
[part~=date-inner][part~=selected][part~=range],
[part~=date-inner][part~=selected][part~=range][part~=preview] {
  border-radius: 0 !important;
}

[part~=date-inner][part~=selected][part~=range],
[part~=date-inner][part~=selected][part~=range][part~=weekend],
[part~=date-inner][part~=selected][part~=range][part~=preview],
[part~=date-inner][part~=selected][part~=range][part~=current] {
  background: hsla(var(--ig-primary-200), var(--ig-primary-a));
  border-color: hsla(var(--ig-primary-200), var(--ig-primary-a));
  color: var(--ig-primary-600-contrast);
}
[part~=date-inner][part~=selected][part~=range]:hover, [part~=date-inner][part~=selected][part~=range]:focus,
[part~=date-inner][part~=selected][part~=range][part~=weekend]:hover,
[part~=date-inner][part~=selected][part~=range][part~=weekend]:focus,
[part~=date-inner][part~=selected][part~=range][part~=preview]:hover,
[part~=date-inner][part~=selected][part~=range][part~=preview]:focus,
[part~=date-inner][part~=selected][part~=range][part~=current]:hover,
[part~=date-inner][part~=selected][part~=range][part~=current]:focus {
  background: hsla(var(--ig-primary-500), var(--ig-primary-a));
  border-color: hsla(var(--ig-primary-500), var(--ig-primary-a));
  color: var(--ig-primary-600-contrast);
}

[part~=date][part~=range],
[part~=date][part~=range][part~=preview] {
  border-top-style: solid;
  border-bottom-style: solid;
  background: hsla(var(--ig-primary-200), var(--ig-primary-a));
  border-top-color: hsla(var(--ig-primary-200), var(--ig-primary-a));
  border-bottom-color: hsla(var(--ig-primary-200), var(--ig-primary-a));
}

[part~=date][part~=first][part~=range],
[part~=date][part~=last][part~=range] {
  background: transparent;
  border-top-color: transparent;
  border-bottom-color: transparent;
}

[part~=date-inner][part~=range][part~=preview],
[part~=date-inner][part~=range] {
  color: var(--ig-primary-600-contrast);
}

[part~=date][part~=first][part~=selected][part~=range]::after,
[part~=date][part~=first][part~=range][part~=preview]::after,
[part~=date][part~=last][part~=selected][part~=range]::after,
[part~=date][part~=last][part~=range][part~=preview]::after {
  background: hsla(var(--ig-primary-200), var(--ig-primary-a));
  border-color: hsla(var(--ig-primary-200), var(--ig-primary-a));
}

[part~=date-inner][part~=current][part~=selected],
[part~=date-inner][part~=current][part~=range][part~=selected],
[part~=date-inner][part~=current][part~=range][part~=preview] {
  border-color: hsla(var(--ig-primary-200), var(--ig-primary-a));
  background: hsla(var(--ig-primary-200), var(--ig-primary-a));
  color: hsla(var(--ig-gray-800), var(--ig-gray-a));
  position: relative;
}
[part~=date-inner][part~=current][part~=selected]:hover,
[part~=date-inner][part~=current][part~=range][part~=selected]:hover,
[part~=date-inner][part~=current][part~=range][part~=preview]:hover {
  border-color: hsla(var(--ig-primary-200), var(--ig-primary-a));
  background: hsla(var(--ig-primary-200), var(--ig-primary-a));
  color: hsla(var(--ig-gray-800), var(--ig-gray-a));
}
[part~=date-inner][part~=current][part~=selected]::after,
[part~=date-inner][part~=current][part~=range][part~=selected]::after,
[part~=date-inner][part~=current][part~=range][part~=preview]::after {
  content: "";
  position: absolute;
  width: calc(100% - 4px);
  height: calc(100% - 4px);
  border-radius: clamp(0rem, calc(var(--ig-radius-factor) * 0.25rem), 0.25rem);
  background: hsla(var(--ig-gray-200), var(--ig-gray-a));
  z-index: -1;
}

[part~=date-inner][part~=current][part~=selected] {
  -webkit-box-shadow: none;
          box-shadow: none;
}

[part~=date-inner][part~=current][part~=single]:focus, [part~=date-inner][part~=current][part~=single]:hover {
  -webkit-box-shadow: none;
          box-shadow: none;
}

[part~=date-inner][part~=current][part~=selected][part~=single] {
  color: hsla(var(--ig-gray-800), var(--ig-gray-a));
  background: hsla(var(--ig-primary-500), var(--ig-primary-a));
  border-color: hsla(var(--ig-primary-500), var(--ig-primary-a));
}
[part~=date-inner][part~=current][part~=selected][part~=single]:focus, [part~=date-inner][part~=current][part~=selected][part~=single]:hover {
  color: hsla(var(--ig-gray-800), var(--ig-gray-a));
  background: hsla(var(--ig-primary-500), var(--ig-primary-a));
  border-color: hsla(var(--ig-primary-500), var(--ig-primary-a));
}

[part~=date-inner][part~=last][part~=range][part~=preview],
[part~=date-inner][part~=first][part~=range][part~=preview],
[part~=date-inner][part~=first][part~=last][part~=selected],
[part~=date-inner][part~=first][part~=selected][part~=range],
[part~=date-inner][part~=last][part~=selected][part~=range],
[part~=date-inner][part~=first][part~=range][part~=pewview],
[part~=date-inner][part~=first][part~=range][part~=current][part~=preview],
[part~=date-inner][part~=last][part~=range][part~=pewview],
[part~=date-inner][part~=last][part~=range][part~=current][part~=preview] {
  border-radius: 0 !important;
  background: hsla(var(--ig-primary-500), var(--ig-primary-a));
  border-color: hsla(var(--ig-primary-500), var(--ig-primary-a));
  color: var(--ig-primary-600-contrast);
}
[part~=date-inner][part~=last][part~=range][part~=preview]:hover, [part~=date-inner][part~=last][part~=range][part~=preview]:focus,
[part~=date-inner][part~=first][part~=range][part~=preview]:hover,
[part~=date-inner][part~=first][part~=range][part~=preview]:focus,
[part~=date-inner][part~=first][part~=last][part~=selected]:hover,
[part~=date-inner][part~=first][part~=last][part~=selected]:focus,
[part~=date-inner][part~=first][part~=selected][part~=range]:hover,
[part~=date-inner][part~=first][part~=selected][part~=range]:focus,
[part~=date-inner][part~=last][part~=selected][part~=range]:hover,
[part~=date-inner][part~=last][part~=selected][part~=range]:focus,
[part~=date-inner][part~=first][part~=range][part~=pewview]:hover,
[part~=date-inner][part~=first][part~=range][part~=pewview]:focus,
[part~=date-inner][part~=first][part~=range][part~=current][part~=preview]:hover,
[part~=date-inner][part~=first][part~=range][part~=current][part~=preview]:focus,
[part~=date-inner][part~=last][part~=range][part~=pewview]:hover,
[part~=date-inner][part~=last][part~=range][part~=pewview]:focus,
[part~=date-inner][part~=last][part~=range][part~=current][part~=preview]:hover,
[part~=date-inner][part~=last][part~=range][part~=current][part~=preview]:focus {
  background: hsla(var(--ig-primary-500), var(--ig-primary-a));
  border-color: hsla(var(--ig-primary-500), var(--ig-primary-a));
  color: var(--ig-primary-600-contrast);
}

[part~=date-inner][part~=first][part~=last][part~=selected][part~=current]:hover, [part~=date-inner][part~=first][part~=last][part~=selected][part~=current]:focus,
[part~=date-inner][part~=last][part~=range][part~=current][part~=preview]:hover,
[part~=date-inner][part~=last][part~=range][part~=current][part~=preview]:focus,
[part~=date-inner][part~=first][part~=range][part~=current][part~=preview]:hover,
[part~=date-inner][part~=first][part~=range][part~=current][part~=preview]:focus {
  color: hsla(var(--ig-gray-800), var(--ig-gray-a));
}

[part~=date-inner][part~=first][part~=current][part~=range][part~=preview],
[part~=date-inner][part~=first][part~=current][part~=range][part~=selected],
[part~=date-inner][part~=last][part~=current][part~=range][part~=preview],
[part~=date-inner][part~=last][part~=current][part~=range][part~=selected] {
  background: hsla(var(--ig-primary-500), var(--ig-primary-a));
  border-color: hsla(var(--ig-primary-500), var(--ig-primary-a));
}

[part~=date-inner][part~=first][part~=selected][part~=current][part~=range],
[part~=date-inner][part~=last][part~=selected][part~=current][part~=range] {
  background: hsla(var(--ig-primary-500), var(--ig-primary-a));
  border-color: hsla(var(--ig-primary-500), var(--ig-primary-a));
  color: hsla(var(--ig-gray-800), var(--ig-gray-a));
}
[part~=date-inner][part~=first][part~=selected][part~=current][part~=range]:hover, [part~=date-inner][part~=first][part~=selected][part~=current][part~=range]:focus,
[part~=date-inner][part~=last][part~=selected][part~=current][part~=range]:hover,
[part~=date-inner][part~=last][part~=selected][part~=current][part~=range]:focus {
  background: hsla(var(--ig-primary-500), var(--ig-primary-a));
  border-color: hsla(var(--ig-primary-500), var(--ig-primary-a));
  color: hsla(var(--ig-gray-800), var(--ig-gray-a));
}

[part~=date-inner][part~=selected][part~=current][part~=range] {
  color: hsla(var(--ig-gray-800), var(--ig-gray-a));
}
[part~=date-inner][part~=selected][part~=current][part~=range]:hover, [part~=date-inner][part~=selected][part~=current][part~=range]:focus {
  background: hsla(var(--ig-primary-500), var(--ig-primary-a));
  border-color: hsla(var(--ig-primary-500), var(--ig-primary-a));
  color: hsla(var(--ig-gray-800), var(--ig-gray-a));
}

[part~=week-number-inner] {
  font-style: italic;
  background: transparent;
  border: none;
}
[part~=week-number-inner]::after {
  display: none;
}

[part~=date-inner][part~=special] {
  background: hsla(var(--ig-primary-300), var(--ig-primary-a));
  border-color: hsla(var(--ig-primary-300), var(--ig-primary-a));
  color: var(--ig-primary-600-contrast);
}

[part~=date-inner][part~=special][part~=selected] {
  background: hsla(var(--ig-primary-500), var(--ig-primary-a));
  border-color: hsla(var(--ig-primary-500), var(--ig-primary-a));
  color: var(--ig-primary-600-contrast);
}`;
