// igniteui-webcomponents/components/navbar/themes/light/navbar.base.css.js
import { css } from 'lit';
export const styles = css `:host {
  --is-large: clamp(0, (var(--component-size, 1) + 1) - var(--ig-size-large, 3), 1);
  --is-medium:
      min(
          clamp(0, (var(--component-size, 1) + 1) - var(--ig-size-medium, 2), 1),
          clamp(0, var(--ig-size-large, 3) - var(--component-size, 1), 1)
      );
  --is-small: clamp(0, var(--ig-size-medium) - var(--component-size, 1), 1);
  position: relative;
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
  scrollbar-width: var(--ig-scrollbar-size);
  scrollbar-color: var(--ig-scrollbar-thumb-background) var(--ig-scrollbar-track-background);
}
:host *,
:host *::before,
:host *::after {
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
}
:host ::-webkit-scrollbar {
  width: var(--ig-scrollbar-size);
  height: var(--ig-scrollbar-size);
  background: var(--ig-scrollbar-track-background);
}
:host ::-webkit-scrollbar-thumb {
  background: var(--ig-scrollbar-thumb-background);
}

[hidden] {
  display: none !important;
}

/* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */
:host {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  min-height: 3.5rem;
  max-height: 8rem;
  padding: 0 1rem;
  background: hsla(var(--ig-primary-500), var(--ig-primary-a));
  color: var(--ig-primary-600-contrast);
  -webkit-box-shadow: var(--ig-elevation-4);
          box-shadow: var(--ig-elevation-4);
  font-family: var(--ig-font-family);
  z-index: 4;
}

[part=base] {
  display: inherit;
  position: relative;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
      -ms-flex-flow: row nowrap;
          flex-flow: row nowrap;
  -webkit-box-flex: 1;
      -ms-flex-positive: 1;
          flex-grow: 1;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
}

[part=middle] {
  display: -webkit-inline-box;
  -webkit-line-clamp: 4;
  -webkit-box-orient: vertical;
  overflow: hidden;
  -webkit-user-select: text;
     -moz-user-select: text;
      -ms-user-select: text;
          user-select: text;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex: 1;
      -ms-flex-positive: 1;
          flex-grow: 1;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  overflow: visible;
}

[part=start],
[part=end] {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  -webkit-user-select: none;
     -moz-user-select: none;
      -ms-user-select: none;
          user-select: none;
  overflow: visible;
}

[name=start]::slotted(*) {
  -webkit-margin-end: 1rem;
          margin-inline-end: 1rem;
  -webkit-margin-start: 0.0625rem;
          margin-inline-start: 0.0625rem;
}

[name=end]::slotted(*) {
  -webkit-margin-start: 1rem;
          margin-inline-start: 1rem;
  -webkit-margin-end: 0.0625rem;
          margin-inline-end: 0.0625rem;
}

::slotted(igc-avatar[part=base]) {
  max-width: 2rem;
  max-height: 2rem;
  min-height: 0;
  min-width: 0;
}

::slotted(igc-icon) {
  cursor: pointer;
  color: var(--ig-primary-600-contrast);
  width: 2rem;
  height: 2rem;
  font-size: 2rem;
}

::slotted(*) {
  font-family: var(--ig-h6-font-family, var(--ig-font-family));
  font-size: var(--ig-h6-font-size);
  font-weight: var(--ig-h6-font-weight);
  font-style: var(--ig-h6-font-style);
  line-height: var(--ig-h6-line-height);
  letter-spacing: var(--ig-h6-letter-spacing);
  text-transform: var(--ig-h6-text-transform);
  margin-top: var(--ig-h6-margin-top);
  margin-bottom: var(--ig-h6-margin-bottom);
  line-height: normal;
  margin: 0;
}`;
