// igniteui-webcomponents/components/calendar/themes/fluent/days-view.fluent.css.js
import { css } from 'lit';
export const styles = css `/* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */
[part~=date-inner]:focus, [part~=date-inner]:hover {
  background: hsla(var(--ig-gray-300), var(--ig-gray-a));
  border-color: hsla(var(--ig-gray-300), var(--ig-gray-a));
  color: var(--date-color-hover, var(--ig-gray-300-contrast));
}

[part~=date-inner][part~=weekend]:focus, [part~=date-inner][part~=weekend]:hover,
[part~=date-inner][part~=inactive]:focus,
[part~=date-inner][part~=inactive]:hover,
[part~=date-inner][part~=selected][part~=single]:focus,
[part~=date-inner][part~=selected][part~=single]:hover {
  background: hsla(var(--ig-gray-300), var(--ig-gray-a));
  border-color: hsla(var(--ig-gray-700), var(--ig-gray-a));
  color: var(--date-color-hover, var(--ig-gray-300-contrast));
}

[part~=current][part~=date-inner][part~=current],
[part~=current][part~=date-inner],
[part~=current][part~=single][part~=date-inner] {
  -webkit-box-shadow: none;
          box-shadow: none;
  position: relative;
  overflow: hidden;
  color: var(--date-current-color, var(--ig-secondary-700-contrast));
}
[part~=current][part~=date-inner][part~=current]::after,
[part~=current][part~=date-inner]::after,
[part~=current][part~=single][part~=date-inner]::after {
  border-radius: clamp(0rem, calc(var(--ig-radius-factor) * 1rem), 1rem);
  content: "";
  position: absolute;
  top: 0.0625rem;
  inset-inline-start: 0.0625rem;
  width: calc(100% - 0.125rem);
  height: calc(100% - 0.125rem);
  background: hsla(var(--ig-secondary-500), var(--ig-secondary-a));
  z-index: -1;
}
[part~=current][part~=date-inner][part~=current]:focus, [part~=current][part~=date-inner][part~=current]:hover,
[part~=current][part~=date-inner]:focus,
[part~=current][part~=date-inner]:hover,
[part~=current][part~=single][part~=date-inner]:focus,
[part~=current][part~=single][part~=date-inner]:hover {
  color: var(--date-current-color, var(--ig-secondary-700-contrast));
}
[part~=current][part~=date-inner][part~=current]:focus::after, [part~=current][part~=date-inner][part~=current]:hover::after,
[part~=current][part~=date-inner]:focus::after,
[part~=current][part~=date-inner]:hover::after,
[part~=current][part~=single][part~=date-inner]:focus::after,
[part~=current][part~=single][part~=date-inner]:hover::after {
  background: hsla(var(--ig-secondary-500), var(--ig-secondary-a));
}

[part~=selected][part~=date-inner][part~=single],
[part~=selected][part~=date-inner][part~=single][part~=weekend] {
  border-color: hsla(var(--ig-gray-700), var(--ig-gray-a));
  background: hsla(var(--ig-gray-200), var(--ig-gray-a));
  color: var(--ig-gray-200-contrast);
  border-radius: 0;
}

[part~=date-inner][part~=selected][part~=current][part~=single] {
  color: var(--date-current-color, var(--ig-secondary-700-contrast));
}

[part~=week-number-inner] {
  border-inline-end-color: hsla(var(--ig-gray-700), var(--ig-gray-a));
}
[part~=week-number-inner]::after {
  border-inline-end-color: hsla(var(--ig-gray-700), var(--ig-gray-a));
}

[part="week-number-inner last"],
[part="week-number-inner first"] {
  border-radius: 0;
}

[part~=date-inner][part~=selected][part~=range][part~=first],
[part~=date-inner][part~=selected][part~=range][part~=last],
[part~=date-inner][part~=selected]:not([part~=range]) {
  border-top-color: hsla(var(--ig-gray-700), var(--ig-gray-a));
  border-bottom-color: hsla(var(--ig-gray-700), var(--ig-gray-a));
  border-inline-start-color: hsla(var(--ig-gray-700), var(--ig-gray-a));
}
[part~=date-inner][part~=selected][part~=range][part~=first]:hover, [part~=date-inner][part~=selected][part~=range][part~=first]:focus,
[part~=date-inner][part~=selected][part~=range][part~=last]:hover,
[part~=date-inner][part~=selected][part~=range][part~=last]:focus,
[part~=date-inner][part~=selected]:not([part~=range]):hover,
[part~=date-inner][part~=selected]:not([part~=range]):focus {
  border-color: hsla(var(--ig-gray-700), var(--ig-gray-a));
}
[part~=date-inner][part~=selected][part~=range][part~=first]:hover,
[part~=date-inner][part~=selected][part~=range][part~=last]:hover,
[part~=date-inner][part~=selected]:not([part~=range]):hover {
  background: hsla(var(--ig-gray-300), var(--ig-gray-a));
}
[part~=date-inner][part~=selected][part~=range][part~=first]:focus,
[part~=date-inner][part~=selected][part~=range][part~=last]:focus,
[part~=date-inner][part~=selected]:not([part~=range]):focus {
  background: hsla(var(--ig-gray-200), var(--ig-gray-a));
}

[part~=date-inner][part~=selected][part~=range][part~=last] {
  border-inline-start-color: transparent;
  -webkit-border-end: 0;
          border-inline-end: 0;
}

[part~=date-inner][part~=selected][part~=range][part~=first] {
  border-inline-end-color: transparent;
  -webkit-border-start: 0;
          border-inline-start: 0;
}

[part~=date-inner][part~=selected][part~=range][part~=weekend],
[part~=date-inner][part~=selected][part~=range][part~=inactive] {
  color: var(--ig-gray-200-contrast);
}
[part~=date-inner][part~=selected][part~=range][part~=weekend]:hover, [part~=date-inner][part~=selected][part~=range][part~=weekend]:focus,
[part~=date-inner][part~=selected][part~=range][part~=inactive]:hover,
[part~=date-inner][part~=selected][part~=range][part~=inactive]:focus {
  color: var(--ig-gray-300-contrast);
  background: hsla(var(--ig-gray-300), var(--ig-gray-a));
  border-color: hsla(var(--ig-gray-700), var(--ig-gray-a));
}

[part~=date-inner][part~=last][part~=range] {
  border-color: transparent;
}

[part~=date-inner][part~=selected][part~=range] {
  border-top: 0;
  border-bottom: 0;
  color: var(--ig-gray-200-contrast);
}
[part~=date-inner][part~=selected][part~=range]:hover {
  color: var(--date-color-hover, var(--ig-gray-300-contrast));
  background: hsla(var(--ig-gray-300), var(--ig-gray-a));
  border-color: hsla(var(--ig-gray-700), var(--ig-gray-a));
}
[part~=date-inner][part~=selected][part~=range]:focus {
  color: var(--date-color-hover, var(--ig-gray-300-contrast));
  background: hsla(var(--ig-gray-200), var(--ig-gray-a));
  border-color: hsla(var(--ig-gray-700), var(--ig-gray-a));
}

[part~=date-inner][part~=selected][part~=range][part~=preview]:focus {
  background: hsla(var(--ig-gray-200), var(--ig-gray-a));
  border-color: hsla(var(--ig-gray-200), var(--ig-gray-a));
}

[part~=date][part~=selected][part~=range] {
  background: hsla(var(--ig-gray-200), var(--ig-gray-a));
  border-top-color: hsla(var(--ig-gray-700), var(--ig-gray-a));
  border-bottom-color: hsla(var(--ig-gray-700), var(--ig-gray-a));
}

[part~=date][part~=first][part~=selected][part~=range],
[part~=date][part~=last][part~=selected][part~=range] {
  background: transparent;
  border-color: transparent;
}
[part~=date][part~=first][part~=selected][part~=range]::after,
[part~=date][part~=last][part~=selected][part~=range]::after {
  background: hsla(var(--ig-gray-200), var(--ig-gray-a));
  border-top-color: hsla(var(--ig-gray-700), var(--ig-gray-a));
  border-bottom-color: hsla(var(--ig-gray-700), var(--ig-gray-a));
}
[part~=date][part~=first][part~=selected][part~=range]::before,
[part~=date][part~=last][part~=selected][part~=range]::before {
  content: "";
  position: absolute;
  top: -0.0625rem;
  width: var(--size);
  height: 100%;
  border: 0.0625rem solid hsla(var(--ig-gray-700), var(--ig-gray-a));
  z-index: 3;
  pointer-events: none;
}

[part~=date][part~=first][part~=selected][part~=range]::before {
  border-inline-end-color: transparent;
}

[part~=date][part~=last][part~=selected][part~=range]::before {
  border-inline-start-color: transparent;
}

:host(igc-calendar[size=large]) igc-days-view::part(date first selected range)::before {
  margin: 0;
  inset-inline-start: calc(50% - 1rem);
}
:host(igc-calendar[size=large]) igc-days-view::part(date last selected range)::before {
  inset-inline-end: calc(50% - 1rem);
}

:host(igc-calendar[size=medium]) igc-days-view::part(date first selected range)::before {
  inset-inline-start: calc(50% - 0.875rem);
}
:host(igc-calendar[size=medium]) igc-days-view::part(date last selected range)::before {
  inset-inline-end: calc(50% - 0.875rem);
}

:host(igc-calendar[size=small]) igc-days-view::part(date first selected range)::before {
  inset-inline-start: calc(50% - 0.75rem);
}
:host(igc-calendar[size=small]) igc-days-view::part(date last selected range)::before {
  inset-inline-end: calc(50% - 0.75rem);
}

[part~=date][part~=first][part~=last][part~=selected][part~=range]::before {
  border-color: hsla(var(--ig-gray-200), var(--ig-gray-a));
  border-inline-end-color: transparent;
}

[part~=date-inner][part~=selected][part~=current][part~=range],
[part~=date-inner][part~=first][part~=last][part~=selected][part~=current][part~=range],
[part~=date-inner][part~=first][part~=last][part~=selected][part~=current],
[part~=date-inner][part~=first][part~=selected][part~=current][part~=range],
[part~=date-inner][part~=last][part~=selected][part~=current][part~=range] {
  color: var(--date-current-color, var(--ig-secondary-700-contrast)) !important;
}
[part~=date-inner][part~=selected][part~=current][part~=range]:focus, [part~=date-inner][part~=selected][part~=current][part~=range]:hover,
[part~=date-inner][part~=first][part~=last][part~=selected][part~=current][part~=range]:focus,
[part~=date-inner][part~=first][part~=last][part~=selected][part~=current][part~=range]:hover,
[part~=date-inner][part~=first][part~=last][part~=selected][part~=current]:focus,
[part~=date-inner][part~=first][part~=last][part~=selected][part~=current]:hover,
[part~=date-inner][part~=first][part~=selected][part~=current][part~=range]:focus,
[part~=date-inner][part~=first][part~=selected][part~=current][part~=range]:hover,
[part~=date-inner][part~=last][part~=selected][part~=current][part~=range]:focus,
[part~=date-inner][part~=last][part~=selected][part~=current][part~=range]:hover {
  border-color: hsla(var(--ig-gray-700), var(--ig-gray-a));
}

[part~=date-inner][part~=first][part~=range][part~=preview],
[part~=date-inner][part~=first][part~=selected][part~=range][part~=preview] {
  border-top-color: hsla(var(--ig-gray-200), var(--ig-gray-a));
  border-bottom-color: hsla(var(--ig-gray-200), var(--ig-gray-a));
  border-inline-start-color: hsla(var(--ig-gray-200), var(--ig-gray-a));
  border-inline-end-color: transparent;
  background: transparent;
}

[part~=date-inner][part~=last][part~=range][part~=preview],
[part~=date-inner][part~=last][part~=selected][part~=range][part~=preview] {
  border-top-color: hsla(var(--ig-gray-200), var(--ig-gray-a));
  border-bottom-color: hsla(var(--ig-gray-200), var(--ig-gray-a));
  border-inline-end-color: hsla(var(--ig-gray-200), var(--ig-gray-a));
  border-inline-start-color: transparent;
  background: transparent;
}

[part~=date][part~=range][part~=preview] {
  border-top-style: solid;
  border-bottom-style: solid;
  border-top-color: hsla(var(--ig-gray-200), var(--ig-gray-a));
  border-bottom-color: hsla(var(--ig-gray-200), var(--ig-gray-a));
}
[part~=date][part~=range][part~=preview]::after {
  border-top-style: solid;
  border-bottom-style: solid;
  border-top-color: hsla(var(--ig-gray-200), var(--ig-gray-a));
  border-bottom-color: hsla(var(--ig-gray-200), var(--ig-gray-a));
  background: transparent;
}

[part~=date][part~=range][part~=preview][part~=first],
[part~=date][part~=range][part~=preview][part~=last] {
  border-top-color: transparent;
  border-bottom-color: transparent;
}
[part~=date][part~=range][part~=preview][part~=first]::after,
[part~=date][part~=range][part~=preview][part~=last]::after {
  border-top-style: solid;
  border-bottom-style: solid;
  border-top-color: hsla(var(--ig-gray-200), var(--ig-gray-a));
  border-bottom-color: hsla(var(--ig-gray-200), var(--ig-gray-a));
}

[part~=date][part~=first][part~=selected][part~=range][part~=preview],
[part~=date][part~=last][part~=selected][part~=range][part~=preview] {
  background: transparent;
}
[part~=date][part~=first][part~=selected][part~=range][part~=preview]::before,
[part~=date][part~=last][part~=selected][part~=range][part~=preview]::before {
  border-block-start-color: hsla(var(--ig-gray-200), var(--ig-gray-a));
  border-block-end-color: hsla(var(--ig-gray-200), var(--ig-gray-a));
}
[part~=date][part~=first][part~=selected][part~=range][part~=preview]::after,
[part~=date][part~=last][part~=selected][part~=range][part~=preview]::after {
  border-style: solid;
  border-top-color: hsla(var(--ig-gray-200), var(--ig-gray-a));
  border-bottom-color: hsla(var(--ig-gray-200), var(--ig-gray-a));
  background: transparent;
}

[part~=date][part~=first][part~=selected][part~=range][part~=preview]::before {
  border-inline-start-color: hsla(var(--ig-gray-200), var(--ig-gray-a));
  border-inline-end-color: transparent;
}
[part~=date][part~=first][part~=selected][part~=range][part~=preview]::after {
  -webkit-border-end: 0;
          border-inline-end: 0;
}

[part~=date][part~=last][part~=selected][part~=range][part~=preview]::before {
  border-inline-end-color: hsla(var(--ig-gray-200), var(--ig-gray-a));
  border-inline-start-color: transparent;
}
[part~=date][part~=last][part~=selected][part~=range][part~=preview]::after {
  -webkit-border-start: 0;
          border-inline-start: 0;
}

[part~=date-inner][part~=first][part~=last][part~=selected] {
  border-color: hsla(var(--ig-gray-700), var(--ig-gray-a));
}

[part~=date-inner][part~=first][part~=last][part~=selected],
[part~=date-inner][part~=first][part~=selected],
[part~=date-inner][part~=last][part~=selected] {
  background: hsla(var(--ig-gray-200), var(--ig-gray-a)) !important;
  color: var(--ig-gray-200-contrast) !important;
}
[part~=date-inner][part~=first][part~=last][part~=selected]:focus,
[part~=date-inner][part~=first][part~=selected]:focus,
[part~=date-inner][part~=last][part~=selected]:focus {
  background: hsla(var(--ig-gray-200), var(--ig-gray-a)) !important;
  color: var(--ig-gray-200-contrast) !important;
}

[part~=date-inner][part~=first][part~=last][part~=selected][part~=current] {
  color: var(--date-current-color, var(--ig-secondary-700-contrast)) !important;
}

[part~=date-inner][part~=first][part~=selected][part~=current][part~=range][part~=preview]:focus,
[part~=date-inner][part~=last][part~=selected][part~=current][part~=range][part~=preview]:focus {
  border-color: transparent;
}`;
