// igniteui-webcomponents/components/tabs/themes/light/tab.bootstrap.css.js
import { css } from 'lit';
export const styles = css `/* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */
:host {
  background: transparent;
  -webkit-transition: none;
  -o-transition: none;
  transition: none;
  border-start-start-radius: 0.25rem;
  border-start-end-radius: 0.25rem;
  color: hsla(var(--ig-primary-500), var(--ig-primary-a));
}

:host(:hover) {
  background: transparent;
  color: var(--hover-color, hsla(var(--ig-primary-700), var(--ig-primary-a)));
  -webkit-box-shadow: inset 0 0 0 0.0625rem hsla(var(--ig-gray-200), var(--ig-gray-a));
          box-shadow: inset 0 0 0 0.0625rem hsla(var(--ig-gray-200), var(--ig-gray-a));
}

:host([selected]) {
  color: hsla(var(--ig-gray-800), var(--ig-gray-a));
  position: relative;
  -webkit-box-shadow: inset 0 0 0 0.0625rem hsla(var(--ig-gray-300), var(--ig-gray-a));
          box-shadow: inset 0 0 0 0.0625rem hsla(var(--ig-gray-300), var(--ig-gray-a));
}
:host([selected]) ::slotted(igc-icon) {
  color: hsla(var(--ig-gray-800), var(--ig-gray-a));
}
:host([selected])::after {
  content: "";
  position: absolute;
  bottom: 0;
  inset-inline-start: 0;
  width: 100%;
  height: 0.0625rem;
  background: hsla(var(--ig-surface-500), var(--ig-surface-a));
  z-index: 1;
}

:host([selected]:hover) {
  color: hsla(var(--ig-gray-800), var(--ig-gray-a));
  border-top-color: hsla(var(--ig-gray-300), var(--ig-gray-a));
  border-inline-color: hsla(var(--ig-gray-300), var(--ig-gray-a));
  border-bottom-color: hsla(var(--ig-surface-500), var(--ig-surface-a));
}

:host(:focus-within),
:host([selected]:focus-within) {
  color: var(--hover-color, hsla(var(--ig-primary-700), var(--ig-primary-a)));
  background: transparent;
  -webkit-box-shadow: inset 0 0 0 0.125rem hsla(var(--ig-primary-700), var(--ig-primary-a));
          box-shadow: inset 0 0 0 0.125rem hsla(var(--ig-primary-700), var(--ig-primary-a));
  border-radius: 0.25rem;
  z-index: 1;
}
:host(:focus-within)::after,
:host([selected]:focus-within)::after {
  display: none;
}

:host([selected]:focus-within) {
  color: hsla(var(--ig-gray-800), var(--ig-gray-a));
}

:host([disabled]) {
  color: hsla(var(--ig-primary-500), 0.5);
}`;
