// igniteui-webcomponents/components/tabs/themes/light/tabs.base.css.js
import { css } from 'lit';
export const styles = css `:host {
  --is-large: clamp(0, (var(--component-size, 1) + 1) - var(--ig-size-large, 3), 1);
  --is-medium:
      min(
          clamp(0, (var(--component-size, 1) + 1) - var(--ig-size-medium, 2), 1),
          clamp(0, var(--ig-size-large, 3) - var(--component-size, 1), 1)
      );
  --is-small: clamp(0, var(--ig-size-medium) - var(--component-size, 1), 1);
  position: relative;
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
  scrollbar-width: var(--ig-scrollbar-size);
  scrollbar-color: var(--ig-scrollbar-thumb-background) var(--ig-scrollbar-track-background);
}
:host *,
:host *::before,
:host *::after {
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
}
:host ::-webkit-scrollbar {
  width: var(--ig-scrollbar-size);
  height: var(--ig-scrollbar-size);
  background: var(--ig-scrollbar-track-background);
}
:host ::-webkit-scrollbar-thumb {
  background: var(--ig-scrollbar-thumb-background);
}

[hidden] {
  display: none !important;
}

/* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */
:host {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
  overflow: hidden;
}

:host([alignment=start]) [part=headers-scroll] ::slotted(igc-tab:last-of-type) {
  -webkit-margin-end: auto;
          margin-inline-end: auto;
}

:host([alignment=end]) [part=headers-scroll] ::slotted(igc-tab:first-of-type) {
  -webkit-margin-start: auto;
          margin-inline-start: auto;
}

:host([alignment=center]) [part=headers-scroll] ::slotted(igc-tab:first-of-type) {
  -webkit-margin-start: auto;
          margin-inline-start: auto;
}
:host([alignment=center]) [part=headers-scroll] ::slotted(igc-tab:last-of-type) {
  -webkit-margin-end: auto;
          margin-inline-end: auto;
}

:host([alignment=justify]) [part=headers-scroll] ::slotted(igc-tab) {
  -ms-flex-preferred-size: 100px;
      flex-basis: 100px;
  -webkit-box-flex: 1;
      -ms-flex-positive: 1;
          flex-grow: 1;
  max-width: 100%;
}

[part=headers] {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  overflow: hidden;
  min-height: 2.625rem;
  background: hsla(var(--ig-surface-500), var(--ig-surface-a));
  -webkit-box-flex: 0;
      -ms-flex: none;
          flex: none;
}

[part=headers-content] {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex: 1;
      -ms-flex: 1 1 auto;
          flex: 1 1 auto;
  scroll-behavior: smooth;
  overflow: hidden;
}

[part=headers-wrapper] {
  position: relative;
  -webkit-box-flex: 1;
      -ms-flex-positive: 1;
          flex-grow: 1;
}

[part=headers-scroll] {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
}

[part=content] {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
  -webkit-box-flex: 1;
      -ms-flex-positive: 1;
          flex-grow: 1;
  overflow: hidden;
}

[part=selected-indicator] {
  position: absolute;
  inset-inline-start: 0;
  bottom: 0;
  -webkit-transform: translateX(0);
      -ms-transform: translateX(0);
          transform: translateX(0);
  height: 0.125rem;
  min-width: 5.625rem;
  background: hsla(var(--ig-primary-500), var(--ig-primary-a));
  -webkit-transition: width 0.2s cubic-bezier(0.35, 0, 0.25, 1), -webkit-transform 0.3s cubic-bezier(0.35, 0, 0.25, 1);
  transition: width 0.2s cubic-bezier(0.35, 0, 0.25, 1), -webkit-transform 0.3s cubic-bezier(0.35, 0, 0.25, 1);
  -o-transition: transform 0.3s cubic-bezier(0.35, 0, 0.25, 1), width 0.2s cubic-bezier(0.35, 0, 0.25, 1);
  transition: transform 0.3s cubic-bezier(0.35, 0, 0.25, 1), width 0.2s cubic-bezier(0.35, 0, 0.25, 1);
  transition: transform 0.3s cubic-bezier(0.35, 0, 0.25, 1), width 0.2s cubic-bezier(0.35, 0, 0.25, 1), -webkit-transform 0.3s cubic-bezier(0.35, 0, 0.25, 1);
}

[part=start-scroll-button],
[part=end-scroll-button] {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
  z-index: 1;
  border: none;
  padding: 0;
  min-width: 3rem;
  width: 3rem;
  cursor: pointer;
  position: relative;
  background: hsla(var(--ig-surface-500), var(--ig-surface-a));
}
[part=start-scroll-button]:hover, [part=start-scroll-button]:focus,
[part=end-scroll-button]:hover,
[part=end-scroll-button]:focus {
  background: hsla(var(--ig-gray-100), var(--ig-gray-a));
}
[part=start-scroll-button] ::slotted(*),
[part=end-scroll-button] ::slotted(*) {
  -webkit-box-shadow: none;
          box-shadow: none;
}

igc-icon-button::part(base) {
  color: hsla(var(--ig-gray-700), var(--ig-gray-a));
}
igc-icon-button::part(base):hover, igc-icon-button::part(base):focus-within {
  color: hsla(var(--ig-gray-900), var(--ig-gray-a));
  background: transparent;
  -webkit-box-shadow: none;
          box-shadow: none;
}

igc-icon-button[disabled]::part(base) {
  color: hsla(var(--ig-gray-500), var(--ig-gray-a));
}

:host([dir=rtl]) [part=start-scroll-button],
:host([dir=rtl]) [part=end-scroll-button] {
  -webkit-transform: scaleX(-1);
      -ms-transform: scaleX(-1);
          transform: scaleX(-1);
}`;
