// igniteui-webcomponents/components/tabs/themes/light/tabs.bootstrap.css.js
import { css } from 'lit';
export const styles = css `/* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */
[part=headers] {
  background: hsla(var(--ig-surface-500), var(--ig-surface-a));
}

[part=headers-wrapper]::after {
  content: "";
  position: absolute;
  bottom: 0;
  inset-inline-start: 0;
  width: 100%;
  height: 0.0625rem;
  background: hsla(var(--ig-gray-300), var(--ig-gray-a));
  z-index: 0;
}

igc-icon-button::part(base) {
  color: hsla(var(--ig-gray-500), var(--ig-gray-a));
}
igc-icon-button::part(base):hover, igc-icon-button::part(base):focus-within {
  color: hsla(var(--ig-gray-600), var(--ig-gray-a));
}

igc-icon-button[disabled]::part(base) {
  color: hsla(var(--ig-primary-500), 0.5);
}

[part=selected-indicator] {
  display: none;
}`;
