// igniteui-webcomponents/components/nav-drawer/themes/light/nav-drawer-item.base.css.js
import { css } from 'lit';
export const styles = css `:host {
  --is-large: clamp(0, (var(--component-size, 1) + 1) - var(--ig-size-large, 3), 1);
  --is-medium:
      min(
          clamp(0, (var(--component-size, 1) + 1) - var(--ig-size-medium, 2), 1),
          clamp(0, var(--ig-size-large, 3) - var(--component-size, 1), 1)
      );
  --is-small: clamp(0, var(--ig-size-medium) - var(--component-size, 1), 1);
  position: relative;
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
  scrollbar-width: var(--ig-scrollbar-size);
  scrollbar-color: var(--ig-scrollbar-thumb-background) var(--ig-scrollbar-track-background);
}
:host *,
:host *::before,
:host *::after {
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
}
:host ::-webkit-scrollbar {
  width: var(--ig-scrollbar-size);
  height: var(--ig-scrollbar-size);
  background: var(--ig-scrollbar-track-background);
}
:host ::-webkit-scrollbar-thumb {
  background: var(--ig-scrollbar-thumb-background);
}

[hidden] {
  display: none !important;
}

/* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */
[part~=base] {
  border-radius: clamp(0rem, calc(var(--ig-radius-factor) * 0.25rem), 0.25rem);
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  gap: 2rem;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
      -ms-flex-flow: row nowrap;
          flex-flow: row nowrap;
  position: relative;
  color: hsla(var(--ig-gray-700), var(--ig-gray-a));
  max-height: 3rem;
  margin: 0.5rem;
  padding: 0.75rem 0.5rem;
  cursor: pointer;
  -webkit-user-select: none;
     -moz-user-select: none;
      -ms-user-select: none;
          user-select: none;
  outline: transparent;
  white-space: nowrap;
  text-decoration: none;
  width: calc(100% - 1rem);
  border: none;
  -webkit-box-pack: start;
      -ms-flex-pack: start;
          justify-content: flex-start;
  -webkit-transition: background 0.15s ease-out;
  -o-transition: background 0.15s ease-out;
  transition: background 0.15s ease-out;
}
[part~=base] ::slotted(igc-icon) {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  font: inherit;
}
[part~=base]:hover, [part~=base]:focus-within {
  background: var(--hover-background, hsla(var(--ig-gray-200), 0.8));
  color: hsla(var(--ig-gray-900), var(--ig-gray-a));
}

[part=content] {
  white-space: nowrap;
  -o-text-overflow: ellipsis;
     text-overflow: ellipsis;
  overflow: hidden;
  font-family: var(--ig-subtitle-2-font-family, var(--ig-font-family));
  font-size: var(--ig-subtitle-2-font-size);
  font-weight: var(--ig-subtitle-2-font-weight);
  font-style: var(--ig-subtitle-2-font-style);
  line-height: var(--ig-subtitle-2-line-height);
  letter-spacing: var(--ig-subtitle-2-letter-spacing);
  text-transform: var(--ig-subtitle-2-text-transform);
  margin-top: var(--ig-subtitle-2-margin-top);
  margin-bottom: var(--ig-subtitle-2-margin-bottom);
}

[part="base mini"] {
  display: block;
}
[part="base mini"] [part=content] {
  display: none;
}
[part="base mini"] ::slotted(igc-icon) {
  width: 100%;
}

:host([disabled]) {
  -webkit-user-select: none;
     -moz-user-select: none;
      -ms-user-select: none;
          user-select: none;
  cursor: initial;
  pointer-events: none;
}
:host([disabled]) [part~=base] {
  color: hsla(var(--ig-gray-400), var(--ig-gray-a));
}

:host([active]) [part~=base] {
  background: var(--active-background, hsla(var(--ig-primary-500), 0.12));
  color: var(--active-text-color, hsla(var(--ig-primary-500), var(--ig-primary-a)));
}`;
