// igniteui-webcomponents/components/nav-drawer/themes/light/nav-drawer.base.css.js
import { css } from 'lit';
export const styles = css `:host {
  --is-large: clamp(0, (var(--component-size, 1) + 1) - var(--ig-size-large, 3), 1);
  --is-medium:
      min(
          clamp(0, (var(--component-size, 1) + 1) - var(--ig-size-medium, 2), 1),
          clamp(0, var(--ig-size-large, 3) - var(--component-size, 1), 1)
      );
  --is-small: clamp(0, var(--ig-size-medium) - var(--component-size, 1), 1);
  position: relative;
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
  scrollbar-width: var(--ig-scrollbar-size);
  scrollbar-color: var(--ig-scrollbar-thumb-background) var(--ig-scrollbar-track-background);
}
:host *,
:host *::before,
:host *::after {
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
}
:host ::-webkit-scrollbar {
  width: var(--ig-scrollbar-size);
  height: var(--ig-scrollbar-size);
  background: var(--ig-scrollbar-track-background);
}
:host ::-webkit-scrollbar-thumb {
  background: var(--ig-scrollbar-thumb-background);
}

[hidden] {
  display: none !important;
}

/* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */
:host {
  -ms-flex-negative: 0;
      flex-shrink: 0;
  font-family: var(--ig-font-family);
}
:host [part=base] {
  position: fixed;
  height: 100%;
  min-height: 100%;
  overflow-x: hidden;
  top: 0;
  bottom: 0;
  inset-inline-start: 0;
  z-index: 999;
  background: hsla(var(--ig-surface-500), var(--ig-surface-a));
  border-radius: 0;
  -webkit-transition: all ease-out 0.3s, 0.3s;
  -o-transition: all ease-out 0.3s, 0.3s;
  transition: all ease-out 0.3s, 0.3s;
}
:host [part=main] {
  width: 16rem;
}
:host [part=mini] {
  width: 3.75rem;
}
:host [part=overlay] {
  background: hsla(var(--ig-gray-500), 0.54);
  position: fixed;
  inset: 0;
  z-index: 2;
}

:host([position=start]) [part=mini],
:host([position=end]) [part=mini] {
  background: hsla(var(--ig-surface-500), var(--ig-surface-a));
  -webkit-box-shadow: var(--ig-elevation-16);
          box-shadow: var(--ig-elevation-16);
  position: fixed;
  top: 0;
  bottom: 0;
}

:host([position=start]) [part=mini] {
  -webkit-border-end: 1px solid hsla(var(--ig-gray-900), 0.14);
          border-inline-end: 1px solid hsla(var(--ig-gray-900), 0.14);
  inset-inline-start: 0;
}

:host([position=end]) [part=mini] {
  -webkit-border-start: 1px solid hsla(var(--ig-gray-900), 0.14);
          border-inline-start: 1px solid hsla(var(--ig-gray-900), 0.14);
  inset-inline-end: 0;
}

:host([position=top]) [part=main],
:host([position=bottom]) [part=main] {
  width: 100%;
}
:host([position=top]) [part=base],
:host([position=bottom]) [part=base] {
  height: 50vh;
  min-height: initial;
  top: auto;
  width: 100%;
}

:host([position=top]) [part=base] {
  top: 0;
  bottom: auto;
}

:host(:not([open])[position=top]) [part=base],
:host(:not([open])[position=bottom]) [part=base] {
  -webkit-transform: translate3d(0, -60vh, 0);
          transform: translate3d(0, -60vh, 0);
}
:host(:not([open])[position=top]) [part=mini],
:host(:not([open])[position=bottom]) [part=mini] {
  display: none;
}

:host(:not([open])[position=bottom]) [part=base] {
  -webkit-transform: translate3d(0, 60vh, 0);
          transform: translate3d(0, 60vh, 0);
}

:host([open][position=end]) [part=base] {
  -webkit-border-end: none;
          border-inline-end: none;
  -webkit-border-start: 1px solid hsla(var(--ig-gray-900), 0.14);
          border-inline-start: 1px solid hsla(var(--ig-gray-900), 0.14);
  inset-inline-start: auto;
  inset-inline-end: 0;
}

:host([position=relative]) [part=base] {
  position: relative;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  -webkit-box-shadow: none;
          box-shadow: none;
  z-index: initial;
  -webkit-border-end: 1px solid hsla(var(--ig-gray-300), var(--ig-gray-a));
          border-inline-end: 1px solid hsla(var(--ig-gray-300), var(--ig-gray-a));
}
:host([position=relative]) [part=overlay] {
  display: none;
}

:host([open]) [part=mini] {
  display: none;
}

:host([open]:not([position=relative])) [part=base] {
  -webkit-border-end: 1px solid hsla(var(--ig-gray-900), 0.14);
          border-inline-end: 1px solid hsla(var(--ig-gray-900), 0.14);
  -webkit-box-shadow: var(--ig-elevation-16);
          box-shadow: var(--ig-elevation-16);
}

:host(:not([open])) [part=main] {
  display: none;
}
:host(:not([open])) [part=overlay] {
  display: none;
}`;
