// igniteui-webcomponents/components/calendar/themes/calendar.base.css.js
import { css } from 'lit';
export const styles = css `/* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */
:host {
  --is-large: clamp(0, (var(--component-size, 1) + 1) - var(--ig-size-large, 3), 1);
  --is-medium:
      min(
          clamp(0, (var(--component-size, 1) + 1) - var(--ig-size-medium, 2), 1),
          clamp(0, var(--ig-size-large, 3) - var(--component-size, 1), 1)
      );
  --is-small: clamp(0, var(--ig-size-medium) - var(--component-size, 1), 1);
  position: relative;
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
  scrollbar-width: var(--ig-scrollbar-size);
  scrollbar-color: var(--ig-scrollbar-thumb-background) var(--ig-scrollbar-track-background);
}
:host *,
:host *::before,
:host *::after {
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
}
:host ::-webkit-scrollbar {
  width: var(--ig-scrollbar-size);
  height: var(--ig-scrollbar-size);
  background: var(--ig-scrollbar-track-background);
}
:host ::-webkit-scrollbar-thumb {
  background: var(--ig-scrollbar-thumb-background);
}

[hidden] {
  display: none !important;
}

:host(igc-calendar) {
  --size: max(var(--is-large, 1) * max(2rem, -1 * 2rem), var(--is-medium, 1) * max(1.75rem, -1 * 1.75rem), var(--is-small, 1) * max(1.5rem, -1 * 1.5rem));
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
  font-family: var(--ig-font-family);
  background: hsla(var(--ig-surface-500), var(--ig-surface-a));
  color: var(--ig-surface-500-contrast);
}
:host(igc-calendar) > * {
  font-family: var(--ig-font-family);
}
:host(igc-calendar) igc-days-view::part(date),
:host(igc-calendar) igc-days-view::part(week-number),
:host(igc-calendar) igc-days-view::part(label),
:host(igc-calendar) igc-days-view::part(days-row) {
  height: var(--size);
}
:host(igc-calendar) igc-days-view::part(week-number-inner),
:host(igc-calendar) igc-days-view::part(date-inner) {
  width: calc(var(--size) - 0.125rem);
  min-width: calc(var(--size) - 0.125rem);
  height: 100%;
}
:host(igc-calendar) igc-days-view::part(label) {
  min-width: var(--size);
  height: var(--size);
}
:host(igc-calendar) igc-days-view::part(date-inner) {
  border-radius: clamp(0rem, calc(var(--ig-radius-factor) * calc(calc(var(--size) / 2) + 0.125rem)), calc(calc(var(--size) / 2) + 0.125rem));
}
:host(igc-calendar) igc-months-view {
  grid-template-rows: repeat(4, minmax(var(--size), 1fr));
}
:host(igc-calendar) igc-months-view::part(month-inner) {
  height: var(--size);
  border-radius: clamp(0rem, calc(var(--ig-radius-factor) * calc(calc(var(--size) / 2) + 0.125rem)), calc(calc(var(--size) / 2) + 0.125rem));
}
:host(igc-calendar) igc-years-view {
  grid-template-rows: repeat(5, minmax(var(--size), 1fr));
}
:host(igc-calendar) igc-years-view::part(year-inner) {
  height: var(--size);
  border-radius: clamp(0rem, calc(var(--ig-radius-factor) * calc(calc(var(--size) / 2) + 0.125rem)), calc(calc(var(--size) / 2) + 0.125rem));
}

:host(igc-calendar[size=large]) {
  --component-size: var(--ig-size, var(--ig-size-large));
}

:host(igc-calendar[size=medium]) {
  --component-size: var(--ig-size, var(--ig-size-medium));
}

:host(igc-calendar[size=small]) {
  --component-size: var(--ig-size, var(--ig-size-small));
}

[part=navigation] {
  position: relative;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
      -ms-flex-pack: justify;
          justify-content: space-between;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  height: 1.25rem;
  padding: 1.5625rem 1.125rem;
  -webkit-user-select: none;
     -moz-user-select: none;
      -ms-user-select: none;
          user-select: none;
}

[part=months-navigation],
[part=years-navigation],
[part~=navigation-button] {
  display: -webkit-inline-box;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-user-select: none;
     -moz-user-select: none;
      -ms-user-select: none;
          user-select: none;
  background: transparent;
  border: none;
  color: hsla(var(--ig-gray-800), var(--ig-gray-a));
  font-size: 1rem;
  padding: 0;
  -webkit-transition: color 150ms ease-in-out;
  -o-transition: color 150ms ease-in-out;
  transition: color 150ms ease-in-out;
}
[part=months-navigation]:hover, [part=months-navigation]:focus,
[part=years-navigation]:hover,
[part=years-navigation]:focus,
[part~=navigation-button]:hover,
[part~=navigation-button]:focus {
  color: hsla(var(--ig-secondary-500), var(--ig-secondary-a));
  cursor: pointer;
}
[part=months-navigation]:focus,
[part=years-navigation]:focus,
[part~=navigation-button]:focus {
  outline: none;
}

[part~=navigation-button][part~=vertical] {
  -webkit-transform: rotate(90deg);
      -ms-transform: rotate(90deg);
          transform: rotate(90deg);
}

[part=navigation-buttons] {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  gap: 1.5rem;
}

:host(igc-calendar[header-orientation=vertical]) {
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
      -ms-flex-direction: row;
          flex-direction: row;
}
:host(igc-calendar[header-orientation=vertical]) [part=header] {
  -webkit-box-pack: start;
      -ms-flex-pack: start;
          justify-content: flex-start;
  width: 8.75rem;
}
:host(igc-calendar[header-orientation=vertical]) [part=header-date] {
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
}

[part=days-view-container] {
  -webkit-box-flex: 1;
      -ms-flex-positive: 1;
          flex-grow: 1;
}

[part=header] {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
  -webkit-box-pack: justify;
      -ms-flex-pack: justify;
          justify-content: space-between;
  padding: 1rem;
  background: hsla(var(--ig-secondary-500), var(--ig-secondary-a));
  color: var(--ig-secondary-600-contrast);
}

[part=header-date],
[part=header-title] {
  margin: 0;
}

[part=header-date] {
  font-family: var(--ig-h4-font-family, var(--ig-font-family));
  font-size: var(--ig-h4-font-size);
  font-weight: var(--ig-h4-font-weight);
  font-style: var(--ig-h4-font-style);
  line-height: var(--ig-h4-line-height);
  letter-spacing: var(--ig-h4-letter-spacing);
  text-transform: var(--ig-h4-text-transform);
  margin-top: var(--ig-h4-margin-top);
  margin-bottom: var(--ig-h4-margin-bottom);
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  white-space: pre-wrap;
}

[part=header-title] {
  font-family: var(--ig-overline-font-family, var(--ig-font-family));
  font-size: var(--ig-overline-font-size);
  font-weight: var(--ig-overline-font-weight);
  font-style: var(--ig-overline-font-style);
  line-height: var(--ig-overline-line-height);
  letter-spacing: var(--ig-overline-letter-spacing);
  text-transform: var(--ig-overline-text-transform);
  margin-top: var(--ig-overline-margin-top);
  margin-bottom: var(--ig-overline-margin-bottom);
}

.aria-off-screen {
  position: absolute;
  width: 1px;
  height: 1px;
  margin: -1px;
  border: none;
  clip: rect(0, 0, 0, 0);
  outline: 0;
  pointer-events: none;
  overflow: hidden;
  -webkit-appearance: none;
     -moz-appearance: none;
          appearance: none;
}`;
