// igniteui-webcomponents/components/calendar/themes/days-view.base.css.js
import { css } from 'lit';
export const styles = css `/* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */
:host([part=days-view]) {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
  min-width: 18.125rem;
  outline: none;
  padding: 0 0.75rem 1rem 0.75rem;
  overflow: hidden;
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
  gap: 0.375rem;
}

[part~=days-row] {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
}

[part~=date-inner] {
  cursor: pointer;
  -webkit-user-select: none;
     -moz-user-select: none;
      -ms-user-select: none;
          user-select: none;
  z-index: 2;
}
[part~=date-inner]:hover, [part~=date-inner]:focus {
  background: hsla(var(--ig-secondary-100), var(--ig-secondary-a));
  color: var(--ig-secondary-100-contrast);
  border-color: hsla(var(--ig-secondary-100), var(--ig-secondary-a));
}

[part~=date-inner][part~=current][part~=selected] {
  -webkit-box-shadow: inset 0 0 0 0.0625rem hsla(var(--ig-surface-500), var(--ig-surface-a)), 0 0 0 0.0625rem hsla(var(--ig-secondary-500), var(--ig-secondary-a));
          box-shadow: inset 0 0 0 0.0625rem hsla(var(--ig-surface-500), var(--ig-surface-a)), 0 0 0 0.0625rem hsla(var(--ig-secondary-500), var(--ig-secondary-a));
}

[part~=current][part~=date-inner] {
  -webkit-box-shadow: inset 0 0 0 0.0625rem hsla(var(--ig-gray-600), var(--ig-gray-a));
          box-shadow: inset 0 0 0 0.0625rem hsla(var(--ig-gray-600), var(--ig-gray-a));
}
[part~=current][part~=date-inner]:hover, [part~=current][part~=date-inner]:focus {
  -webkit-box-shadow: inset 0 0 0 0.0625rem hsla(var(--ig-surface-500), var(--ig-surface-a));
          box-shadow: inset 0 0 0 0.0625rem hsla(var(--ig-surface-500), var(--ig-surface-a));
}

[part~=date],
[part~=week-number],
[part~=label] {
  font-size: var(--ig-subtitle-1-font-size);
  width: 100%;
  color: hsla(var(--ig-gray-800), var(--ig-gray-a));
  position: relative;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  outline: none;
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
  -webkit-user-select: none;
     -moz-user-select: none;
      -ms-user-select: none;
          user-select: none;
  border-top: 0.0625rem solid transparent;
  border-bottom: 0.0625rem solid transparent;
}

[part=label-inner] {
  -o-text-overflow: ellipsis;
     text-overflow: ellipsis;
  overflow: hidden;
  min-width: 4ch;
  text-align: center;
}

[part~=date-inner],
[part~=week-number-inner] {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
  border: 0.0625rem solid transparent;
}
[part~=date-inner]:focus,
[part~=week-number-inner]:focus {
  outline: none;
}

[part~=date][part~=range] {
  background: hsla(var(--ig-secondary-50), var(--ig-secondary-a));
}

[part~=date][part~=range][part~=preview] {
  background: transparent;
  border-top: 0.0625rem dashed hsla(var(--ig-gray-300), var(--ig-gray-a));
  border-bottom: 0.0625rem dashed hsla(var(--ig-gray-300), var(--ig-gray-a));
}

[part~=date][part~=range][part~=preview][part~=first],
[part~=date][part~=range][part~=preview][part~=last] {
  background: transparent;
  border-top-color: transparent;
  border-bottom-color: transparent;
}
[part~=date][part~=range][part~=preview][part~=first]::after,
[part~=date][part~=range][part~=preview][part~=last]::after {
  border-top: 0.0625rem dashed hsla(var(--ig-gray-300), var(--ig-gray-a));
  border-bottom: 0.0625rem dashed hsla(var(--ig-gray-300), var(--ig-gray-a));
  background: transparent;
}

[part~=date][part~=range][part~=first],
[part~=date][part~=range][part~=last] {
  background: transparent;
}
[part~=date][part~=range][part~=first]::after,
[part~=date][part~=range][part~=last]::after {
  content: "";
  position: absolute;
  height: 100%;
  width: calc(50% + 0.0625rem);
  background: hsla(var(--ig-secondary-50), var(--ig-secondary-a));
  border-top: 0.0625rem solid hsla(var(--ig-secondary-50), var(--ig-secondary-a));
  border-bottom: 0.0625rem solid hsla(var(--ig-secondary-50), var(--ig-secondary-a));
  z-index: 1;
}

[part~=date][part~=range][part~=first]::after {
  inset-inline-start: 50%;
}

[part~=date][part~=range][part~=last]::after {
  inset-inline-end: calc(50% - 0.0625rem);
}

[part~=date][part~=range] + [part~=date][part~=range][part~=last]::after {
  inset-inline-end: 50%;
}

[part~=date-inner][part~=inactive],
[part~=date-inner][part~=label],
[part~=date-inner][part~=disabled] {
  color: hsla(var(--ig-gray-500), var(--ig-gray-a));
}

[part~=date-inner][part~=weekend]:hover, [part~=date-inner][part~=weekend]:focus,
[part~=date-inner][part~=inactive]:hover,
[part~=date-inner][part~=inactive]:focus {
  color: var(--ig-secondary-100-contrast);
}

[part~=date-inner][part~=selected][part~=range][part~=first],
[part~=date-inner][part~=selected][part~=range][part~=last],
[part~=date-inner][part~=selected][part~=range][part~=weekend][part~=first],
[part~=date-inner][part~=selected][part~=range][part~=weekend][part~=last],
[part~=date-inner][part~=selected][part~=range][part~=inactive][part~=first],
[part~=date-inner][part~=selected][part~=range][part~=inactive][part~=last] {
  color: var(--ig-secondary-700-contrast);
}

[part~=date-inner][part~=selected][part~=range][part~=weekend],
[part~=date-inner][part~=selected][part~=range][part~=inactive],
[part~=date-inner][part~=selected][part~=range] {
  color: var(--ig-secondary-50-contrast);
}
[part~=date-inner][part~=selected][part~=range][part~=weekend]:hover, [part~=date-inner][part~=selected][part~=range][part~=weekend]:focus,
[part~=date-inner][part~=selected][part~=range][part~=inactive]:hover,
[part~=date-inner][part~=selected][part~=range][part~=inactive]:focus,
[part~=date-inner][part~=selected][part~=range]:hover,
[part~=date-inner][part~=selected][part~=range]:focus {
  color: var(--ig-secondary-100-contrast);
  background: hsla(var(--ig-secondary-100), var(--ig-secondary-a));
  border-color: hsla(var(--ig-secondary-100), var(--ig-secondary-a));
}

[part~=date-inner][part~=selected][part~=range][part~=first],
[part~=date-inner][part~=selected][part~=range][part~=last],
[part~=date-inner][part~=selected]:not([part~=range]) {
  background: hsla(var(--ig-secondary-500), var(--ig-secondary-a));
  border-color: hsla(var(--ig-secondary-500), var(--ig-secondary-a));
}
[part~=date-inner][part~=selected][part~=range][part~=first]:hover, [part~=date-inner][part~=selected][part~=range][part~=first]:focus,
[part~=date-inner][part~=selected][part~=range][part~=last]:hover,
[part~=date-inner][part~=selected][part~=range][part~=last]:focus,
[part~=date-inner][part~=selected]:not([part~=range]):hover,
[part~=date-inner][part~=selected]:not([part~=range]):focus {
  background: hsla(var(--ig-secondary-700), var(--ig-secondary-a));
  border-color: hsla(var(--ig-secondary-700), var(--ig-secondary-a));
  color: var(--ig-secondary-700-contrast);
}

[part~=date-inner][part~=selected][part~=weekend]:not([part~=range]),
[part~=date-inner][part~=selected][part~=inactive]:not([part~=range]),
[part~=date-inner][part~=selected]:not([part~=range]) {
  color: var(--ig-secondary-700-contrast);
}

[part~=week-number] {
  -webkit-box-ordinal-group: 1;
      -ms-flex-order: 0;
          order: 0;
  color: hsla(var(--ig-gray-500), var(--ig-gray-a));
  cursor: default;
}

[part="week-number-inner first"] {
  border-top-left-radius: clamp(0rem, calc(var(--ig-radius-factor) * 0.25rem), 0.25rem);
  border-top-right-radius: clamp(0rem, calc(var(--ig-radius-factor) * 0.25rem), 0.25rem);
}
[part="week-number-inner first"]::after {
  display: none;
}

[part="week-number-inner last"] {
  border-bottom-left-radius: clamp(0rem, calc(var(--ig-radius-factor) * 0.25rem), 0.25rem);
  border-bottom-right-radius: clamp(0rem, calc(var(--ig-radius-factor) * 0.25rem), 0.25rem);
}

[part~=week-number-inner] {
  position: relative;
  background: hsla(var(--ig-gray-200), var(--ig-gray-a));
  border-inline-end-color: hsla(var(--ig-gray-200), var(--ig-gray-a));
}
[part~=week-number-inner]::after {
  content: "";
  position: absolute;
  bottom: 100%;
  height: calc(0.375rem + (0.0625rem * 2 + 2px));
  width: 100%;
  background: hsla(var(--ig-gray-200), var(--ig-gray-a));
  -webkit-border-start: 0.0625rem solid hsla(var(--ig-gray-200), var(--ig-gray-a));
          border-inline-start: 0.0625rem solid hsla(var(--ig-gray-200), var(--ig-gray-a));
  -webkit-border-end: 0.0625rem solid hsla(var(--ig-gray-200), var(--ig-gray-a));
          border-inline-end: 0.0625rem solid hsla(var(--ig-gray-200), var(--ig-gray-a));
}

[part~=date-inner][part~=disabled] {
  pointer-events: none;
  -webkit-user-select: none;
     -moz-user-select: none;
      -ms-user-select: none;
          user-select: none;
  cursor: not-allowed;
}

[part~=date-inner][part~=special][part~=range],
[part~=date-inner][part~=special] {
  background: hsla(var(--ig-gray-100), var(--ig-gray-a));
  border-color: hsla(var(--ig-gray-100), var(--ig-gray-a));
  color: hsla(var(--ig-gray-900), var(--ig-gray-a));
  font-weight: 900;
}
[part~=date-inner][part~=special][part~=range]:hover, [part~=date-inner][part~=special][part~=range]:focus,
[part~=date-inner][part~=special]:hover,
[part~=date-inner][part~=special]:focus {
  background: hsla(var(--ig-secondary-100), var(--ig-secondary-a));
  border-color: hsla(var(--ig-secondary-100), var(--ig-secondary-a));
}

[part~=hidden] {
  visibility: hidden;
}`;
