// igniteui-webcomponents/components/calendar/themes/year-month-view.base.css.js
import { css } from 'lit';
export const styles = css `/* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */
:host {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
  min-width: 18.125rem;
  outline: none;
  padding: 0 0.75rem 1rem 0.75rem;
  overflow: hidden;
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
  gap: 0.25rem;
}

[part=months-row],
[part=years-row] {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
}

[part~=month],
[part~=year] {
  -webkit-box-flex: 1;
      -ms-flex: 1;
          flex: 1;
}

[part~=month],
[part~=year],
[part~=month-inner],
[part~=year-inner] {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
}

[part~=month-inner],
[part~=year-inner] {
  -webkit-user-select: none;
     -moz-user-select: none;
      -ms-user-select: none;
          user-select: none;
  cursor: pointer;
  padding: 0 0.75rem;
}
[part~=month-inner]:hover, [part~=month-inner]:focus,
[part~=year-inner]:hover,
[part~=year-inner]:focus {
  background: hsla(var(--ig-gray-200), var(--ig-gray-a));
}
[part~=month-inner]:hover,
[part~=year-inner]:hover {
  background: hsla(var(--ig-gray-200), var(--ig-gray-a));
}

[part~=month-inner][part~=selected],
[part~=year-inner][part~=selected] {
  background: hsla(var(--ig-secondary-500), var(--ig-secondary-a));
  color: var(--ig-secondary-600-contrast);
}
[part~=month-inner][part~=selected]:hover,
[part~=year-inner][part~=selected]:hover {
  background: hsla(var(--ig-secondary-500), var(--ig-secondary-a));
  color: var(--ig-secondary-600-contrast);
}
[part~=month-inner][part~=selected]:focus,
[part~=year-inner][part~=selected]:focus {
  outline: none;
  background: hsla(var(--ig-secondary-700), var(--ig-secondary-a));
  color: var(--ig-secondary-700-contrast);
}

[part~=month-inner][part~=current],
[part~=year-inner][part~=current] {
  -webkit-box-shadow: inset 0 0 0 0.0625rem hsla(var(--ig-gray-600), var(--ig-gray-a));
          box-shadow: inset 0 0 0 0.0625rem hsla(var(--ig-gray-600), var(--ig-gray-a));
}

[part~=month-inner][part~=selected][part~=current],
[part~=year-inner][part~=selected][part~=current] {
  -webkit-box-shadow: inset 0 0 0 0.0625rem hsla(var(--ig-surface-500), var(--ig-surface-a)), 0 0 0 0.0625rem hsla(var(--ig-secondary-500), var(--ig-secondary-a));
          box-shadow: inset 0 0 0 0.0625rem hsla(var(--ig-surface-500), var(--ig-surface-a)), 0 0 0 0.0625rem hsla(var(--ig-secondary-500), var(--ig-secondary-a));
}`;
