// igniteui-webcomponents/components/chip/chip.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
import { html, LitElement, nothing } from 'lit';
import { property } from 'lit/decorators.js';
import { themes } from '../../theming/theming-decorator.js';
import { blazorTwoWayBind } from '../common/decorators/blazorTwoWayBind.js';
import { EventEmitterMixin } from '../common/mixins/event-emitter.js';
import { SizableMixin } from '../common/mixins/sizable.js';
import { styles } from './themes/chip.base.css.js';
import { styles as bootstrap } from './themes/chip.bootstrap.css.js';
import { styles as fluent } from './themes/chip.fluent.css.js';
import { styles as indigo } from './themes/chip.indigo.css.js';
import { styles as material } from './themes/chip.material.css.js';
import { defineComponents } from '../common/definitions/defineComponents.js';
import IgcIconComponent from '../icon/icon.js';
defineComponents(IgcIconComponent);
let IgcChipComponent = class IgcChipComponent extends SizableMixin(EventEmitterMixin(LitElement)) {
    constructor() {
        super();
        this.disabled = false;
        this.removable = false;
        this.selectable = false;
        this.selected = false;
        this.size = 'medium';
    }
    handleSelect() {
        if (this.selectable) {
            this.selected = !this.selected;
            this.emitEvent('igcSelect', { detail: this.selected });
        }
    }
    handleRemove(e) {
        this.emitEvent('igcRemove');
        e.stopPropagation();
    }
    handleKeyup(e) {
        if (/\s|enter/i.test(e.key)) {
            this.handleRemove(e);
        }
    }
    render() {
        return html `
      <button
        part="base"
        .disabled="${this.disabled}"
        aria-selected="${this.selected ? 'true' : 'false'}"
        aria-disabled="${this.disabled ? 'true' : 'false'}"
        @click=${this.handleSelect}
      >
        <span part="prefix">
          ${this.selectable && this.selected
            ? html `<slot @slotchange=${this.slotChanges} name="select">
                <igc-icon
                  size=${this.size}
                  name="chip_select"
                  collection="internal"
                ></igc-icon>
              </slot>`
            : nothing}
          <slot name="start"></slot>
          <slot name="prefix"></slot>
        </span>
        <slot></slot>
        <span part="suffix">
          <slot name="end"></slot>
          <slot name="suffix"></slot>
          ${this.removable && !this.disabled
            ? html `<slot
                @slotchange=${this.slotChanges}
                @click=${this.handleRemove}
                @keyup=${this.handleKeyup}
                name="remove"
              >
                <igc-icon
                  size=${this.size}
                  name="chip_cancel"
                  collection="internal"
                  tabindex="0"
                  role="button"
                  aria-label="remove"
                ></igc-icon>
              </slot>`
            : nothing}
        </span>
      </button>
    `;
    }
    slotChanges() {
        this.requestUpdate();
    }
};
IgcChipComponent.tagName = 'igc-chip';
IgcChipComponent.styles = styles;
__decorate([
    property({ type: Boolean, reflect: true })
], IgcChipComponent.prototype, "disabled", void 0);
__decorate([
    property({ type: Boolean, reflect: true })
], IgcChipComponent.prototype, "removable", void 0);
__decorate([
    property({ type: Boolean, reflect: true })
], IgcChipComponent.prototype, "selectable", void 0);
__decorate([
    property({ type: Boolean, reflect: true }),
    blazorTwoWayBind('igcSelect', 'detail')
], IgcChipComponent.prototype, "selected", void 0);
__decorate([
    property({ reflect: true })
], IgcChipComponent.prototype, "variant", void 0);
IgcChipComponent = __decorate([
    themes({ material, bootstrap, fluent, indigo })
], IgcChipComponent);
export default IgcChipComponent;
