// igniteui-webcomponents/components/checkbox/themes/light/checkbox.base.css.js
import { css } from 'lit';
export const styles = css `:host {
  --is-large: clamp(0, (var(--component-size, 1) + 1) - var(--ig-size-large, 3), 1);
  --is-medium:
      min(
          clamp(0, (var(--component-size, 1) + 1) - var(--ig-size-medium, 2), 1),
          clamp(0, var(--ig-size-large, 3) - var(--component-size, 1), 1)
      );
  --is-small: clamp(0, var(--ig-size-medium) - var(--component-size, 1), 1);
  position: relative;
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
  scrollbar-width: var(--ig-scrollbar-size);
  scrollbar-color: var(--ig-scrollbar-thumb-background) var(--ig-scrollbar-track-background);
}
:host *,
:host *::before,
:host *::after {
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
}
:host ::-webkit-scrollbar {
  width: var(--ig-scrollbar-size);
  height: var(--ig-scrollbar-size);
  background: var(--ig-scrollbar-track-background);
}
:host ::-webkit-scrollbar-thumb {
  background: var(--ig-scrollbar-thumb-background);
}

[hidden] {
  display: none !important;
}

/* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */
:host {
  display: -webkit-inline-box;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: var(--ig-font-family);
}

input[type=checkbox] {
  position: absolute;
  width: 1px;
  height: 1px;
  margin: -1px;
  border: none;
  clip: rect(0, 0, 0, 0);
  outline: 0;
  pointer-events: none;
  overflow: hidden;
  -webkit-appearance: none;
     -moz-appearance: none;
          appearance: none;
}

[part~=control] {
  --size: 1.25rem;
  position: relative;
  display: -webkit-inline-box;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-user-select: none;
     -moz-user-select: none;
      -ms-user-select: none;
          user-select: none;
}
[part~=control]::after {
  border-radius: clamp(0rem, calc(var(--ig-radius-factor) * 0.125rem), 0.125rem);
  position: absolute;
  content: "";
  left: 50%;
  top: 50%;
  -webkit-transform: translate(-50%, -50%);
      -ms-transform: translate(-50%, -50%);
          transform: translate(-50%, -50%);
  width: var(--size);
  height: var(--size);
  border: 0.125rem solid hsla(var(--ig-gray-600), var(--ig-gray-a));
  -webkit-transition: all 0.15s ease-in;
  -o-transition: all 0.15s ease-in;
  transition: all 0.15s ease-in;
}

[part~=label] {
  display: inline-block;
  color: hsla(var(--ig-gray-900), var(--ig-gray-a));
  cursor: pointer;
  -webkit-user-select: none;
     -moz-user-select: none;
      -ms-user-select: none;
          user-select: none;
  word-wrap: break-all;
  -webkit-transition: color 0.2s ease-in-out;
  -o-transition: color 0.2s ease-in-out;
  transition: color 0.2s ease-in-out;
}

[part~=indicator] {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  stroke-linecap: square;
  stroke-width: 3;
  stroke-dasharray: 24;
  stroke-dashoffset: 24;
  fill: none;
  opacity: 0;
  z-index: 1;
  -webkit-transform: scale(0.45);
      -ms-transform: scale(0.45);
          transform: scale(0.45);
  -webkit-transform-origin: center;
      -ms-transform-origin: center;
          transform-origin: center;
}
[part~=indicator] svg {
  width: 100%;
  height: 100%;
}

label {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  cursor: pointer;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
      -ms-flex-flow: row nowrap;
          flex-flow: row nowrap;
  gap: 0.5rem;
}
label:hover [part="control checked"] {
  background: hsla(var(--ig-secondary-500), 0.06);
}
label:focus-within [part="control checked"] {
  background: hsla(var(--ig-secondary-500), 0.12);
}

[part="control checked"]::after {
  border-color: hsla(var(--ig-secondary-500), var(--ig-secondary-a));
  background: hsla(var(--ig-secondary-500), var(--ig-secondary-a));
}

[part="indicator checked"] {
  -webkit-transition: all 0.2s ease-out;
  -o-transition: all 0.2s ease-out;
  transition: all 0.2s ease-out;
  stroke-dashoffset: 0;
  opacity: 1;
}

:host([indeterminate]) [part~=base]:hover [part~=control] {
  background: hsla(var(--ig-secondary-500), 0.06);
}
:host([indeterminate]) [part~=base]:focus-within [part~=control] {
  background: hsla(var(--ig-secondary-500), 0.12);
}
:host([indeterminate]) [part~=control]::after {
  border-color: hsla(var(--ig-secondary-500), var(--ig-secondary-a));
  background: hsla(var(--ig-secondary-500), var(--ig-secondary-a));
}
:host([indeterminate]) [part~=indicator] {
  stroke-dashoffset: 41;
  opacity: 1;
  -webkit-transform: rotate(45deg) scale(0.45) translateX(-0.25rem);
      -ms-transform: rotate(45deg) scale(0.45) translateX(-0.25rem);
          transform: rotate(45deg) scale(0.45) translateX(-0.25rem);
}

:host(:not([disabled])[invalid]) [part~=label] {
  color: hsla(var(--ig-error-500), var(--ig-error-a));
}
:host(:not([disabled])[invalid]) [part~=control]::after {
  border-color: hsla(var(--ig-error-500), var(--ig-error-a));
}
:host(:not([disabled])[invalid]) [part="control checked"]::after {
  background: hsla(var(--ig-error-500), var(--ig-error-a));
}

:host(:not([disabled])[invalid][indeterminate]) [part~=control]::after {
  background: hsla(var(--ig-error-500), var(--ig-error-a));
}

:host([disabled]) {
  -webkit-user-select: none;
     -moz-user-select: none;
      -ms-user-select: none;
          user-select: none;
  cursor: initial;
  pointer-events: none;
}

:host([disabled][indeterminate]) [part~=control]::after {
  background: hsla(var(--ig-gray-300), var(--ig-gray-a));
}

:host([label-position=before]) [part~=label] {
  -webkit-box-ordinal-group: 0;
      -ms-flex-order: -1;
          order: -1;
}`;
