// igniteui-webcomponents/components/checkbox/themes/light/checkbox.indigo.css.js
import { css } from 'lit';
export const styles = css `/* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */
:host {
  font-weight: 600;
}

[part~=control] {
  border-radius: clamp(0rem, calc(var(--ig-radius-factor) * 0.1875rem), 0.1875rem);
  width: var(--size, 1.25rem);
  height: var(--size, 1.25rem);
  min-width: var(--size, 1.25rem);
}
[part~=control]::after {
  border-radius: inherit;
  border-color: var(--border-color, hsla(var(--ig-gray-500), var(--ig-gray-a)));
}

[part~=label] {
  font-family: var(--ig-body-1-font-family, var(--ig-font-family));
  font-size: var(--ig-body-1-font-size);
  font-weight: var(--ig-body-1-font-weight);
  font-style: var(--ig-body-1-font-style);
  line-height: var(--ig-body-1-line-height);
  letter-spacing: var(--ig-body-1-letter-spacing);
  text-transform: var(--ig-body-1-text-transform);
  margin-top: var(--ig-body-1-margin-top);
  margin-bottom: var(--ig-body-1-margin-bottom);
}

[part~=indicator] {
  stroke: hsla(var(--ig-surface-500), var(--ig-surface-a));
  -webkit-transform: scale(0.8);
      -ms-transform: scale(0.8);
          transform: scale(0.8);
}

:host([indeterminate]) [part~=indicator] {
  -webkit-transform: rotate(45deg) translateX(-0.125rem) scale(0.8);
      -ms-transform: rotate(45deg) translateX(-0.125rem) scale(0.8);
          transform: rotate(45deg) translateX(-0.125rem) scale(0.8);
}

:host(:not([invalid])) [part="control checked"]::after,
:host(:not([invalid])[indeterminate]) [part~=control]::after {
  background: var(--checked-color, hsla(var(--ig-secondary-500), var(--ig-secondary-a)));
  border-color: var(--checked-color, hsla(var(--ig-secondary-500), var(--ig-secondary-a)));
}

:host(:hover) [part~=control] {
  -webkit-box-shadow: 0 0 0 4px hsla(var(--ig-gray-200), 0.87);
          box-shadow: 0 0 0 4px hsla(var(--ig-gray-200), 0.87);
}

:host(:focus-within) [part~=control] {
  -webkit-box-shadow: 0 0 0 4px hsla(var(--ig-gray-300), 0.87);
          box-shadow: 0 0 0 4px hsla(var(--ig-gray-300), 0.87);
}

:host(:hover) [part="control checked"],
:host([indeterminate]:hover) [part~=control] {
  -webkit-box-shadow: 0 0 0 4px hsla(var(--ig-secondary-100), 0.87);
          box-shadow: 0 0 0 4px hsla(var(--ig-secondary-100), 0.87);
}

:host(:focus-within) [part="control checked"],
:host([indeterminate]:focus-within) [part~=control] {
  -webkit-box-shadow: 0 0 0 4px hsla(var(--ig-secondary-200), 0.87);
          box-shadow: 0 0 0 4px hsla(var(--ig-secondary-200), 0.87);
}

:host(:not([disabled])[invalid]) [part="control checked"],
:host(:not([disabled])[invalid][indeterminate]) [part~=control] {
  -webkit-box-shadow: inset 0 0 0 4px hsla(var(--ig-error-500), var(--ig-error-a));
          box-shadow: inset 0 0 0 4px hsla(var(--ig-error-500), var(--ig-error-a));
  background: hsla(var(--ig-error-500), var(--ig-error-a));
}

:host([invalid]:hover) [part~=control],
:host([invalid]:hover) [part="control checked"],
:host([invalid][indeterminate]:hover) [part~=control] {
  -webkit-box-shadow: 0 0 0 4px hsla(var(--ig-error-100), 0.87);
          box-shadow: 0 0 0 4px hsla(var(--ig-error-100), 0.87);
}

:host([invalid]:focus-within) [part~=control],
:host([invalid]:focus-within) [part="control checked"],
:host([invalid][indeterminate]:focus-within) [part~=control] {
  -webkit-box-shadow: 0 0 0 4px hsla(var(--ig-error-200), 0.87);
          box-shadow: 0 0 0 4px hsla(var(--ig-error-200), 0.87);
}

:host([disabled]) [part~=control]::after,
:host([disabled][invalid]) [part~=control]::after {
  border-color: var(--disabled-color, hsla(var(--ig-gray-300), var(--ig-gray-a)));
}
:host([disabled]) [part="control checked"]::after,
:host([disabled][invalid]) [part="control checked"]::after {
  background: var(--disabled-color, hsla(var(--ig-gray-300), var(--ig-gray-a)));
}
:host([disabled]) [part~=label],
:host([disabled][invalid]) [part~=label] {
  color: var(--disabled-color, hsla(var(--ig-gray-300), var(--ig-gray-a)));
}

:host([disabled][indeterminate]) [part~=control]::after {
  background: var(--disabled-color, hsla(var(--ig-gray-300), var(--ig-gray-a)));
  border-color: var(--disabled-color, hsla(var(--ig-gray-300), var(--ig-gray-a)));
}`;
