// igniteui-webcomponents/components/checkbox/themes/light/checkbox.material.css.js
import { css } from 'lit';
export const styles = css `/* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */
label:hover [part=control] {
  background: hsla(var(--ig-gray-600), 0.06);
}
label:focus-within [part=control] {
  background: hsla(var(--ig-gray-600), 0.12);
}

[part~=label] {
  font-family: var(--ig-subtitle-1-font-family, var(--ig-font-family));
  font-size: var(--ig-subtitle-1-font-size);
  font-weight: var(--ig-subtitle-1-font-weight);
  font-style: var(--ig-subtitle-1-font-style);
  line-height: var(--ig-subtitle-1-line-height);
  letter-spacing: var(--ig-subtitle-1-letter-spacing);
  text-transform: var(--ig-subtitle-1-text-transform);
  margin-top: var(--ig-subtitle-1-margin-top);
  margin-bottom: var(--ig-subtitle-1-margin-bottom);
}

[part~=indicator] {
  stroke: hsla(var(--ig-surface-500), var(--ig-surface-a));
}

[part~=control] {
  --size: 1.25rem;
  border-radius: clamp(0rem, calc(var(--ig-radius-factor) * calc(var(--size))), calc(var(--size)));
  width: calc(var(--size) * 2);
  height: calc(var(--size) * 2);
  min-width: calc(var(--size) * 2);
  -webkit-transition: background 0.15s ease-in;
  -o-transition: background 0.15s ease-in;
  transition: background 0.15s ease-in;
}

:host([invalid]) label:hover [part~=control],
:host([invalid][indeterminate]) label:hover [part~=control] {
  background: hsla(var(--ig-error-500), 0.06);
}
:host([invalid]) label:focus-within [part~=control],
:host([invalid][indeterminate]) label:focus-within [part~=control] {
  background: hsla(var(--ig-error-500), 0.12);
}

:host([disabled]) [part~=label] {
  color: hsla(var(--ig-gray-400), var(--ig-gray-a));
}
:host([disabled]) [part~=control]::after {
  border-color: hsla(var(--ig-gray-400), var(--ig-gray-a));
}
:host([disabled]) [part="control checked"]::after {
  background: hsla(var(--ig-gray-400), var(--ig-gray-a));
}

:host([disabled][indeterminate]) [part~=control]::after {
  background: hsla(var(--ig-secondary-100), var(--ig-secondary-a));
  border-color: transparent;
}`;
