// igniteui-webcomponents/components/checkbox/themes/light/checkbox.bootstrap.css.js
import { css } from 'lit';
export const styles = css `/* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */
[part~=control] {
  border-radius: clamp(0rem, calc(var(--ig-radius-factor) * 0.25rem), 0.25rem);
  --size: 1rem;
  width: var(--size);
  height: var(--size);
  min-width: var(--size);
  -webkit-transition: background-color 0.15s ease-in-out;
  -o-transition: background-color 0.15s ease-in-out;
  transition: background-color 0.15s ease-in-out;
}
[part~=control]::after {
  width: inherit;
  height: inherit;
  min-width: inherit;
  border-radius: inherit;
  border: none;
  -webkit-box-shadow: inset 0 0 0 1px hsla(var(--ig-gray-400), var(--ig-gray-a));
          box-shadow: inset 0 0 0 1px hsla(var(--ig-gray-400), var(--ig-gray-a));
  -webkit-transition: none;
  -o-transition: none;
  transition: none;
}

[part~=indicator] {
  stroke-width: 3;
  -webkit-transform: scale(0.7);
      -ms-transform: scale(0.7);
          transform: scale(0.7);
  -webkit-transition: none;
  -o-transition: none;
  transition: none;
  stroke: white;
}

[part~=label] {
  font-family: var(--ig-body-2-font-family, var(--ig-font-family));
  font-size: var(--ig-body-2-font-size);
  font-weight: var(--ig-body-2-font-weight);
  font-style: var(--ig-body-2-font-style);
  line-height: var(--ig-body-2-line-height);
  letter-spacing: var(--ig-body-2-letter-spacing);
  text-transform: var(--ig-body-2-text-transform);
  margin-top: var(--ig-body-2-margin-top);
  margin-bottom: var(--ig-body-2-margin-bottom);
  -webkit-transition: none;
  -o-transition: none;
  transition: none;
}

:host([indeterminate]) [part=indicator] {
  -webkit-transform: rotate(45deg) translateX(-0.075rem) scale(0.7);
      -ms-transform: rotate(45deg) translateX(-0.075rem) scale(0.7);
          transform: rotate(45deg) translateX(-0.075rem) scale(0.7);
}

:host(:focus-within) [part~=control] {
  -webkit-box-shadow: 0 0 0 4px hsla(var(--ig-primary-100), var(--ig-primary-a));
          box-shadow: 0 0 0 4px hsla(var(--ig-primary-100), var(--ig-primary-a));
}
:host(:focus-within) [part~=control]::after {
  -webkit-box-shadow: inset 0 0 0 1px hsla(var(--ig-primary-500), var(--ig-primary-a));
          box-shadow: inset 0 0 0 1px hsla(var(--ig-primary-500), var(--ig-primary-a));
}

[part="control checked"]::after,
:host([indeterminate]) [part~=control]::after {
  background: hsla(var(--ig-primary-500), var(--ig-primary-a));
  -webkit-box-shadow: inset 0 0 0 1px hsla(var(--ig-primary-500), var(--ig-primary-a));
          box-shadow: inset 0 0 0 1px hsla(var(--ig-primary-500), var(--ig-primary-a));
}

:host([invalid]) [part=control]::after {
  -webkit-box-shadow: inset 0 0 0 1px hsla(var(--ig-error-500), var(--ig-error-a));
          box-shadow: inset 0 0 0 1px hsla(var(--ig-error-500), var(--ig-error-a));
}

:host([invalid]) [part="control checked"]::after,
:host([invalid][indeterminate]) [part=control]::after {
  -webkit-box-shadow: inset 0 0 0 1px hsla(var(--ig-error-500), var(--ig-error-a));
          box-shadow: inset 0 0 0 1px hsla(var(--ig-error-500), var(--ig-error-a));
}

:host([invalid]:focus-within) [part=control],
:host([invalid]:focus-within) [part="control checked"],
:host([invalid][indeterminate]:focus-within) [part=control] {
  -webkit-box-shadow: 0 0 0 4px hsla(var(--ig-error-100), var(--ig-error-a));
          box-shadow: 0 0 0 4px hsla(var(--ig-error-100), var(--ig-error-a));
}

:host([disabled]) [part=control]::after,
:host([disabled][invalid]) [part=control]::after {
  -webkit-box-shadow: inset 0 0 0 1px hsla(var(--ig-gray-300), var(--ig-gray-a));
          box-shadow: inset 0 0 0 1px hsla(var(--ig-gray-300), var(--ig-gray-a));
}

:host([disabled]) [part="control checked"]::after,
:host([disabled][invalid]) [part="control checked"]::after {
  background: hsla(var(--ig-primary-200), var(--ig-primary-a));
  -webkit-box-shadow: inset 0 0 0 1px hsla(var(--ig-primary-200), var(--ig-primary-a));
          box-shadow: inset 0 0 0 1px hsla(var(--ig-primary-200), var(--ig-primary-a));
}

:host([disabled][indeterminate]) [part=control]::after,
:host([disabled][invalid][indeterminate]) [part=control]::after {
  background: hsla(var(--ig-primary-200), var(--ig-primary-a));
  -webkit-box-shadow: inset 0 0 0 1px hsla(var(--ig-primary-200), var(--ig-primary-a));
          box-shadow: inset 0 0 0 1px hsla(var(--ig-primary-200), var(--ig-primary-a));
}`;
