// igniteui-webcomponents/components/checkbox/themes/light/checkbox.fluent.css.js
import { css } from 'lit';
export const styles = css `/* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */
[part~=control] {
  width: var(--size, 1.25rem);
  height: var(--size, 1.25rem);
  min-width: var(--size, 1.25rem);
}
[part~=control]::after {
  border: none;
  background: transparent;
  -webkit-box-shadow: inset 0 0 0 1px var(--border-color, hsla(var(--ig-gray-900), var(--ig-gray-a)));
          box-shadow: inset 0 0 0 1px var(--border-color, hsla(var(--ig-gray-900), var(--ig-gray-a)));
}

[part~=label] {
  font-family: var(--ig-body-2-font-family, var(--ig-font-family));
  font-size: var(--ig-body-2-font-size);
  font-weight: var(--ig-body-2-font-weight);
  font-style: var(--ig-body-2-font-style);
  line-height: var(--ig-body-2-line-height);
  letter-spacing: var(--ig-body-2-letter-spacing);
  text-transform: var(--ig-body-2-text-transform);
  margin-top: var(--ig-body-2-margin-top);
  margin-bottom: var(--ig-body-2-margin-bottom);
}

[part~=indicator] {
  stroke: hsla(var(--ig-surface-500), var(--ig-surface-a));
  stroke-width: 1.5;
  -webkit-transform: scale(0.8);
      -ms-transform: scale(0.8);
          transform: scale(0.8);
}

:host(:hover) [part~=indicator] {
  opacity: 1;
  stroke: hsla(var(--ig-gray-700), var(--ig-gray-a));
  stroke-dashoffset: 0;
  -webkit-transition: opacity 0.1s ease-out;
  -o-transition: opacity 0.1s ease-out;
  transition: opacity 0.1s ease-out;
}
:host(:hover) [part="indicator checked"] {
  stroke: hsla(var(--ig-surface-500), var(--ig-surface-a));
}

:host(:hover) [part~=control]::after,
:host(:focus-within) [part~=control]::after {
  -webkit-box-shadow: inset 0 0 0 1px var(--hover-color, hsla(var(--ig-gray-800), var(--ig-gray-a)));
          box-shadow: inset 0 0 0 1px var(--hover-color, hsla(var(--ig-gray-800), var(--ig-gray-a)));
}

:host([indeterminate]) [part~=indicator] {
  display: none;
}
:host([indeterminate]) [part~=control]::before {
  border-radius: clamp(0rem, calc(var(--ig-radius-factor) * 0.125rem), 0.125rem);
  display: block;
  width: inherit;
  height: inherit;
  content: "";
  -webkit-transform: scale(0.5);
      -ms-transform: scale(0.5);
          transform: scale(0.5);
  -webkit-transition-duration: 0.1s;
       -o-transition-duration: 0.1s;
          transition-duration: 0.1s;
  background: hsla(var(--ig-primary-500), var(--ig-primary-a));
  z-index: 1;
}
:host([indeterminate]) [part~=control]::after {
  -webkit-box-shadow: inset 0 0 0 1px hsla(var(--ig-primary-500), var(--ig-primary-a));
          box-shadow: inset 0 0 0 1px hsla(var(--ig-primary-500), var(--ig-primary-a));
  background: transparent;
}

:host([indeterminate]:hover) [part~=control]::before,
:host([indeterminate]:focus-within) [part~=control]::before {
  background: var(--checked-hover-color, hsla(var(--ig-primary-900), var(--ig-primary-a)));
}
:host([indeterminate]:hover) [part~=control]::after,
:host([indeterminate]:focus-within) [part~=control]::after {
  -webkit-box-shadow: inset 0 0 0 1px var(--checked-hover-color, hsla(var(--ig-primary-900), var(--ig-primary-a)));
          box-shadow: inset 0 0 0 1px var(--checked-hover-color, hsla(var(--ig-primary-900), var(--ig-primary-a)));
  background: transparent;
}

[part="control checked"]::after {
  background: hsla(var(--ig-primary-500), var(--ig-primary-a));
  -webkit-box-shadow: inset 0 0 0 1px hsla(var(--ig-primary-500), var(--ig-primary-a));
          box-shadow: inset 0 0 0 1px hsla(var(--ig-primary-500), var(--ig-primary-a));
}

:host(:hover) [part="control checked"]::after,
:host(:focus-within) [part="control checked"]::after {
  background: var(--checked-hover-color, hsla(var(--ig-primary-900), var(--ig-primary-a)));
  -webkit-box-shadow: inset 0 0 0 1px var(--checked-hover-color, hsla(var(--ig-primary-900), var(--ig-primary-a)));
          box-shadow: inset 0 0 0 1px var(--checked-hover-color, hsla(var(--ig-primary-900), var(--ig-primary-a)));
}

:host(:not([disabled])[invalid]) [part~=label] {
  color: var(--invalid-color, hsla(var(--ig-error-500), var(--ig-error-a)));
}

:host([invalid]) [part~=control]::after {
  -webkit-box-shadow: inset 0 0 0 1px var(--invalid-color, hsla(var(--ig-error-500), var(--ig-error-a)));
          box-shadow: inset 0 0 0 1px var(--invalid-color, hsla(var(--ig-error-500), var(--ig-error-a)));
}

:host([invalid]:hover) [part~=indicator] {
  stroke: var(--invalid-color, hsla(var(--ig-error-500), var(--ig-error-a)));
}

:host([invalid]:hover) [part="indicator checked"] {
  stroke: hsla(var(--ig-surface-500), var(--ig-surface-a));
}

:host([invalid][checked]) [part="control checked"]::after {
  -webkit-box-shadow: inset 0 0 0 1px var(--invalid-color, hsla(var(--ig-error-500), var(--ig-error-a)));
          box-shadow: inset 0 0 0 1px var(--invalid-color, hsla(var(--ig-error-500), var(--ig-error-a)));
  background: var(--invalid-color, hsla(var(--ig-error-500), var(--ig-error-a)));
}

:host(:not([disabled])[invalid][indeterminate]) [part~=control]::before {
  background: var(--invalid-color, hsla(var(--ig-error-500), var(--ig-error-a)));
}
:host(:not([disabled])[invalid][indeterminate]) [part~=control]::after {
  -webkit-box-shadow: inset 0 0 0 1px var(--invalid-color, hsla(var(--ig-error-500), var(--ig-error-a)));
          box-shadow: inset 0 0 0 1px var(--invalid-color, hsla(var(--ig-error-500), var(--ig-error-a)));
  background: transparent;
}

:host([invalid]:hover) [part~=control]::after,
:host([invalid]:focus-within) [part~=control]::after {
  -webkit-box-shadow: inset 0 0 0 1px var(--invalid-hover-color, hsla(var(--ig-error-700), var(--ig-error-a)));
          box-shadow: inset 0 0 0 1px var(--invalid-hover-color, hsla(var(--ig-error-700), var(--ig-error-a)));
}

:host([invalid]:hover) [part="control checked"]::after,
:host([invalid]:focus-within) [part="control checked"]::after {
  background: var(--invalid-hover-color, hsla(var(--ig-error-700), var(--ig-error-a)));
}

:host([invalid][indeterminate]:hover) [part~=control]::before,
:host([invalid][indeterminate]:focus-within) [part~=control]::before {
  background: var(--invalid-hover-color, hsla(var(--ig-error-700), var(--ig-error-a)));
}
:host([invalid][indeterminate]:hover) [part~=control]::after,
:host([invalid][indeterminate]:focus-within) [part~=control]::after {
  -webkit-box-shadow: inset 0 0 0 1px var(--invalid-hover-color, hsla(var(--ig-error-700), var(--ig-error-a)));
          box-shadow: inset 0 0 0 1px var(--invalid-hover-color, hsla(var(--ig-error-700), var(--ig-error-a)));
  background: transparent;
}

:host([disabled]) [part~=label] {
  color: var(--disabled-color, hsla(var(--ig-gray-400), var(--ig-gray-a)));
}

:host([disabled]) [part~=control]::after,
:host([disabled][invalid]) [part~=control]::after {
  -webkit-box-shadow: inset 0 0 0 1px var(--disabled-color, hsla(var(--ig-gray-400), var(--ig-gray-a)));
          box-shadow: inset 0 0 0 1px var(--disabled-color, hsla(var(--ig-gray-400), var(--ig-gray-a)));
}

:host([disabled]) [part="control checked"]::after,
:host([disabled][invalid]) [part="control checked"]::after {
  background: var(--disabled-color, hsla(var(--ig-gray-400), var(--ig-gray-a)));
}

:host([disabled][indeterminate]) [part~=control]::before,
:host([disabled][invalid][indeterminate]) [part~=control]::before {
  background: var(--disabled-color, hsla(var(--ig-gray-400), var(--ig-gray-a)));
}
:host([disabled][indeterminate]) [part~=control]::after,
:host([disabled][invalid][indeterminate]) [part~=control]::after {
  -webkit-box-shadow: inset 0 0 0 1px var(--disabled-color, hsla(var(--ig-gray-400), var(--ig-gray-a)));
          box-shadow: inset 0 0 0 1px var(--disabled-color, hsla(var(--ig-gray-400), var(--ig-gray-a)));
  background: transparent;
}

[part~=focused] {
  position: relative;
}
[part~=focused]::after {
  content: "";
  position: absolute;
  top: -0.125rem;
  left: -0.125rem;
  -webkit-box-shadow: 0 0 0 1px hsla(var(--ig-gray-700), var(--ig-gray-a));
          box-shadow: 0 0 0 1px hsla(var(--ig-gray-700), var(--ig-gray-a));
  width: calc(100% + (0.125rem * 2));
  height: calc(100% + (0.125rem * 2));
}`;
