// igniteui-webcomponents/components/checkbox/themes/light/switch.base.css.js
import { css } from 'lit';
export const styles = css `:host {
  --is-large: clamp(0, (var(--component-size, 1) + 1) - var(--ig-size-large, 3), 1);
  --is-medium:
      min(
          clamp(0, (var(--component-size, 1) + 1) - var(--ig-size-medium, 2), 1),
          clamp(0, var(--ig-size-large, 3) - var(--component-size, 1), 1)
      );
  --is-small: clamp(0, var(--ig-size-medium) - var(--component-size, 1), 1);
  position: relative;
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
  scrollbar-width: var(--ig-scrollbar-size);
  scrollbar-color: var(--ig-scrollbar-thumb-background) var(--ig-scrollbar-track-background);
}
:host *,
:host *::before,
:host *::after {
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
}
:host ::-webkit-scrollbar {
  width: var(--ig-scrollbar-size);
  height: var(--ig-scrollbar-size);
  background: var(--ig-scrollbar-track-background);
}
:host ::-webkit-scrollbar-thumb {
  background: var(--ig-scrollbar-thumb-background);
}

[hidden] {
  display: none !important;
}

/* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */
:host {
  display: -webkit-inline-box;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: var(--ig-font-family);
  --size: 2.25rem;
  --thumb-size: 1.25rem;
  --thumb-offset: -0.0625rem;
}

input[type=checkbox] {
  position: absolute;
  width: 1px;
  height: 1px;
  margin: -1px;
  border: none;
  clip: rect(0, 0, 0, 0);
  outline: 0;
  pointer-events: none;
  overflow: hidden;
  -webkit-appearance: none;
     -moz-appearance: none;
          appearance: none;
}

label {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
      -ms-flex-flow: row nowrap;
          flex-flow: row nowrap;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  cursor: pointer;
  gap: 0.5rem;
}
label:hover [part~=thumb]::after {
  background: hsla(var(--ig-gray-600), 0.06);
}
label:focus-within [part~=thumb]::after {
  background: hsla(var(--ig-gray-600), 0.12);
}

[part~=control] {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -ms-flex-negative: 0;
      flex-shrink: 0;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  -webkit-user-select: none;
     -moz-user-select: none;
      -ms-user-select: none;
          user-select: none;
  -webkit-transition: all 0.2s ease-in-out;
  -o-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
}

[part~=thumb] {
  border-radius: clamp(0rem, calc(var(--ig-radius-factor) * 0.625rem), 0.625rem);
  position: relative;
  -webkit-animation-name: thumb-off;
          animation-name: thumb-off;
  -webkit-animation-duration: 0.2s;
          animation-duration: 0.2s;
  -webkit-animation-fill-mode: none;
          animation-fill-mode: none;
  -webkit-transition: all 0.2s ease-in-out;
  -o-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
}

[part~=thumb][part~=checked] {
  -webkit-animation-name: thumb-on;
          animation-name: thumb-on;
  -webkit-animation-duration: 0.2s;
          animation-duration: 0.2s;
  -webkit-animation-fill-mode: forwards;
          animation-fill-mode: forwards;
}

[part~=label] {
  font-family: var(--ig-subtitle-1-font-family, var(--ig-font-family));
  font-size: var(--ig-subtitle-1-font-size);
  font-weight: var(--ig-subtitle-1-font-weight);
  font-style: var(--ig-subtitle-1-font-style);
  line-height: var(--ig-subtitle-1-line-height);
  letter-spacing: var(--ig-subtitle-1-letter-spacing);
  text-transform: var(--ig-subtitle-1-text-transform);
  margin-top: var(--ig-subtitle-1-margin-top);
  margin-bottom: var(--ig-subtitle-1-margin-bottom);
  display: inline-block;
  color: hsla(var(--ig-gray-900), var(--ig-gray-a));
  cursor: pointer;
  -webkit-user-select: none;
     -moz-user-select: none;
      -ms-user-select: none;
          user-select: none;
  word-wrap: break-all;
  -webkit-transition: all 0.2s ease-in-out;
  -o-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
}

:host([disabled]) {
  -webkit-user-select: none;
     -moz-user-select: none;
      -ms-user-select: none;
          user-select: none;
  pointer-events: none;
  cursor: initial;
}
:host([disabled]) [part~=label] {
  color: hsla(var(--ig-gray-400), var(--ig-gray-a));
}

:host(:not([disabled])[invalid]) [part~=label] {
  color: hsla(var(--ig-error-500), var(--ig-error-a));
}
:host(:not([disabled])[invalid]) [part~=thumb] {
  background: hsla(var(--ig-error-500), var(--ig-error-a));
}

:host([label-position=before]) [part~=label] {
  -webkit-box-ordinal-group: 0;
      -ms-flex-order: -1;
          order: -1;
}

@-webkit-keyframes thumb-on {
  from {
    -webkit-margin-start: var(--thumb-offset);
            margin-inline-start: var(--thumb-offset);
  }
  to {
    -webkit-margin-start: calc(var(--size) - var(--thumb-size) - var(--thumb-offset));
            margin-inline-start: calc(var(--size) - var(--thumb-size) - var(--thumb-offset));
  }
}

@keyframes thumb-on {
  from {
    -webkit-margin-start: var(--thumb-offset);
            margin-inline-start: var(--thumb-offset);
  }
  to {
    -webkit-margin-start: calc(var(--size) - var(--thumb-size) - var(--thumb-offset));
            margin-inline-start: calc(var(--size) - var(--thumb-size) - var(--thumb-offset));
  }
}
@-webkit-keyframes thumb-off {
  from {
    -webkit-margin-start: calc(var(--size) - var(--thumb-size) - var(--thumb-offset));
            margin-inline-start: calc(var(--size) - var(--thumb-size) - var(--thumb-offset));
  }
  to {
    -webkit-margin-start: var(--thumb-offset);
            margin-inline-start: var(--thumb-offset);
  }
}
@keyframes thumb-off {
  from {
    -webkit-margin-start: calc(var(--size) - var(--thumb-size) - var(--thumb-offset));
            margin-inline-start: calc(var(--size) - var(--thumb-size) - var(--thumb-offset));
  }
  to {
    -webkit-margin-start: var(--thumb-offset);
            margin-inline-start: var(--thumb-offset);
  }
}`;
