// igniteui-webcomponents/components/checkbox/themes/light/switch.indigo.css.js
import { css } from 'lit';
export const styles = css `/* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */
:host {
  font-weight: 600;
  --size: 2.125rem;
  --thumb-size: 0.5rem;
  --thumb-offset: calc(var(--thumb-size) / 2);
}

[part~=control] {
  border-radius: clamp(0rem, calc(var(--ig-radius-factor) * 0.5rem), 0.5rem);
  width: var(--size);
  height: 1rem;
  -webkit-box-shadow: inset 0 0 0 2px var(--border-color, hsla(var(--ig-gray-500), var(--ig-gray-a)));
          box-shadow: inset 0 0 0 2px var(--border-color, hsla(var(--ig-gray-500), var(--ig-gray-a)));
  background: transparent;
}
[part~=control]::after {
  content: "";
  position: absolute;
  width: inherit;
  height: inherit;
  border-radius: inherit;
}

:host(:hover) [part~=control] {
  -webkit-box-shadow: 0 0 0 2px hsla(var(--ig-gray-200), 0.87);
          box-shadow: 0 0 0 2px hsla(var(--ig-gray-200), 0.87);
}
:host(:hover) [part~=control]::after {
  border: 2px solid var(--border-color, hsla(var(--ig-gray-500), var(--ig-gray-a)));
}

:host(:focus-within) [part~=control] {
  -webkit-box-shadow: 0 0 0 2px hsla(var(--ig-gray-300), 0.87);
          box-shadow: 0 0 0 2px hsla(var(--ig-gray-300), 0.87);
}
:host(:focus-within) [part~=control]::after {
  border: 2px solid var(--border-color, hsla(var(--ig-gray-500), var(--ig-gray-a)));
}

[part~=control][part~=checked] {
  background: var(--checked-color, hsla(var(--ig-secondary-500), var(--ig-secondary-a)));
  -webkit-box-shadow: inset 0 0 0 2px var(--checked-color, hsla(var(--ig-secondary-500), var(--ig-secondary-a)));
          box-shadow: inset 0 0 0 2px var(--checked-color, hsla(var(--ig-secondary-500), var(--ig-secondary-a)));
}

:host(:hover) [part~=control][part~=checked] {
  -webkit-box-shadow: 0 0 0 2px hsla(var(--ig-secondary-100), 0.87);
          box-shadow: 0 0 0 2px hsla(var(--ig-secondary-100), 0.87);
}
:host(:hover) [part~=control][part~=checked]::after {
  border: 2px solid var(--checked-color, hsla(var(--ig-secondary-500), var(--ig-secondary-a)));
}

:host(:focus-within) [part~=control][part~=checked] {
  -webkit-box-shadow: 0 0 0 2px hsla(var(--ig-secondary-200), 0.87);
          box-shadow: 0 0 0 2px hsla(var(--ig-secondary-200), 0.87);
}
:host(:focus-within) [part~=control][part~=checked]::after {
  border: 2px solid var(--checked-color, hsla(var(--ig-secondary-500), var(--ig-secondary-a)));
}

:host([invalid]) [part~=control],
:host([invalid]) [part~=control][part~=checked] {
  -webkit-box-shadow: inset 0 0 0 2px hsla(var(--ig-error-500), var(--ig-error-a));
          box-shadow: inset 0 0 0 2px hsla(var(--ig-error-500), var(--ig-error-a));
}

:host([invalid]) [part~=control][part~=checked],
:host([invalid]) [part~=thumb] {
  background: hsla(var(--ig-error-500), var(--ig-error-a));
}

:host([invalid]:hover) [part~=control] {
  -webkit-box-shadow: 0 0 0 2px hsla(var(--ig-error-100), 0.87);
          box-shadow: 0 0 0 2px hsla(var(--ig-error-100), 0.87);
}
:host([invalid]:hover) [part~=control]::after {
  border: 2px solid hsla(var(--ig-error-500), var(--ig-error-a));
}

:host([invalid]:focus-within) [part~=control],
:host([invalid]:focus-within) [part~=control][part~=checked] {
  -webkit-box-shadow: 0 0 0 2px hsla(var(--ig-error-200), 0.87);
          box-shadow: 0 0 0 2px hsla(var(--ig-error-200), 0.87);
}
:host([invalid]:focus-within) [part~=control]::after,
:host([invalid]:focus-within) [part~=control][part~=checked]::after {
  border: 2px solid hsla(var(--ig-error-500), var(--ig-error-a));
}

:host([disabled]) [part~=control],
:host([disabled][invalid]) [part~=control] {
  -webkit-box-shadow: inset 0 0 0 2px var(--disabled-color, hsla(var(--ig-gray-300), var(--ig-gray-a)));
          box-shadow: inset 0 0 0 2px var(--disabled-color, hsla(var(--ig-gray-300), var(--ig-gray-a)));
}

:host([disabled]) [part~=control][part~=checked],
:host([disabled][invalid]) [part~=control][part~=checked] {
  background: transparent;
}

[part~=thumb] {
  width: var(--thumb-size);
  height: var(--thumb-size);
  min-width: var(--thumb-size);
  -webkit-animation-name: thumb-off;
          animation-name: thumb-off;
  -webkit-margin-start: var(--thumb-offset);
          margin-inline-start: var(--thumb-offset);
  background: var(--thumb-color, hsla(var(--ig-gray-600), var(--ig-gray-a)));
}

[part~=thumb][part~=checked],
:host([invalid]) [part~=thumb][part~=checked] {
  background: hsla(var(--ig-surface-500), var(--ig-surface-a));
  -webkit-animation-name: thumb-on;
          animation-name: thumb-on;
  -webkit-margin-start: calc(var(--size) - var(--thumb-size) - var(--thumb-offset));
          margin-inline-start: calc(var(--size) - var(--thumb-size) - var(--thumb-offset));
}

:host([disabled]) [part~=thumb],
:host([disabled][invalid]) [part~=thumb] {
  background: var(--disabled-color, hsla(var(--ig-gray-300), var(--ig-gray-a)));
}

:host([disabled]) [part~=label] {
  color: var(--disabled-color, hsla(var(--ig-gray-300), var(--ig-gray-a)));
}`;
