// igniteui-webcomponents/components/checkbox/themes/light/switch.material.css.js
import { css } from 'lit';
export const styles = css `/* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */
[part~=control] {
  border-radius: clamp(0rem, calc(var(--ig-radius-factor) * 0.4375rem), 0.4375rem);
  width: var(--size);
  height: 0.875rem;
  background: hsla(var(--ig-gray-500), var(--ig-gray-a));
}

[part~=thumb] {
  width: var(--thumb-size);
  height: var(--thumb-size);
  min-width: var(--thumb-size);
  background: hsla(var(--ig-gray-50), var(--ig-gray-a));
  -webkit-box-shadow: var(--ig-elevation-2);
          box-shadow: var(--ig-elevation-2);
  -webkit-margin-start: var(--thumb-offset);
          margin-inline-start: var(--thumb-offset);
}
[part~=thumb]::after {
  border-radius: clamp(0rem, calc(var(--ig-radius-factor) * calc(2.5rem / 2)), calc(2.5rem / 2));
  position: absolute;
  content: "";
  top: calc(50% - calc(2.5rem / 2));
  left: calc(50% - calc(2.5rem / 2));
  width: 2.5rem;
  height: 2.5rem;
  overflow: hidden;
}
[part~=thumb]:hover {
  -webkit-box-shadow: var(--ig-elevation-3);
          box-shadow: var(--ig-elevation-3);
}

label:hover [part~=thumb][part~=checked]::after {
  background: hsla(var(--ig-secondary-500), 0.06);
}
label:focus-within [part~=thumb][part~=checked]::after {
  background: hsla(var(--ig-secondary-500), 0.12);
}

:host([invalid]) label:hover [part~=thumb]::after {
  background: hsla(var(--ig-error-500), 0.06);
}
:host([invalid]) label:focus-within [part~=thumb]::after {
  background: hsla(var(--ig-error-500), 0.12);
}

[part~=control][part~=checked] {
  background: hsla(var(--ig-secondary-200), var(--ig-secondary-a));
}

[part~=thumb][part~=checked] {
  background: hsla(var(--ig-secondary-500), var(--ig-secondary-a));
  -webkit-margin-start: calc(var(--size) - var(--thumb-size) - var(--thumb-offset));
          margin-inline-start: calc(var(--size) - var(--thumb-size) - var(--thumb-offset));
  -webkit-animation-name: thumb-on;
          animation-name: thumb-on;
  -webkit-animation-duration: 0.2s;
          animation-duration: 0.2s;
  -webkit-animation-fill-mode: forwards;
          animation-fill-mode: forwards;
}

:host([disabled]) [part~=thumb] {
  -webkit-box-shadow: var(--ig-elevation-1);
          box-shadow: var(--ig-elevation-1);
  background: hsla(var(--ig-gray-200), var(--ig-gray-a));
}
:host([disabled]) [part~=control] {
  background: hsla(var(--ig-gray-300), var(--ig-gray-a));
}
:host([disabled]) [part~=thumb][part~=checked] {
  background: hsla(var(--ig-secondary-100), var(--ig-secondary-a));
}
:host([disabled]) [part~=control][part~=checked] {
  background: hsla(var(--ig-secondary-50), var(--ig-secondary-a));
}

:host(:not([disabled])[invalid]) [part~=control] {
  background: hsla(var(--ig-error-500), 0.5);
}`;
