// igniteui-webcomponents/components/checkbox/themes/light/switch.fluent.css.js
import { css } from 'lit';
export const styles = css `/* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */
:host {
  --size: 2.5rem;
  --thumb-size: 0.75rem;
  --thumb-offset: calc(var(--thumb-size) / 2);
}

[part~=control] {
  border-radius: clamp(0rem, calc(var(--ig-radius-factor) * 0.625rem), 0.625rem);
  width: var(--size);
  height: 1.25rem;
  -webkit-box-shadow: inset 0 0 0 1px hsla(var(--ig-gray-700), var(--ig-gray-a));
          box-shadow: inset 0 0 0 1px hsla(var(--ig-gray-700), var(--ig-gray-a));
  background: transparent;
}

[part~=label] {
  font-family: var(--ig-body-2-font-family, var(--ig-font-family));
  font-size: var(--ig-body-2-font-size);
  font-weight: var(--ig-body-2-font-weight);
  font-style: var(--ig-body-2-font-style);
  line-height: var(--ig-body-2-line-height);
  letter-spacing: var(--ig-body-2-letter-spacing);
  text-transform: var(--ig-body-2-text-transform);
  margin-top: var(--ig-body-2-margin-top);
  margin-bottom: var(--ig-body-2-margin-bottom);
}

:host(:hover) [part~=control] {
  -webkit-box-shadow: inset 0 0 0 1px var(--hover-color, hsla(var(--ig-gray-800), var(--ig-gray-a)));
          box-shadow: inset 0 0 0 1px var(--hover-color, hsla(var(--ig-gray-800), var(--ig-gray-a)));
}

[part~=focused] {
  position: relative;
}
[part~=focused]::after {
  content: "";
  position: absolute;
  top: -0.125rem;
  left: -0.125rem;
  -webkit-box-shadow: 0 0 0 1px hsla(var(--ig-gray-700), var(--ig-gray-a));
          box-shadow: 0 0 0 1px hsla(var(--ig-gray-700), var(--ig-gray-a));
  width: calc(100% + (0.125rem * 2));
  height: calc(100% + (0.125rem * 2));
}

:host(:hover) [part~=thumb] {
  background: var(--hover-color, hsla(var(--ig-gray-800), var(--ig-gray-a)));
}

[part~=control][part~=checked] {
  background: hsla(var(--ig-primary-500), var(--ig-primary-a));
  -webkit-box-shadow: inset 0 0 0 1px hsla(var(--ig-primary-500), var(--ig-primary-a));
          box-shadow: inset 0 0 0 1px hsla(var(--ig-primary-500), var(--ig-primary-a));
}

:host(:hover) [part~=control][part~=checked] {
  background: var(--checked-hover-color, hsla(var(--ig-primary-900), var(--ig-primary-a)));
  -webkit-box-shadow: inset 0 0 0 1px var(--checked-hover-color, hsla(var(--ig-primary-900), var(--ig-primary-a)));
          box-shadow: inset 0 0 0 1px var(--checked-hover-color, hsla(var(--ig-primary-900), var(--ig-primary-a)));
}

:host([invalid]) [part~=control] {
  -webkit-box-shadow: inset 0 0 0 1px var(--invalid-color, hsla(var(--ig-error-500), var(--ig-error-a)));
          box-shadow: inset 0 0 0 1px var(--invalid-color, hsla(var(--ig-error-500), var(--ig-error-a)));
}

:host([invalid]) [part~=control][part~=checked] {
  background: var(--invalid-color, hsla(var(--ig-error-500), var(--ig-error-a)));
  -webkit-box-shadow: inset 0 0 0 1px var(--invalid-color, hsla(var(--ig-error-500), var(--ig-error-a)));
          box-shadow: inset 0 0 0 1px var(--invalid-color, hsla(var(--ig-error-500), var(--ig-error-a)));
}

:host([invalid]:hover) [part~=control] {
  -webkit-box-shadow: inset 0 0 0 1px var(--invalid-hover-color, hsla(var(--ig-error-700), var(--ig-error-a)));
          box-shadow: inset 0 0 0 1px var(--invalid-hover-color, hsla(var(--ig-error-700), var(--ig-error-a)));
}

:host([invalid]:hover) [part~=thumb] {
  background: var(--invalid-hover-color, hsla(var(--ig-error-700), var(--ig-error-a)));
}

:host([invalid]) [part~=thumb][part~=checked],
:host([invalid]:hover) [part~=thumb][part~=checked] {
  background: hsla(var(--ig-surface-500), var(--ig-surface-a));
}

:host([invalid]:hover) [part~=control][part~=checked] {
  -webkit-box-shadow: inset 0 0 0 1px var(--invalid-hover-color, hsla(var(--ig-error-700), var(--ig-error-a)));
          box-shadow: inset 0 0 0 1px var(--invalid-hover-color, hsla(var(--ig-error-700), var(--ig-error-a)));
  background: var(--invalid-hover-color, hsla(var(--ig-error-700), var(--ig-error-a)));
}

:host([disabled]) [part~=label] {
  color: var(--disabled-color, hsla(var(--ig-gray-400), var(--ig-gray-a)));
}

:host([disabled]) [part~=control],
:host([disabled][invalid]) [part~=control] {
  -webkit-box-shadow: inset 0 0 0 1px var(--disabled-color, hsla(var(--ig-gray-400), var(--ig-gray-a)));
          box-shadow: inset 0 0 0 1px var(--disabled-color, hsla(var(--ig-gray-400), var(--ig-gray-a)));
}

:host([disabled]) [part~=control][part~=checked],
:host([disabled][invalid]) [part~=control][part~=checked],
:host([disabled]) [part~=thumb],
:host([disabled][invalid]) [part~=thumb] {
  background: var(--disabled-color, hsla(var(--ig-gray-400), var(--ig-gray-a)));
}

[part~=thumb] {
  width: var(--thumb-size);
  height: var(--thumb-size);
  min-width: var(--thumb-size);
  -webkit-margin-start: var(--thumb-offset);
          margin-inline-start: var(--thumb-offset);
  background: var(--thumb-color, hsla(var(--ig-gray-700), var(--ig-gray-a)));
}

[part~=thumb][part~=checked],
:host(:hover) [part~=thumb][part~=checked] {
  background: hsla(var(--ig-gray-50), var(--ig-gray-a));
  -webkit-margin-start: calc(var(--size) - var(--thumb-size) - var(--thumb-offset));
          margin-inline-start: calc(var(--size) - var(--thumb-size) - var(--thumb-offset));
}

:host([disabled]) [part~=thumb][part~=checked],
:host([disabled][invalid]) [part~=thumb][part~=checked] {
  background: var(--disabled-thumb-color, hsla(var(--ig-gray-100), var(--ig-gray-a)));
}`;
