// igniteui-webcomponents/components/checkbox/themes/light/switch.bootstrap.css.js
import { css } from 'lit';
export const styles = css `/* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */
:host {
  --size: 2rem;
  --thumb-size: 0.625rem;
  --thumb-offset: calc(var(--thumb-size) / 4);
}

[part~=control] {
  border-radius: clamp(0rem, calc(var(--ig-radius-factor) * 0.25rem), 0.25rem);
  width: var(--size);
  height: 1rem;
  -webkit-box-shadow: inset 0 0 0 1px hsla(var(--ig-gray-400), var(--ig-gray-a));
          box-shadow: inset 0 0 0 1px hsla(var(--ig-gray-400), var(--ig-gray-a));
  background: transparent;
  -webkit-transition: background-color 0.15s ease-in-out;
  -o-transition: background-color 0.15s ease-in-out;
  transition: background-color 0.15s ease-in-out;
}
[part~=control]::after {
  content: "";
  position: absolute;
  width: inherit;
  height: inherit;
  border-radius: inherit;
}

[part~=label] {
  -webkit-transition: none;
  -o-transition: none;
  transition: none;
}

:host(:focus-within) [part~=control] {
  -webkit-box-shadow: 0 0 0 4px hsla(var(--ig-primary-100), var(--ig-primary-a));
          box-shadow: 0 0 0 4px hsla(var(--ig-primary-100), var(--ig-primary-a));
}
:host(:focus-within) [part~=control]::after {
  border: 1px solid hsla(var(--ig-primary-200), var(--ig-primary-a));
}
:host(:focus-within) [part~=thumb] {
  background: hsla(var(--ig-primary-200), var(--ig-primary-a));
}

:host(:focus-within) [part~=control][part~=checked] {
  -webkit-box-shadow: 0 0 0 4px hsla(var(--ig-primary-200), var(--ig-primary-a));
          box-shadow: 0 0 0 4px hsla(var(--ig-primary-200), var(--ig-primary-a));
}
:host(:focus-within) [part~=control][part~=checked]::after {
  border: 1px solid hsla(var(--ig-primary-500), var(--ig-primary-a));
}

[part~=control][part~=checked] {
  background: hsla(var(--ig-primary-500), var(--ig-primary-a));
  -webkit-box-shadow: inset 0 0 0 1px hsla(var(--ig-primary-500), var(--ig-primary-a));
          box-shadow: inset 0 0 0 1px hsla(var(--ig-primary-500), var(--ig-primary-a));
}

:host([invalid]) [part~=control],
:host([invalid]) [part~=control][part~=checked] {
  -webkit-box-shadow: inset 0 0 0 1px hsla(var(--ig-error-500), var(--ig-error-a));
          box-shadow: inset 0 0 0 1px hsla(var(--ig-error-500), var(--ig-error-a));
}

:host([invalid]) [part~=control][part~=checked],
:host([invalid]) [part~=thumb] {
  background: hsla(var(--ig-error-500), var(--ig-error-a));
}

:host([invalid]:focus-within) [part~=control] {
  -webkit-box-shadow: 0 0 0 4px hsla(var(--ig-error-100), var(--ig-error-a));
          box-shadow: 0 0 0 4px hsla(var(--ig-error-100), var(--ig-error-a));
  background: transparent;
}
:host([invalid]:focus-within) [part~=control]::after {
  border: 1px solid hsla(var(--ig-error-500), var(--ig-error-a));
}

:host([invalid]:focus-within) [part~=control][part~=checked] {
  -webkit-box-shadow: 0 0 0 4px hsla(var(--ig-error-200), var(--ig-error-a));
          box-shadow: 0 0 0 4px hsla(var(--ig-error-200), var(--ig-error-a));
  background: hsla(var(--ig-error-500), var(--ig-error-a));
}
:host([invalid]:focus-within) [part~=control][part~=checked]::after {
  border: 1px solid hsla(var(--ig-error-500), var(--ig-error-a));
}

:host([disabled]) [part~=control],
:host([disabled][invalid]) [part~=control] {
  -webkit-box-shadow: inset 0 0 0 1px hsla(var(--ig-gray-300), var(--ig-gray-a));
          box-shadow: inset 0 0 0 1px hsla(var(--ig-gray-300), var(--ig-gray-a));
}

:host([disabled]) [part~=control][part~=checked],
:host([disabled][invalid]) [part~=control][part~=checked] {
  background: hsla(var(--ig-primary-200), var(--ig-primary-a));
  -webkit-box-shadow: inset 0 0 0 1px hsla(var(--ig-primary-200), var(--ig-primary-a));
          box-shadow: inset 0 0 0 1px hsla(var(--ig-primary-200), var(--ig-primary-a));
}

[part~=thumb] {
  border-radius: clamp(0rem, calc(var(--ig-radius-factor) * 0.125rem), 0.125rem);
  width: var(--thumb-size);
  height: var(--thumb-size);
  min-width: var(--thumb-size);
  -webkit-margin-start: var(--thumb-offset);
          margin-inline-start: var(--thumb-offset);
  background: hsla(var(--ig-gray-400), var(--ig-gray-a));
}

:host(:focus-within) [part~=thumb][part~=checked],
[part~=thumb][part~=checked] {
  background: white;
  -webkit-margin-start: calc(var(--size) - var(--thumb-size) - var(--thumb-offset));
          margin-inline-start: calc(var(--size) - var(--thumb-size) - var(--thumb-offset));
}

:host([disabled]) [part~=thumb],
:host([disabled][invalid]) [part~=thumb] {
  background: hsla(var(--ig-gray-300), var(--ig-gray-a));
}

:host([disabled]) [part~=thumb][part~=checked],
:host([invalid]) [part~=thumb][part~=checked],
:host([disabled][invalid]) [part~=thumb][part~=checked] {
  background: white;
}`;
