// igniteui-webcomponents/components/radio/themes/light/radio.indigo.css.js
import { css } from 'lit';
export const styles = css `/* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */
:host {
  font-weight: 600;
}

:host [part~=control] {
  width: var(--size, 1.25rem);
  height: var(--size, 1.25rem);
  min-width: var(--size, 1.25rem);
  -webkit-transition: none;
  -o-transition: none;
  transition: none;
  background: none;
}

:host(:not([invalid])) [part=control]::after {
  border-color: var(--border-color, hsla(var(--ig-gray-600), var(--ig-gray-a)));
}

:host(:hover) [part=control] {
  -webkit-box-shadow: 0 0 0 0.25rem hsla(var(--ig-gray-200), 0.87);
          box-shadow: 0 0 0 0.25rem hsla(var(--ig-gray-200), 0.87);
}

:host(:focus-within) [part=control] {
  -webkit-box-shadow: 0 0 0 0.25rem hsla(var(--ig-gray-300), 0.87);
          box-shadow: 0 0 0 0.25rem hsla(var(--ig-gray-300), 0.87);
}

:host(:not([invalid])) [part="control checked"]::after {
  border-color: var(--checked-color, hsla(var(--ig-secondary-500), var(--ig-secondary-a)));
}

:host(:not([invalid])) [part="control checked"]::before {
  background: var(--checked-color, hsla(var(--ig-secondary-500), var(--ig-secondary-a)));
  border-color: var(--checked-color, hsla(var(--ig-secondary-500), var(--ig-secondary-a)));
}

:host(:hover) [part="control checked"] {
  background: transparent;
  -webkit-box-shadow: 0 0 0 0.25rem hsla(var(--ig-secondary-100), 0.87);
          box-shadow: 0 0 0 0.25rem hsla(var(--ig-secondary-100), 0.87);
}

:host(:focus-within) [part="control checked"] {
  background: transparent;
  -webkit-box-shadow: 0 0 0 0.25rem hsla(var(--ig-secondary-200), 0.87);
          box-shadow: 0 0 0 0.25rem hsla(var(--ig-secondary-200), 0.87);
}

:host(:not([disabled])[invalid]) [part="control checked"]::before {
  background: hsla(var(--ig-error-500), var(--ig-error-a));
}

:host([invalid]:hover) [part~=control] {
  -webkit-box-shadow: 0 0 0 0.25rem hsla(var(--ig-error-100), 0.87);
          box-shadow: 0 0 0 0.25rem hsla(var(--ig-error-100), 0.87);
}

:host([invalid]:focus-within) [part~=control] {
  -webkit-box-shadow: 0 0 0 0.25rem hsla(var(--ig-error-200), 0.87);
          box-shadow: 0 0 0 0.25rem hsla(var(--ig-error-200), 0.87);
}

:host([disabled]) {
  pointer-events: none;
  -webkit-user-select: none;
     -moz-user-select: none;
      -ms-user-select: none;
          user-select: none;
}
:host([disabled]) [part=control]::after {
  background: transparent;
}

:host([disabled]) [part=control]::after,
:host([disabled]) [part="control checked"]::after {
  border-color: var(--disabled-color, hsla(var(--ig-gray-300), var(--ig-gray-a)));
}

:host([disabled]) [part="control checked"]::before {
  background: var(--disabled-color, hsla(var(--ig-gray-300), var(--ig-gray-a)));
}

:host([disabled]) [part~=label] {
  color: var(--disabled-color, hsla(var(--ig-gray-300), var(--ig-gray-a)));
}`;
