// igniteui-webcomponents/components/radio/themes/light/radio.bootstrap.css.js
import { css } from 'lit';
export const styles = css `/* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */
:host::part(base) {
  -webkit-transition: none;
  -o-transition: none;
  transition: none;
}

:host [part~=control] {
  --size: 1rem;
  width: var(--size);
  height: var(--size);
  min-width: var(--size);
  background: none;
  -webkit-transition: transparent;
  -o-transition: transparent;
  transition: transparent;
}
:host [part~=control]::before, :host [part~=control]::after {
  -webkit-transition: none;
  -o-transition: none;
  transition: none;
}
:host [part~=control]::before {
  -webkit-transform: translate(-50%, -50%) scale(0.375);
      -ms-transform: translate(-50%, -50%) scale(0.375);
          transform: translate(-50%, -50%) scale(0.375);
}
:host [part~=control]::after {
  width: inherit;
  height: inherit;
  min-width: inherit;
  border-color: hsla(var(--ig-gray-400), var(--ig-gray-a));
  border-width: calc(var(--size, 0.625rem) / 10);
}

:host(:focus-within) [part~=control] {
  -webkit-box-shadow: 0 0 0 calc(var(--size, 1rem) / 4) hsla(var(--ig-primary-100), var(--ig-primary-a));
          box-shadow: 0 0 0 calc(var(--size, 1rem) / 4) hsla(var(--ig-primary-100), var(--ig-primary-a));
}
:host(:focus-within) [part~=control]::after {
  border-color: hsla(var(--ig-primary-500), var(--ig-primary-a));
}

:host [part="control checked"]::after {
  background: hsla(var(--ig-primary-500), var(--ig-primary-a));
  border-color: hsla(var(--ig-primary-500), var(--ig-primary-a));
}
:host [part="control checked"]::before {
  background: white;
}

:host(:not([disabled])[invalid]) [part~=control]::after {
  border-color: hsla(var(--ig-error-500), var(--ig-error-a));
}
:host(:not([disabled])[invalid]) [part="control checked"]::after {
  background: hsla(var(--ig-error-500), var(--ig-error-a));
}

:host([invalid]:focus-within) [part~=control] {
  -webkit-box-shadow: 0 0 0 calc(var(--size, 1rem) / 4) hsla(var(--ig-error-100), var(--ig-error-a));
          box-shadow: 0 0 0 calc(var(--size, 1rem) / 4) hsla(var(--ig-error-100), var(--ig-error-a));
}

:host([disabled]) [part=control]::after {
  border-color: hsla(var(--ig-gray-300), var(--ig-gray-a));
}
:host([disabled]) [part="control checked"]::after {
  background: hsla(var(--ig-primary-200), var(--ig-primary-a));
  border-color: hsla(var(--ig-primary-200), var(--ig-primary-a));
}
:host([disabled]) [part="control checked"]::before {
  background: white;
}`;
