// igniteui-webcomponents/components/radio/themes/light/radio.fluent.css.js
import { css } from 'lit';
export const styles = css `/* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */
:host [part~=base] {
  -webkit-transition-duration: 0.1s;
       -o-transition-duration: 0.1s;
          transition-duration: 0.1s;
  gap: 0.375rem;
}

:host [part~=control] {
  width: var(--size, 1.25rem);
  height: var(--size, 1.25rem);
  min-width: var(--size, 1.25rem);
  background: transparent;
  -webkit-transition: none;
  -o-transition: none;
  transition: none;
}
:host [part~=control]::before, :host [part~=control]::after {
  -webkit-transition-duration: 0.1s;
       -o-transition-duration: 0.1s;
          transition-duration: 0.1s;
}
:host [part~=control]::before {
  -webkit-transform: translate(-50%, -50%) scale(0.5);
      -ms-transform: translate(-50%, -50%) scale(0.5);
          transform: translate(-50%, -50%) scale(0.5);
}
:host [part~=control]::after {
  border: calc(var(--size, 20px) / 20) solid hsla(var(--ig-gray-900), var(--ig-gray-a));
}

:host [part~=label] {
  font-family: var(--ig-body-2-font-family, var(--ig-font-family));
  font-size: var(--ig-body-2-font-size);
  font-weight: var(--ig-body-2-font-weight);
  font-style: var(--ig-body-2-font-style);
  line-height: var(--ig-body-2-line-height);
  letter-spacing: var(--ig-body-2-letter-spacing);
  text-transform: var(--ig-body-2-text-transform);
  margin-top: var(--ig-body-2-margin-top);
  margin-bottom: var(--ig-body-2-margin-bottom);
}

:host(:hover) [part="control checked"]::after {
  border-color: var(--checked-hover-color, hsla(var(--ig-primary-800), var(--ig-primary-a)));
}
:host(:hover) [part="control checked"]::before {
  background: var(--checked-hover-color, hsla(var(--ig-primary-800), var(--ig-primary-a)));
}

:host(:hover) [part=control]::before,
:host(:focus-within) [part=control]::before {
  background: var(--hover-color, hsla(var(--ig-gray-700), var(--ig-gray-a)));
}

:host [part="control checked"]::before {
  background: var(--checked-color, hsla(var(--ig-primary-500), var(--ig-primary-a)));
}
:host [part="control checked"]::after {
  border-color: var(--checked-color, hsla(var(--ig-primary-500), var(--ig-primary-a)));
}

:host(:not([disabled])[invalid]) [part="control checked"]::before {
  background: hsla(var(--ig-error-500), var(--ig-error-a));
}
:host(:not([disabled])[invalid]) [part~=control]::after {
  border-color: hsla(var(--ig-error-500), var(--ig-error-a));
}

:host(:not([disabled])[invalid]:hover) [part="control checked"]::after {
  border-color: var(--invalid-hover-color, hsla(var(--ig-error-700), var(--ig-error-a)));
}
:host(:not([disabled])[invalid]:hover) [part="control checked"]::before {
  background: var(--invalid-hover-color, hsla(var(--ig-error-700), var(--ig-error-a)));
}

:host(:not([disabled])[invalid]:hover) [part=control]::after,
:host(:not([disabled])[invalid]:focus-within) [part=control]::after {
  border-color: hsla(var(--ig-error-500), var(--ig-error-a));
}
:host(:not([disabled])[invalid]:hover) [part=control]::before,
:host(:not([disabled])[invalid]:focus-within) [part=control]::before {
  background: hsla(var(--ig-error-500), var(--ig-error-a));
}

:host([disabled]) [part~=label] {
  color: var(--disabled-label-color, hsla(var(--ig-gray-500), var(--ig-gray-a)));
}
:host([disabled]) [part~=control]::after {
  border-color: var(--disabled-color, hsla(var(--ig-gray-400), var(--ig-gray-a)));
}
:host([disabled]) [part="control checked"]::before {
  background: var(--disabled-color, hsla(var(--ig-gray-400), var(--ig-gray-a)));
}

[part~=focused] {
  position: relative;
}
[part~=focused]::after {
  content: "";
  position: absolute;
  top: -0.125rem;
  left: -0.125rem;
  -webkit-box-shadow: 0 0 0 1px hsla(var(--ig-gray-700), var(--ig-gray-a));
          box-shadow: 0 0 0 1px hsla(var(--ig-gray-700), var(--ig-gray-a));
  width: calc(100% + (0.125rem * 2));
  height: calc(100% + (0.125rem * 2));
}`;
