// igniteui-webcomponents/components/radio/themes/light/radio.base.css.js
import { css } from 'lit';
export const styles = css `:host {
  --is-large: clamp(0, (var(--component-size, 1) + 1) - var(--ig-size-large, 3), 1);
  --is-medium:
      min(
          clamp(0, (var(--component-size, 1) + 1) - var(--ig-size-medium, 2), 1),
          clamp(0, var(--ig-size-large, 3) - var(--component-size, 1), 1)
      );
  --is-small: clamp(0, var(--ig-size-medium) - var(--component-size, 1), 1);
  position: relative;
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
  scrollbar-width: var(--ig-scrollbar-size);
  scrollbar-color: var(--ig-scrollbar-thumb-background) var(--ig-scrollbar-track-background);
}
:host *,
:host *::before,
:host *::after {
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
}
:host ::-webkit-scrollbar {
  width: var(--ig-scrollbar-size);
  height: var(--ig-scrollbar-size);
  background: var(--ig-scrollbar-track-background);
}
:host ::-webkit-scrollbar-thumb {
  background: var(--ig-scrollbar-thumb-background);
}

[hidden] {
  display: none !important;
}

/* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */
:host {
  display: -webkit-inline-box;
  display: -ms-inline-flexbox;
  display: inline-flex;
  color: hsla(var(--ig-gray-900), var(--ig-gray-a));
  font-family: var(--ig-font-family);
}

input[type=radio] {
  position: absolute;
  width: 1px;
  height: 1px;
  margin: -1px;
  border: none;
  clip: rect(0, 0, 0, 0);
  outline: 0;
  pointer-events: none;
  overflow: hidden;
  -webkit-appearance: none;
     -moz-appearance: none;
          appearance: none;
}

[part~=base] {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  gap: 0.5rem;
  cursor: pointer;
  -webkit-transition: color 0.15s ease-in;
  -o-transition: color 0.15s ease-in;
  transition: color 0.15s ease-in;
}

[part~=label] {
  font-family: var(--ig-subtitle-1-font-family, var(--ig-font-family));
  font-size: var(--ig-subtitle-1-font-size);
  font-weight: var(--ig-subtitle-1-font-weight);
  font-style: var(--ig-subtitle-1-font-style);
  line-height: var(--ig-subtitle-1-line-height);
  letter-spacing: var(--ig-subtitle-1-letter-spacing);
  text-transform: var(--ig-subtitle-1-text-transform);
  margin-top: var(--ig-subtitle-1-margin-top);
  margin-bottom: var(--ig-subtitle-1-margin-bottom);
}

[part~=control] {
  border-radius: clamp(0rem, calc(var(--ig-radius-factor) * var(--size)), var(--size));
  --size: 1.25rem;
  position: relative;
  display: -webkit-inline-box;
  display: -ms-inline-flexbox;
  display: inline-flex;
  width: calc(var(--size) * 2);
  height: calc(var(--size) * 2);
  min-width: calc(var(--size) * 2);
  -webkit-user-select: none;
     -moz-user-select: none;
      -ms-user-select: none;
          user-select: none;
  -webkit-transition: background 0.15s ease-in;
  -o-transition: background 0.15s ease-in;
  transition: background 0.15s ease-in;
}
[part~=control]::before, [part~=control]::after {
  border-radius: clamp(0rem, calc(var(--ig-radius-factor) * calc(var(--size) / 2)), calc(var(--size) / 2));
  position: absolute;
  content: "";
  width: var(--size);
  height: var(--size);
  left: 50%;
  top: 50%;
  -webkit-transform: translate(-50%, -50%);
      -ms-transform: translate(-50%, -50%);
          transform: translate(-50%, -50%);
  -webkit-transition: all 0.15s ease-in;
  -o-transition: all 0.15s ease-in;
  transition: all 0.15s ease-in;
}
[part~=control]::before {
  -webkit-backface-visibility: hidden;
          backface-visibility: hidden;
  -webkit-transform: translate(-50%, -50%) scale(0.1);
      -ms-transform: translate(-50%, -50%) scale(0.1);
          transform: translate(-50%, -50%) scale(0.1);
  -webkit-transform-origin: center;
      -ms-transform-origin: center;
          transform-origin: center;
  z-index: 1;
}
[part~=control]::after {
  border: calc(var(--size) / 10) solid hsla(var(--ig-gray-600), var(--ig-gray-a));
}

[part="control checked"]::before {
  border-color: hsla(var(--ig-secondary-500), var(--ig-secondary-a));
  background: hsla(var(--ig-secondary-500), var(--ig-secondary-a));
  -webkit-transform: translate(-50%, -50%) scale(0.5);
      -ms-transform: translate(-50%, -50%) scale(0.5);
          transform: translate(-50%, -50%) scale(0.5);
}
[part="control checked"]::after {
  border-color: hsla(var(--ig-secondary-500), var(--ig-secondary-a));
}

:host(:not([disabled])[invalid]) [part~=label] {
  color: hsla(var(--ig-error-500), var(--ig-error-a));
}
:host(:not([disabled])[invalid]) [part~=control]::after {
  border-color: hsla(var(--ig-error-500), var(--ig-error-a));
}

:host([disabled]) {
  pointer-events: none;
  -webkit-user-select: none;
     -moz-user-select: none;
      -ms-user-select: none;
          user-select: none;
}
:host([disabled]) [part~=label] {
  color: hsla(var(--ig-gray-400), var(--ig-gray-a));
}
:host([disabled]) [part~=control]::after {
  border-color: hsla(var(--ig-gray-400), var(--ig-gray-a));
}
:host([disabled]) [part="control checked"]::before {
  background: hsla(var(--ig-gray-400), var(--ig-gray-a));
  border-color: transparent;
}

:host([label-position=before]) [part~=label] {
  -webkit-box-ordinal-group: 0;
      -ms-flex-order: -1;
          order: -1;
}`;
