// igniteui-webcomponents/components/button/button.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
import { html } from 'lit';
import { property } from 'lit/decorators.js';
import { themes } from '../../theming/theming-decorator.js';
import { IgcButtonBaseComponent } from './button-base.js';
import { styles } from './themes/button/light/button.base.css.js';
import { styles as bootstrap } from './themes/button/light/button.bootstrap.css.js';
import { styles as fluent } from './themes/button/light/button.fluent.css.js';
import { styles as indigo } from './themes/button/light/button.indigo.css.js';
import { styles as material } from './themes/button/light/button.material.css.js';
let IgcButtonComponent = class IgcButtonComponent extends IgcButtonBaseComponent {
    constructor() {
        super();
        this.variant = 'contained';
        this.size = 'medium';
    }
    get classes() {
        const { size, variant } = this;
        return {
            flat: variant === 'flat',
            outlined: variant === 'outlined',
            contained: variant === 'contained',
            fab: variant === 'fab',
            small: size === 'small',
            medium: size === 'medium',
            large: size === 'large',
            disabled: this.disabled,
        };
    }
    renderContent() {
        return html `
      <slot name="prefix"></slot>
      <slot></slot>
      <slot name="suffix"></slot>
    `;
    }
};
IgcButtonComponent.tagName = 'igc-button';
IgcButtonComponent.styles = styles;
__decorate([
    property({ reflect: true })
], IgcButtonComponent.prototype, "variant", void 0);
IgcButtonComponent = __decorate([
    themes({ bootstrap, indigo, fluent, material })
], IgcButtonComponent);
export default IgcButtonComponent;
