// igniteui-webcomponents/components/button/button-base.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
import { html, LitElement } from 'lit';
import { property, query } from 'lit/decorators.js';
import { classMap } from 'lit/directives/class-map.js';
import { ifDefined } from 'lit/directives/if-defined.js';
import { EventEmitterMixin } from '../common//mixins/event-emitter.js';
import { alternateName } from '../common/decorators/alternateName.js';
import { blazorDeepImport } from '../common/decorators/blazorDeepImport.js';
import { SizableMixin } from '../common/mixins/sizable.js';
let IgcButtonBaseComponent = class IgcButtonBaseComponent extends SizableMixin(EventEmitterMixin(LitElement)) {
    constructor() {
        super(...arguments);
        this.disabled = false;
    }
    set ariaLabel(value) {
        const oldVal = this._ariaLabel;
        this._ariaLabel = value;
        if (this.hasAttribute('aria-label')) {
            this.removeAttribute('aria-label');
        }
        this.requestUpdate('ariaLabel', oldVal);
    }
    get ariaLabel() {
        return this._ariaLabel;
    }
    focus(options) {
        this.nativeElement.focus(options);
    }
    blur() {
        this.nativeElement.blur();
    }
    handleFocus() {
        this.emitEvent('igcFocus');
    }
    handleBlur() {
        this.emitEvent('igcBlur');
    }
    get classes() {
        return {};
    }
    renderButton() {
        return html `
      <button
        part="base"
        aria-label=${ifDefined(this.ariaLabel)}
        .disabled=${this.disabled}
        class=${classMap(this.classes)}
        type=${ifDefined(this.type)}
        @focus=${this.handleFocus}
        @blur=${this.handleBlur}
      >
        ${this.renderContent()}
      </button>
    `;
    }
    renderLinkButton() {
        return html `
      <a
        part="base"
        role="button"
        href=${ifDefined(this.href)}
        target=${ifDefined(this.target)}
        download=${ifDefined(this.download)}
        rel=${ifDefined(this.rel)}
        aria-disabled=${this.disabled ? 'true' : 'false'}
        aria-label=${ifDefined(this.ariaLabel)}
        class=${classMap(this.classes)}
        @focus=${!this.disabled && this.handleFocus}
        @blur=${!this.disabled && this.handleBlur}
      >
        ${this.renderContent()}
      </a>
    `;
    }
    render() {
        const link = this.href !== undefined;
        return link ? this.renderLinkButton() : this.renderButton();
    }
};
IgcButtonBaseComponent.shadowRootOptions = {
    ...LitElement.shadowRootOptions,
    delegatesFocus: true,
};
__decorate([
    query('[part="base"]', true)
], IgcButtonBaseComponent.prototype, "nativeElement", void 0);
__decorate([
    alternateName('displayType'),
    property()
], IgcButtonBaseComponent.prototype, "type", void 0);
__decorate([
    property()
], IgcButtonBaseComponent.prototype, "href", void 0);
__decorate([
    property()
], IgcButtonBaseComponent.prototype, "download", void 0);
__decorate([
    property()
], IgcButtonBaseComponent.prototype, "target", void 0);
__decorate([
    property()
], IgcButtonBaseComponent.prototype, "rel", void 0);
__decorate([
    property({ type: Boolean, reflect: true })
], IgcButtonBaseComponent.prototype, "disabled", void 0);
__decorate([
    property({ attribute: 'aria-label' })
], IgcButtonBaseComponent.prototype, "ariaLabel", null);
__decorate([
    alternateName('focusComponent')
], IgcButtonBaseComponent.prototype, "focus", null);
__decorate([
    alternateName('blurComponent')
], IgcButtonBaseComponent.prototype, "blur", null);
IgcButtonBaseComponent = __decorate([
    blazorDeepImport
], IgcButtonBaseComponent);
export { IgcButtonBaseComponent };
