// igniteui-webcomponents/components/slider/slider.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
import { html } from 'lit';
import { property } from 'lit/decorators.js';
import { EventEmitterMixin } from '../common/mixins/event-emitter.js';
import { IgcSliderBaseComponent } from './slider-base.js';
export default class IgcSliderComponent extends EventEmitterMixin(IgcSliderBaseComponent) {
    constructor() {
        super(...arguments);
        this._value = 0;
    }
    set value(val) {
        const oldVal = this._value;
        this._value = this.validateValue(val);
        this.requestUpdate('value', oldVal);
    }
    get value() {
        return this._value;
    }
    set ariaLabel(value) {
        const oldVal = this._ariaLabel;
        this._ariaLabel = value;
        if (this.hasAttribute('aria-label')) {
            this.removeAttribute('aria-label');
        }
        this.requestUpdate('ariaLabel', oldVal);
    }
    get ariaLabel() {
        return this._ariaLabel;
    }
    get activeValue() {
        return this.value;
    }
    normalizeValue() {
        this._value = this.validateValue(this._value);
    }
    getTrackStyle() {
        const position = this.valueToFraction(this.value);
        const filledTrackStyle = {
            width: `${position * 100}%`,
        };
        return filledTrackStyle;
    }
    updateValue(increment) {
        const oldValue = this.value;
        this.value = this.value + increment;
        if (oldValue === this.value) {
            return false;
        }
        this.emitInputEvent();
        return true;
    }
    emitInputEvent() {
        this.emitEvent('igcInput', { detail: this.value });
    }
    emitChangeEvent() {
        this.emitEvent('igcChange', { detail: this.value });
    }
    stepUp(stepIncrement = 1) {
        this.value = this.value + stepIncrement * this.step;
    }
    stepDown(stepDecrement = 1) {
        this.value = this.value - stepDecrement * this.step;
    }
    renderThumbs() {
        return html `${this.renderThumb(this.value, this.ariaLabel)}`;
    }
}
IgcSliderComponent.tagName = 'igc-slider';
__decorate([
    property({ type: Number })
], IgcSliderComponent.prototype, "value", null);
__decorate([
    property({ attribute: 'aria-label' })
], IgcSliderComponent.prototype, "ariaLabel", null);
