// igniteui-webcomponents/components/slider/themes/light/slider.material.css.js
import { css } from 'lit';
export const styles = css `/* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */
[part=track] {
  height: 0.375rem;
}

[part=fill] {
  height: 0.375rem;
}

[part=inactive] {
  height: 0.25rem;
  top: 0.0625rem;
}

[part=steps] {
  color: hsla(var(--ig-secondary-200), var(--ig-secondary-a));
}

[part~=thumb] {
  width: 1.25rem;
  height: 1.25rem;
  top: calc((1.25rem - 0.625rem) * -1);
  -webkit-margin-start: calc((1.25rem - 0.625rem) * -1);
          margin-inline-start: calc((1.25rem - 0.625rem) * -1);
}
[part~=thumb]::after {
  border-radius: clamp(0rem, calc(var(--ig-radius-factor) * 1.25rem), 1.25rem);
  position: absolute;
  content: "";
  width: 2.5rem;
  height: 2.5rem;
  background: hsla(var(--ig-secondary-500), var(--ig-secondary-a));
  top: calc(50% - 1.25rem);
  inset-inline-start: calc(50% - 1.25rem);
  opacity: 0;
  -webkit-transform: scale(0);
      -ms-transform: scale(0);
          transform: scale(0);
  -webkit-transform-origin: center center;
      -ms-transform-origin: center center;
          transform-origin: center center;
  -webkit-transition: opacity 0.1s ease-out, -webkit-transform 0.1s ease-out;
  transition: opacity 0.1s ease-out, -webkit-transform 0.1s ease-out;
  -o-transition: transform 0.1s ease-out, opacity 0.1s ease-out;
  transition: transform 0.1s ease-out, opacity 0.1s ease-out;
  transition: transform 0.1s ease-out, opacity 0.1s ease-out, -webkit-transform 0.1s ease-out;
  overflow: hidden;
}
[part~=thumb]:hover::after {
  opacity: 0.12;
  -webkit-transform: scale(1);
      -ms-transform: scale(1);
          transform: scale(1);
}
[part~=thumb]:focus::after {
  opacity: 0.18;
  -webkit-transform: scale(1);
      -ms-transform: scale(1);
          transform: scale(1);
}
[part~=thumb]:active::after {
  opacity: 0.24;
}`;
