// igniteui-webcomponents/components/slider/themes/light/slider.fluent.css.js
import { css } from 'lit';
export const styles = css `/* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */
:host(:hover) [part=fill] {
  background: var(--fill-hover-background, hsla(var(--ig-primary-500), var(--ig-primary-a)));
}
:host(:hover) [part~=thumb] {
  border-color: var(--thumb-hover-border, hsla(var(--ig-primary-700), var(--ig-primary-a)));
}

[part~=focused]::after {
  position: absolute;
  content: "";
  left: 50%;
  top: 50%;
  -webkit-transform: translate(-50%, -50%);
      -ms-transform: translate(-50%, -50%);
          transform: translate(-50%, -50%);
  -webkit-box-shadow: 0 0 0 0.0625rem hsla(var(--ig-gray-700), var(--ig-gray-a));
          box-shadow: 0 0 0 0.0625rem hsla(var(--ig-gray-700), var(--ig-gray-a));
  width: calc(1rem + (0.125rem * 2));
  height: calc(1rem + (0.125rem * 2));
}

[part=track],
[part=fill],
[part=inactive] {
  height: 0.25rem;
}

[part=inactive] {
  background: var(--inactive-background, hsla(var(--ig-gray-600), var(--ig-gray-a)));
  top: initial;
}

[part=fill] {
  background: var(--fill-background, hsla(var(--ig-gray-800), var(--ig-gray-a)));
}

[part~=thumb] {
  background: hsla(var(--ig-surface-500), var(--ig-surface-a));
  border: 0.125rem solid var(--thumb-border, hsla(var(--ig-gray-700), var(--ig-gray-a)));
  width: 1rem;
  height: 1rem;
  top: calc((1rem - 0.5rem) * -1);
  -webkit-margin-start: calc((1rem - 0.5rem) * -1);
          margin-inline-start: calc((1rem - 0.5rem) * -1);
}

[part=thumb-label-inner] {
  background: var(--fill-hover-background, hsla(var(--ig-primary-500), var(--ig-primary-a)));
}
[part=thumb-label-inner]::after {
  border-top: 0.625rem solid var(--fill-hover-background, hsla(var(--ig-primary-500), var(--ig-primary-a)));
}

:host([disabled]) [part=inactive] {
  background: var(--inactive-disabled-background, hsla(var(--ig-gray-100), var(--ig-gray-a)));
}
:host([disabled]) [part=fill] {
  background: var(--fill-disabled-background, hsla(var(--ig-gray-500), var(--ig-gray-a)));
}
:host([disabled]) [part~=thumb] {
  background: hsla(var(--ig-surface-500), var(--ig-surface-a));
  border-color: var(--thumb-disabled-border, hsla(var(--ig-gray-400), var(--ig-gray-a)));
}`;
