// igniteui-webcomponents/components/slider/themes/light/slider.bootstrap.css.js
import { css } from 'lit';
export const styles = css `/* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */
[part=track],
[part=inactive] {
  height: 0.5rem;
}

[part=fill] {
  display: none;
}

[part=inactive] {
  background: var(--inactive-background, hsla(var(--ig-secondary-500), 0.24));
}

[part~=thumb] {
  width: 1rem;
  height: 1rem;
  background: hsla(var(--ig-primary-500), var(--ig-primary-a));
  top: calc((1rem - calc(1rem / 2)) * -1);
  -webkit-margin-start: calc((1rem - calc(1rem / 2)) * -1);
          margin-inline-start: calc((1rem - calc(1rem / 2)) * -1);
}
[part~=thumb]:focus {
  border: 0.0625rem solid hsla(var(--ig-surface-500), var(--ig-surface-a));
  -webkit-box-shadow: 0 0 0 0.1875rem hsla(var(--ig-primary-200), var(--ig-primary-a));
          box-shadow: 0 0 0 0.1875rem hsla(var(--ig-primary-200), var(--ig-primary-a));
}`;
