// igniteui-webcomponents/components/slider/themes/light/slider.indigo.css.js
import { css } from 'lit';
export const styles = css `/* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */
[part=base]:hover [part=inactive] {
  background: var(--inactive-hover-background, hsla(var(--ig-gray-500), var(--ig-gray-a)));
}
[part=base]:hover [part=fill] {
  background: hsla(var(--ig-primary-400), var(--ig-primary-a));
}

[part=inactive] {
  background: var(--inactive-background, hsla(var(--ig-gray-400), var(--ig-gray-a)));
}

[part=track],
[part=fill],
[part=inactive] {
  height: 0.125rem;
  top: initial;
  border-radius: initial;
}

[part=steps] {
  height: 0.125rem;
}
[part=steps] line {
  -webkit-transform: none;
      -ms-transform: none;
          transform: none;
}

[part~=thumb] {
  width: 0.875rem;
  height: 0.875rem;
  top: calc((0.875rem - 0.4375rem) * -1);
  -webkit-margin-start: calc((0.875rem - 0.4375rem) * -1);
          margin-inline-start: calc((0.875rem - 0.4375rem) * -1);
}
[part~=thumb]:focus {
  -webkit-box-shadow: 0 0 0 0.1875rem hsla(var(--ig-primary-200), var(--ig-primary-a));
          box-shadow: 0 0 0 0.1875rem hsla(var(--ig-primary-200), var(--ig-primary-a));
}
[part~=thumb]::after {
  display: none;
}

:host([disabled]) [part~=thumb],
:host([disabled]) [part=fill],
:host([disabled]) [part=inactive] {
  background: hsla(var(--ig-gray-300), var(--ig-gray-a));
}`;
