// igniteui-webcomponents/components/progress/themes/linear/linear.progress.fluent.css.js
import { css } from 'lit';
export const styles = css `/* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */
:host {
  --track-size: 0.125rem;
  --linear-animation-duration: 3000ms;
}

[part~=track] {
  background: hsla(var(--ig-gray-200), var(--ig-gray-a));
  margin: 0.5rem 0;
}

[part~=label] {
  font-size: 0.875rem;
  font-weight: 400;
}

[part~=indeterminate]:not([part~=secondary]) {
  width: 33% !important;
  min-width: 33%;
  -webkit-animation-name: indeterminate-bar-fluent;
          animation-name: indeterminate-bar-fluent;
  -webkit-animation-duration: var(--linear-animation-duration);
          animation-duration: var(--linear-animation-duration);
  -webkit-animation-timing-function: ease;
          animation-timing-function: ease;
  -webkit-animation-iteration-count: infinite;
          animation-iteration-count: infinite;
  left: auto;
}
[part~=indeterminate]:not([part~=secondary])::after {
  display: none;
}

[part~=secondary] {
  display: none;
}

:host([dir=rtl]) [part~=indeterminate] {
  -webkit-animation-name: indeterminate-bar-fluent-rtl;
          animation-name: indeterminate-bar-fluent-rtl;
}

[part~=primary][part~=indeterminate] {
  background: -webkit-gradient(linear, left top, right top, from(transparent), color-stop(50%, hsla(var(--ig-primary-500), var(--ig-primary-a))), to(transparent));
  background: -o-linear-gradient(left, transparent 0%, hsla(var(--ig-primary-500), var(--ig-primary-a)) 50%, transparent 100%);
  background: linear-gradient(90deg, transparent 0%, hsla(var(--ig-primary-500), var(--ig-primary-a)) 50%, transparent 100%);
}

[part~=danger][part~=indeterminate] {
  background: -webkit-gradient(linear, left top, right top, from(transparent), color-stop(50%, hsla(var(--ig-error-500), var(--ig-error-a))), to(transparent));
  background: -o-linear-gradient(left, transparent 0%, hsla(var(--ig-error-500), var(--ig-error-a)) 50%, transparent 100%);
  background: linear-gradient(90deg, transparent 0%, hsla(var(--ig-error-500), var(--ig-error-a)) 50%, transparent 100%);
}

[part~=warning][part~=indeterminate] {
  background: -webkit-gradient(linear, left top, right top, from(transparent), color-stop(50%, hsla(var(--ig-warn-500), var(--ig-warn-a))), to(transparent));
  background: -o-linear-gradient(left, transparent 0%, hsla(var(--ig-warn-500), var(--ig-warn-a)) 50%, transparent 100%);
  background: linear-gradient(90deg, transparent 0%, hsla(var(--ig-warn-500), var(--ig-warn-a)) 50%, transparent 100%);
}

[part~=info][part~=indeterminate] {
  background: -webkit-gradient(linear, left top, right top, from(transparent), color-stop(50%, hsla(var(--ig-info-500), var(--ig-info-a))), to(transparent));
  background: -o-linear-gradient(left, transparent 0%, hsla(var(--ig-info-500), var(--ig-info-a)) 50%, transparent 100%);
  background: linear-gradient(90deg, transparent 0%, hsla(var(--ig-info-500), var(--ig-info-a)) 50%, transparent 100%);
}

[part~=success][part~=indeterminate] {
  background: -webkit-gradient(linear, left top, right top, from(transparent), color-stop(50%, hsla(var(--ig-success-500), var(--ig-success-a))), to(transparent));
  background: -o-linear-gradient(left, transparent 0%, hsla(var(--ig-success-500), var(--ig-success-a)) 50%, transparent 100%);
  background: linear-gradient(90deg, transparent 0%, hsla(var(--ig-success-500), var(--ig-success-a)) 50%, transparent 100%);
}`;
