// igniteui-webcomponents/components/progress/themes/linear/linear.progress.bootstrap.css.js
import { css } from 'lit';
export const styles = css `:host {
  --is-large: clamp(0, (var(--component-size, 1) + 1) - var(--ig-size-large, 3), 1);
  --is-medium:
      min(
          clamp(0, (var(--component-size, 1) + 1) - var(--ig-size-medium, 2), 1),
          clamp(0, var(--ig-size-large, 3) - var(--component-size, 1), 1)
      );
  --is-small: clamp(0, var(--ig-size-medium) - var(--component-size, 1), 1);
  position: relative;
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
  scrollbar-width: var(--ig-scrollbar-size);
  scrollbar-color: var(--ig-scrollbar-thumb-background) var(--ig-scrollbar-track-background);
}
:host *,
:host *::before,
:host *::after {
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
}
:host ::-webkit-scrollbar {
  width: var(--ig-scrollbar-size);
  height: var(--ig-scrollbar-size);
  background: var(--ig-scrollbar-track-background);
}
:host ::-webkit-scrollbar-thumb {
  background: var(--ig-scrollbar-thumb-background);
}

[hidden] {
  display: none !important;
}

/* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */
@-webkit-keyframes indeterminate-primary {
  0% {
    -webkit-transform: translateX(0);
            transform: translateX(0);
  }
  20% {
    -webkit-animation-timing-function: cubic-bezier(0.5, 0, 0.7017, 0.4958);
            animation-timing-function: cubic-bezier(0.5, 0, 0.7017, 0.4958);
    -webkit-transform: translateX(0);
            transform: translateX(0);
  }
  59.15% {
    -webkit-animation-timing-function: cubic-bezier(0.3024, 0.3813, 0.55, 0.9563);
            animation-timing-function: cubic-bezier(0.3024, 0.3813, 0.55, 0.9563);
    -webkit-transform: translateX(83.671%);
            transform: translateX(83.671%);
  }
  100% {
    -webkit-transform: translateX(200.611%);
            transform: translateX(200.611%);
  }
}
@keyframes indeterminate-primary {
  0% {
    -webkit-transform: translateX(0);
            transform: translateX(0);
  }
  20% {
    -webkit-animation-timing-function: cubic-bezier(0.5, 0, 0.7017, 0.4958);
            animation-timing-function: cubic-bezier(0.5, 0, 0.7017, 0.4958);
    -webkit-transform: translateX(0);
            transform: translateX(0);
  }
  59.15% {
    -webkit-animation-timing-function: cubic-bezier(0.3024, 0.3813, 0.55, 0.9563);
            animation-timing-function: cubic-bezier(0.3024, 0.3813, 0.55, 0.9563);
    -webkit-transform: translateX(83.671%);
            transform: translateX(83.671%);
  }
  100% {
    -webkit-transform: translateX(200.611%);
            transform: translateX(200.611%);
  }
}
@-webkit-keyframes indeterminate-primary-scale {
  0% {
    -webkit-transform: scaleX(0.08);
            transform: scaleX(0.08);
  }
  36.65% {
    -webkit-animation-timing-function: cubic-bezier(0.3347, 0.124, 0.7858, 1);
            animation-timing-function: cubic-bezier(0.3347, 0.124, 0.7858, 1);
    -webkit-transform: scaleX(0.08);
            transform: scaleX(0.08);
  }
  69.15% {
    -webkit-animation-timing-function: cubic-bezier(0.3347, 0.124, 0.7858, 1);
            animation-timing-function: cubic-bezier(0.3347, 0.124, 0.7858, 1);
    -webkit-transform: scaleX(0.6614);
            transform: scaleX(0.6614);
  }
  100% {
    -webkit-transform: scaleX(0.08);
            transform: scaleX(0.08);
  }
}
@keyframes indeterminate-primary-scale {
  0% {
    -webkit-transform: scaleX(0.08);
            transform: scaleX(0.08);
  }
  36.65% {
    -webkit-animation-timing-function: cubic-bezier(0.3347, 0.124, 0.7858, 1);
            animation-timing-function: cubic-bezier(0.3347, 0.124, 0.7858, 1);
    -webkit-transform: scaleX(0.08);
            transform: scaleX(0.08);
  }
  69.15% {
    -webkit-animation-timing-function: cubic-bezier(0.3347, 0.124, 0.7858, 1);
            animation-timing-function: cubic-bezier(0.3347, 0.124, 0.7858, 1);
    -webkit-transform: scaleX(0.6614);
            transform: scaleX(0.6614);
  }
  100% {
    -webkit-transform: scaleX(0.08);
            transform: scaleX(0.08);
  }
}
@-webkit-keyframes indeterminate-secondary {
  0% {
    -webkit-animation-timing-function: cubic-bezier(0.15, 0, 0.515, 0.4096);
            animation-timing-function: cubic-bezier(0.15, 0, 0.515, 0.4096);
    -webkit-transform: translateX(0);
            transform: translateX(0);
  }
  25% {
    -webkit-animation-timing-function: cubic-bezier(0.31, 0.284, 0.8, 0.7337);
            animation-timing-function: cubic-bezier(0.31, 0.284, 0.8, 0.7337);
    -webkit-transform: translateX(37.6519%);
            transform: translateX(37.6519%);
  }
  48.35% {
    -webkit-animation-timing-function: cubic-bezier(0.4, 0.627, 0.6, 0.902);
            animation-timing-function: cubic-bezier(0.4, 0.627, 0.6, 0.902);
    -webkit-transform: translateX(84.3861%);
            transform: translateX(84.3861%);
  }
  100% {
    -webkit-transform: translateX(160.2777%);
            transform: translateX(160.2777%);
  }
}
@keyframes indeterminate-secondary {
  0% {
    -webkit-animation-timing-function: cubic-bezier(0.15, 0, 0.515, 0.4096);
            animation-timing-function: cubic-bezier(0.15, 0, 0.515, 0.4096);
    -webkit-transform: translateX(0);
            transform: translateX(0);
  }
  25% {
    -webkit-animation-timing-function: cubic-bezier(0.31, 0.284, 0.8, 0.7337);
            animation-timing-function: cubic-bezier(0.31, 0.284, 0.8, 0.7337);
    -webkit-transform: translateX(37.6519%);
            transform: translateX(37.6519%);
  }
  48.35% {
    -webkit-animation-timing-function: cubic-bezier(0.4, 0.627, 0.6, 0.902);
            animation-timing-function: cubic-bezier(0.4, 0.627, 0.6, 0.902);
    -webkit-transform: translateX(84.3861%);
            transform: translateX(84.3861%);
  }
  100% {
    -webkit-transform: translateX(160.2777%);
            transform: translateX(160.2777%);
  }
}
@-webkit-keyframes indeterminate-secondary-scale {
  0% {
    -webkit-animation-timing-function: cubic-bezier(0.15, 0, 0.515, 0.4096);
            animation-timing-function: cubic-bezier(0.15, 0, 0.515, 0.4096);
    -webkit-transform: scaleX(0.08);
            transform: scaleX(0.08);
  }
  19.15% {
    -webkit-animation-timing-function: cubic-bezier(0.31, 0.284, 0.8, 0.7337);
            animation-timing-function: cubic-bezier(0.31, 0.284, 0.8, 0.7337);
    -webkit-transform: scaleX(0.4571);
            transform: scaleX(0.4571);
  }
  44.15% {
    -webkit-animation-timing-function: cubic-bezier(0.4, 0.627, 0.6, 0.902);
            animation-timing-function: cubic-bezier(0.4, 0.627, 0.6, 0.902);
    -webkit-transform: scaleX(0.727);
            transform: scaleX(0.727);
  }
  100% {
    -webkit-transform: scaleX(0.08);
            transform: scaleX(0.08);
  }
}
@keyframes indeterminate-secondary-scale {
  0% {
    -webkit-animation-timing-function: cubic-bezier(0.15, 0, 0.515, 0.4096);
            animation-timing-function: cubic-bezier(0.15, 0, 0.515, 0.4096);
    -webkit-transform: scaleX(0.08);
            transform: scaleX(0.08);
  }
  19.15% {
    -webkit-animation-timing-function: cubic-bezier(0.31, 0.284, 0.8, 0.7337);
            animation-timing-function: cubic-bezier(0.31, 0.284, 0.8, 0.7337);
    -webkit-transform: scaleX(0.4571);
            transform: scaleX(0.4571);
  }
  44.15% {
    -webkit-animation-timing-function: cubic-bezier(0.4, 0.627, 0.6, 0.902);
            animation-timing-function: cubic-bezier(0.4, 0.627, 0.6, 0.902);
    -webkit-transform: scaleX(0.727);
            transform: scaleX(0.727);
  }
  100% {
    -webkit-transform: scaleX(0.08);
            transform: scaleX(0.08);
  }
}
@-webkit-keyframes indeterminate-bar-fluent {
  0% {
    -webkit-transform: translateX(-100%);
            transform: translateX(-100%);
    -webkit-transform-origin: left;
            transform-origin: left;
  }
  100% {
    -webkit-transform: translateX(310%);
            transform: translateX(310%);
    -webkit-transform-origin: right;
            transform-origin: right;
  }
}
@keyframes indeterminate-bar-fluent {
  0% {
    -webkit-transform: translateX(-100%);
            transform: translateX(-100%);
    -webkit-transform-origin: left;
            transform-origin: left;
  }
  100% {
    -webkit-transform: translateX(310%);
            transform: translateX(310%);
    -webkit-transform-origin: right;
            transform-origin: right;
  }
}
@-webkit-keyframes indeterminate-bar-fluent-rtl {
  0% {
    -webkit-transform: translateX(-310%);
            transform: translateX(-310%);
    -webkit-transform-origin: right;
            transform-origin: right;
  }
  100% {
    -webkit-transform: translateX(100%);
            transform: translateX(100%);
    -webkit-transform-origin: left;
            transform-origin: left;
  }
}
@keyframes indeterminate-bar-fluent-rtl {
  0% {
    -webkit-transform: translateX(-310%);
            transform: translateX(-310%);
    -webkit-transform-origin: right;
            transform-origin: right;
  }
  100% {
    -webkit-transform: translateX(100%);
            transform: translateX(100%);
    -webkit-transform-origin: left;
            transform-origin: left;
  }
}
@-webkit-keyframes indeterminate-accordion {
  0% {
    stroke-dashoffset: calc(calc(calc(var(--diameter) / 2 - var(--stroke-thickness) * 0.5) * 2 * 3.1416) * 2);
    stroke-dasharray: calc(calc(calc(var(--diameter) / 2 - var(--stroke-thickness) * 0.5) * 2 * 3.1416) * 9 / 10);
  }
  100% {
    stroke-dashoffset: calc(calc(calc(var(--diameter) / 2 - var(--stroke-thickness) * 0.5) * 2 * 3.1416) * 2 / 5);
  }
}
@keyframes indeterminate-accordion {
  0% {
    stroke-dashoffset: calc(calc(calc(var(--diameter) / 2 - var(--stroke-thickness) * 0.5) * 2 * 3.1416) * 2);
    stroke-dasharray: calc(calc(calc(var(--diameter) / 2 - var(--stroke-thickness) * 0.5) * 2 * 3.1416) * 9 / 10);
  }
  100% {
    stroke-dashoffset: calc(calc(calc(var(--diameter) / 2 - var(--stroke-thickness) * 0.5) * 2 * 3.1416) * 2 / 5);
  }
}
@-webkit-keyframes indeterminate-accordion-rtl {
  0% {
    stroke-dashoffset: calc(calc(calc(var(--diameter) / 2 - var(--stroke-thickness) * 0.5) * 2 * 3.1416) * -2);
    stroke-dasharray: calc(calc(calc(var(--diameter) / 2 - var(--stroke-thickness) * 0.5) * 2 * 3.1416) * 9 / 10);
  }
  100% {
    stroke-dashoffset: calc(calc(calc(var(--diameter) / 2 - var(--stroke-thickness) * 0.5) * 2 * 3.1416) * -2 / 5);
  }
}
@keyframes indeterminate-accordion-rtl {
  0% {
    stroke-dashoffset: calc(calc(calc(var(--diameter) / 2 - var(--stroke-thickness) * 0.5) * 2 * 3.1416) * -2);
    stroke-dasharray: calc(calc(calc(var(--diameter) / 2 - var(--stroke-thickness) * 0.5) * 2 * 3.1416) * 9 / 10);
  }
  100% {
    stroke-dashoffset: calc(calc(calc(var(--diameter) / 2 - var(--stroke-thickness) * 0.5) * 2 * 3.1416) * -2 / 5);
  }
}
@-webkit-keyframes rotate-center {
  0% {
    -webkit-transform: rotate(0);
            transform: rotate(0);
  }
  100% {
    -webkit-transform: rotate(360deg);
            transform: rotate(360deg);
  }
}
@keyframes rotate-center {
  0% {
    -webkit-transform: rotate(0);
            transform: rotate(0);
  }
  100% {
    -webkit-transform: rotate(360deg);
            transform: rotate(360deg);
  }
}
:host {
  --track-size: 1rem;
}

[part~=track] {
  border-radius: clamp(0rem, calc(var(--ig-radius-factor) * 0.25rem), 0.25rem);
}

[part~=striped][part~=primary]:not([part~=indeterminate]) {
  /* stylelint-disable */
  background: hsla(var(--ig-primary-500), var(--ig-primary-a)) -o-repeating-linear-gradient(45deg, hsla(var(--ig-primary-500), var(--ig-primary-a)), hsla(var(--ig-primary-500), var(--ig-primary-a)) 0.3125rem, hsla(var(--ig-gray-50), 0.2) 0.3125rem, hsla(var(--ig-gray-50), 0.2) 0.625rem);
  background: hsla(var(--ig-primary-500), var(--ig-primary-a)) repeating-linear-gradient(45deg, hsla(var(--ig-primary-500), var(--ig-primary-a)), hsla(var(--ig-primary-500), var(--ig-primary-a)) 0.3125rem, hsla(var(--ig-gray-50), 0.2) 0.3125rem, hsla(var(--ig-gray-50), 0.2) 0.625rem);
  /* stylelint-enable */
}

[part~=striped][part~=primary]::after {
  /* stylelint-disable */
  background: hsla(var(--ig-primary-500), var(--ig-primary-a)) -o-repeating-linear-gradient(45deg, hsla(var(--ig-primary-500), var(--ig-primary-a)), hsla(var(--ig-primary-500), var(--ig-primary-a)) 0.3125rem, hsla(var(--ig-gray-50), 0.2) 0.3125rem, hsla(var(--ig-gray-50), 0.2) 0.625rem);
  background: hsla(var(--ig-primary-500), var(--ig-primary-a)) repeating-linear-gradient(45deg, hsla(var(--ig-primary-500), var(--ig-primary-a)), hsla(var(--ig-primary-500), var(--ig-primary-a)) 0.3125rem, hsla(var(--ig-gray-50), 0.2) 0.3125rem, hsla(var(--ig-gray-50), 0.2) 0.625rem);
  /* stylelint-enable */
}

[part~=striped][part~=danger]:not([part~=indeterminate]) {
  /* stylelint-disable */
  background: hsla(var(--ig-error-500), var(--ig-error-a)) -o-repeating-linear-gradient(45deg, hsla(var(--ig-error-500), var(--ig-error-a)), hsla(var(--ig-error-500), var(--ig-error-a)) 0.3125rem, hsla(var(--ig-gray-50), 0.2) 0.3125rem, hsla(var(--ig-gray-50), 0.2) 0.625rem);
  background: hsla(var(--ig-error-500), var(--ig-error-a)) repeating-linear-gradient(45deg, hsla(var(--ig-error-500), var(--ig-error-a)), hsla(var(--ig-error-500), var(--ig-error-a)) 0.3125rem, hsla(var(--ig-gray-50), 0.2) 0.3125rem, hsla(var(--ig-gray-50), 0.2) 0.625rem);
  /* stylelint-enable */
}

[part~=striped][part~=danger]::after {
  /* stylelint-disable */
  background: hsla(var(--ig-error-500), var(--ig-error-a)) -o-repeating-linear-gradient(45deg, hsla(var(--ig-error-500), var(--ig-error-a)), hsla(var(--ig-error-500), var(--ig-error-a)) 0.3125rem, hsla(var(--ig-gray-50), 0.2) 0.3125rem, hsla(var(--ig-gray-50), 0.2) 0.625rem);
  background: hsla(var(--ig-error-500), var(--ig-error-a)) repeating-linear-gradient(45deg, hsla(var(--ig-error-500), var(--ig-error-a)), hsla(var(--ig-error-500), var(--ig-error-a)) 0.3125rem, hsla(var(--ig-gray-50), 0.2) 0.3125rem, hsla(var(--ig-gray-50), 0.2) 0.625rem);
  /* stylelint-enable */
}

[part~=striped][part~=warning]:not([part~=indeterminate]) {
  /* stylelint-disable */
  background: hsla(var(--ig-warn-500), var(--ig-warn-a)) -o-repeating-linear-gradient(45deg, hsla(var(--ig-warn-500), var(--ig-warn-a)), hsla(var(--ig-warn-500), var(--ig-warn-a)) 0.3125rem, hsla(var(--ig-gray-50), 0.2) 0.3125rem, hsla(var(--ig-gray-50), 0.2) 0.625rem);
  background: hsla(var(--ig-warn-500), var(--ig-warn-a)) repeating-linear-gradient(45deg, hsla(var(--ig-warn-500), var(--ig-warn-a)), hsla(var(--ig-warn-500), var(--ig-warn-a)) 0.3125rem, hsla(var(--ig-gray-50), 0.2) 0.3125rem, hsla(var(--ig-gray-50), 0.2) 0.625rem);
  /* stylelint-enable */
}

[part~=striped][part~=warning]::after {
  /* stylelint-disable */
  background: hsla(var(--ig-warn-500), var(--ig-warn-a)) -o-repeating-linear-gradient(45deg, hsla(var(--ig-warn-500), var(--ig-warn-a)), hsla(var(--ig-warn-500), var(--ig-warn-a)) 0.3125rem, hsla(var(--ig-gray-50), 0.2) 0.3125rem, hsla(var(--ig-gray-50), 0.2) 0.625rem);
  background: hsla(var(--ig-warn-500), var(--ig-warn-a)) repeating-linear-gradient(45deg, hsla(var(--ig-warn-500), var(--ig-warn-a)), hsla(var(--ig-warn-500), var(--ig-warn-a)) 0.3125rem, hsla(var(--ig-gray-50), 0.2) 0.3125rem, hsla(var(--ig-gray-50), 0.2) 0.625rem);
  /* stylelint-enable */
}

[part~=striped][part~=info]:not([part~=indeterminate]) {
  /* stylelint-disable */
  background: hsla(var(--ig-info-500), var(--ig-info-a)) -o-repeating-linear-gradient(45deg, hsla(var(--ig-info-500), var(--ig-info-a)), hsla(var(--ig-info-500), var(--ig-info-a)) 0.3125rem, hsla(var(--ig-gray-50), 0.2) 0.3125rem, hsla(var(--ig-gray-50), 0.2) 0.625rem);
  background: hsla(var(--ig-info-500), var(--ig-info-a)) repeating-linear-gradient(45deg, hsla(var(--ig-info-500), var(--ig-info-a)), hsla(var(--ig-info-500), var(--ig-info-a)) 0.3125rem, hsla(var(--ig-gray-50), 0.2) 0.3125rem, hsla(var(--ig-gray-50), 0.2) 0.625rem);
  /* stylelint-enable */
}

[part~=striped][part~=info]::after {
  /* stylelint-disable */
  background: hsla(var(--ig-info-500), var(--ig-info-a)) -o-repeating-linear-gradient(45deg, hsla(var(--ig-info-500), var(--ig-info-a)), hsla(var(--ig-info-500), var(--ig-info-a)) 0.3125rem, hsla(var(--ig-gray-50), 0.2) 0.3125rem, hsla(var(--ig-gray-50), 0.2) 0.625rem);
  background: hsla(var(--ig-info-500), var(--ig-info-a)) repeating-linear-gradient(45deg, hsla(var(--ig-info-500), var(--ig-info-a)), hsla(var(--ig-info-500), var(--ig-info-a)) 0.3125rem, hsla(var(--ig-gray-50), 0.2) 0.3125rem, hsla(var(--ig-gray-50), 0.2) 0.625rem);
  /* stylelint-enable */
}

[part~=striped][part~=success]:not([part~=indeterminate]) {
  /* stylelint-disable */
  background: hsla(var(--ig-success-500), var(--ig-success-a)) -o-repeating-linear-gradient(45deg, hsla(var(--ig-success-500), var(--ig-success-a)), hsla(var(--ig-success-500), var(--ig-success-a)) 0.3125rem, hsla(var(--ig-gray-50), 0.2) 0.3125rem, hsla(var(--ig-gray-50), 0.2) 0.625rem);
  background: hsla(var(--ig-success-500), var(--ig-success-a)) repeating-linear-gradient(45deg, hsla(var(--ig-success-500), var(--ig-success-a)), hsla(var(--ig-success-500), var(--ig-success-a)) 0.3125rem, hsla(var(--ig-gray-50), 0.2) 0.3125rem, hsla(var(--ig-gray-50), 0.2) 0.625rem);
  /* stylelint-enable */
}

[part~=striped][part~=success]::after {
  /* stylelint-disable */
  background: hsla(var(--ig-success-500), var(--ig-success-a)) -o-repeating-linear-gradient(45deg, hsla(var(--ig-success-500), var(--ig-success-a)), hsla(var(--ig-success-500), var(--ig-success-a)) 0.3125rem, hsla(var(--ig-gray-50), 0.2) 0.3125rem, hsla(var(--ig-gray-50), 0.2) 0.625rem);
  background: hsla(var(--ig-success-500), var(--ig-success-a)) repeating-linear-gradient(45deg, hsla(var(--ig-success-500), var(--ig-success-a)), hsla(var(--ig-success-500), var(--ig-success-a)) 0.3125rem, hsla(var(--ig-gray-50), 0.2) 0.3125rem, hsla(var(--ig-gray-50), 0.2) 0.625rem);
  /* stylelint-enable */
}`;
