// igniteui-webcomponents/components/progress/themes/linear/linear.progress.indigo.css.js
import { css } from 'lit';
export const styles = css `:host {
  --is-large: clamp(0, (var(--component-size, 1) + 1) - var(--ig-size-large, 3), 1);
  --is-medium:
      min(
          clamp(0, (var(--component-size, 1) + 1) - var(--ig-size-medium, 2), 1),
          clamp(0, var(--ig-size-large, 3) - var(--component-size, 1), 1)
      );
  --is-small: clamp(0, var(--ig-size-medium) - var(--component-size, 1), 1);
  position: relative;
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
  scrollbar-width: var(--ig-scrollbar-size);
  scrollbar-color: var(--ig-scrollbar-thumb-background) var(--ig-scrollbar-track-background);
}
:host *,
:host *::before,
:host *::after {
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
}
:host ::-webkit-scrollbar {
  width: var(--ig-scrollbar-size);
  height: var(--ig-scrollbar-size);
  background: var(--ig-scrollbar-track-background);
}
:host ::-webkit-scrollbar-thumb {
  background: var(--ig-scrollbar-thumb-background);
}

[hidden] {
  display: none !important;
}

/* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */
[part~=striped]:not([part~=indeterminate]) {
  /* stylelint-disable */
  background: hsla(var(--ig-secondary-500), var(--ig-secondary-a)) -o-repeating-linear-gradient(45deg, hsla(var(--ig-secondary-500), var(--ig-secondary-a)), hsla(var(--ig-secondary-500), var(--ig-secondary-a)) 0.375rem, hsla(var(--ig-gray-50), 0.7) 0.375rem, hsla(var(--ig-gray-50), 0.7) 0.875rem);
  background: hsla(var(--ig-secondary-500), var(--ig-secondary-a)) repeating-linear-gradient(45deg, hsla(var(--ig-secondary-500), var(--ig-secondary-a)), hsla(var(--ig-secondary-500), var(--ig-secondary-a)) 0.375rem, hsla(var(--ig-gray-50), 0.7) 0.375rem, hsla(var(--ig-gray-50), 0.7) 0.875rem);
  /* stylelint-enable */
}`;
