// igniteui-webcomponents/components/input/themes/light/input.bootstrap.css.js
import { css } from 'lit';
export const styles = css `/* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */
:host([size=large]) {
  --component-size: var(--ig-size, var(--ig-size-large));
}

:host([size=medium]) {
  --component-size: var(--ig-size, var(--ig-size-medium));
}

:host([size=small]) {
  --component-size: var(--ig-size, var(--ig-size-small));
}

:host {
  --size: max(var(--is-large, 1) * max(3rem, -1 * 3rem), var(--is-medium, 1) * max(2.375rem, -1 * 2.375rem), var(--is-small, 1) * max(1.9375rem, -1 * 1.9375rem));
  --font: max(var(--is-large, 1) * max(1.25rem, -1 * 1.25rem), var(--is-medium, 1) * max(1rem, -1 * 1rem), var(--is-small, 1) * max(0.875rem, -1 * 0.875rem));
  --input-font: max(var(--is-large, 1) * max(1.25rem, -1 * 1.25rem), var(--is-medium, 1) * max(1rem, -1 * 1rem), var(--is-small, 1) * max(1rem, -1 * 1rem));
}
:host::after {
  background: none;
}
:host [part~=container] {
  border-radius: clamp(0rem, calc(var(--ig-radius-factor) * 0.25rem), 0.25rem);
  border-style: none;
  grid-template-columns: auto 1fr auto;
  background: var(--input-background, #fff);
  height: var(--size);
}
:host [part~=container]::after {
  display: none;
}
:host [part~=input] {
  border-radius: clamp(0rem, calc(var(--ig-radius-factor) * 0.25rem), 0.25rem);
  height: var(--size);
  font-size: var(--input-font);
  padding-inline: max(var(--is-large, 1) * max(16px, -1 * 16px) * var(--ig-spacing-inline-large, var(--ig-spacing-inline, --ig-spacing)), var(--is-medium, 1) * max(12px, -1 * 12px) * var(--ig-spacing-inline-medium, var(--ig-spacing-inline, --ig-spacing)), var(--is-small, 1) * max(8px, -1 * 8px) * var(--ig-spacing-inline-small, var(--ig-spacing-inline, --ig-spacing)));
  padding-block: max(var(--is-large, 1) * max(8px, -1 * 8px) * var(--ig-spacing-block-large, var(--ig-spacing-block, --ig-spacing)), var(--is-medium, 1) * max(6px, -1 * 6px) * var(--ig-spacing-block-medium, var(--ig-spacing-block, --ig-spacing)), var(--is-small, 1) * max(4px, -1 * 4px) * var(--ig-spacing-block-small, var(--ig-spacing-block, --ig-spacing)));
  background: inherit;
  -webkit-box-shadow: 0 0 0 0 transparent;
          box-shadow: 0 0 0 0 transparent;
  z-index: 1;
  border: 1px solid hsla(var(--ig-gray-400), var(--ig-gray-a));
  -webkit-transition: border 0.15s ease-out, -webkit-box-shadow 0.15s ease-out;
  transition: border 0.15s ease-out, -webkit-box-shadow 0.15s ease-out;
  -o-transition: box-shadow 0.15s ease-out, border 0.15s ease-out;
  transition: box-shadow 0.15s ease-out, border 0.15s ease-out;
  transition: box-shadow 0.15s ease-out, border 0.15s ease-out, -webkit-box-shadow 0.15s ease-out;
  grid-area: 1/2;
  background-clip: padding-box;
}
:host [part*="input prefixed"] {
  border-start-start-radius: 0;
  border-end-start-radius: 0;
}
:host [part*="input suffixed"] {
  border-start-end-radius: 0;
  border-end-end-radius: 0;
}
:host [part*="input prefixed suffixed"] {
  border-radius: 0;
}
:host label {
  line-height: 1.5rem;
  margin-bottom: 0.25rem;
  color: hsla(var(--ig-gray-700), var(--ig-gray-a));
}
:host label:empty {
  display: none;
}
:host [part=prefix],
:host [part=suffix] {
  position: relative;
  border-radius: none;
  min-width: auto;
  color: hsla(var(--ig-gray-800), var(--ig-gray-a));
  background: hsla(var(--ig-gray-300), var(--ig-gray-a));
  -webkit-transition: border 0.15s ease-out, -webkit-box-shadow 0.15s ease-out;
  transition: border 0.15s ease-out, -webkit-box-shadow 0.15s ease-out;
  -o-transition: box-shadow 0.15s ease-out, border 0.15s ease-out;
  transition: box-shadow 0.15s ease-out, border 0.15s ease-out;
  transition: box-shadow 0.15s ease-out, border 0.15s ease-out, -webkit-box-shadow 0.15s ease-out;
  font-size: var(--font);
}
:host [part=prefix]:focus-within,
:host [part=suffix]:focus-within {
  -webkit-box-shadow: 0 0 0 0.25rem hsla(var(--ig-gray-300), 0.38);
          box-shadow: 0 0 0 0.25rem hsla(var(--ig-gray-300), 0.38);
}
:host [part=prefix] {
  grid-area: 1/1;
  border-style: solid;
  border-color: hsla(var(--ig-gray-400), var(--ig-gray-a));
  border-inline-start-width: 1px;
  border-inline-end-width: 0;
  border-block-start-width: 1px;
  border-block-end-width: 1px;
  border-start-start-radius: 0.25rem;
  border-end-start-radius: 0.25rem;
}
:host [name=prefix]::slotted(*),
:host [name=suffix]::slotted(*) {
  display: -webkit-inline-box;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  height: 100%;
  padding-inline: max(var(--is-large, 1) * max(16px, -1 * 16px) * var(--ig-spacing-inline-large, var(--ig-spacing-inline, --ig-spacing)), var(--is-medium, 1) * max(12px, -1 * 12px) * var(--ig-spacing-inline-medium, var(--ig-spacing-inline, --ig-spacing)), var(--is-small, 1) * max(8px, -1 * 8px) * var(--ig-spacing-inline-small, var(--ig-spacing-inline, --ig-spacing)));
}
:host [part=suffix] {
  grid-area: 1/3;
  border-style: solid;
  border-color: hsla(var(--ig-gray-400), var(--ig-gray-a));
  border-inline-start-width: 0;
  border-inline-end-width: 1px;
  border-block-start-width: 1px;
  border-block-end-width: 1px;
  border-start-end-radius: 0.25rem;
  border-end-end-radius: 0.25rem;
}
:host [part=helper-text] {
  font-family: var(--ig-body-2-font-family, var(--ig-font-family));
  font-size: var(--ig-body-2-font-size);
  font-weight: var(--ig-body-2-font-weight);
  font-style: var(--ig-body-2-font-style);
  line-height: var(--ig-body-2-line-height);
  letter-spacing: var(--ig-body-2-letter-spacing);
  text-transform: var(--ig-body-2-text-transform);
  margin-top: var(--ig-body-2-margin-top);
  margin-bottom: var(--ig-body-2-margin-bottom);
  margin-top: 0.25rem;
  color: hsla(var(--ig-gray-700), var(--ig-gray-a));
}

:host(:focus-within) [part~=input] {
  -webkit-box-shadow: 0 0 0 0.25rem hsla(var(--ig-primary-500), 0.38);
          box-shadow: 0 0 0 0.25rem hsla(var(--ig-primary-500), 0.38);
  border-color: hsla(var(--ig-primary-500), var(--ig-primary-a));
}

:host(:not([disabled])[invalid]) [part~=input],
:host(:not([disabled])[invalid]:focus-within) [part~=input] {
  -webkit-box-shadow: 0 0 0 0.25rem hsla(var(--ig-error-500), 0.38);
          box-shadow: 0 0 0 0.25rem hsla(var(--ig-error-500), 0.38);
  border-color: hsla(var(--ig-error-500), var(--ig-error-a));
}

:host([disabled]) [part=prefix],
:host([disabled]) [part=suffix],
:host([disabled]) [part~=container],
:host([readonly]) [part=prefix],
:host([readonly]) [part=suffix],
:host([readonly]) [part~=container] {
  background: hsla(var(--ig-gray-100), var(--ig-gray-a));
  color: hsla(var(--ig-gray-400), var(--ig-gray-a));
  border-block-color: hsla(var(--ig-gray-300), var(--ig-gray-a));
}
:host([disabled]) [part=prefix],
:host([readonly]) [part=prefix] {
  border-inline-start-color: hsla(var(--ig-gray-300), var(--ig-gray-a));
}
:host([disabled]) [part=suffix],
:host([readonly]) [part=suffix] {
  border-inline-end-color: hsla(var(--ig-gray-300), var(--ig-gray-a));
}
:host([disabled]) [part~=input],
:host([readonly]) [part~=input] {
  border-color: hsla(var(--ig-gray-300), var(--ig-gray-a));
}

:host([disabled]) [part~=input]::-webkit-input-placeholder {
  color: hsla(var(--ig-gray-400), var(--ig-gray-a));
}

:host([disabled]) [part~=input]::-moz-placeholder {
  color: hsla(var(--ig-gray-400), var(--ig-gray-a));
}

:host([disabled]) [part~=input]:-ms-input-placeholder {
  color: hsla(var(--ig-gray-400), var(--ig-gray-a));
}

:host([disabled]) [part~=input]::-ms-input-placeholder {
  color: hsla(var(--ig-gray-400), var(--ig-gray-a));
}

:host([disabled]) [part=label],
:host([disabled]) [part=helper-text],
:host([disabled]) [part~=input]::placeholder {
  color: hsla(var(--ig-gray-400), var(--ig-gray-a));
}

:host([readonly]) [part=prefix],
:host([readonly]) [part=suffix],
:host([readonly]) [part~=container] {
  color: hsla(var(--ig-gray-800), var(--ig-gray-a));
}

:host(:focus-within[readonly]) [part~=input] {
  -webkit-box-shadow: 0 0 0 0.25rem hsla(var(--ig-gray-400), 0.38);
          box-shadow: 0 0 0 0.25rem hsla(var(--ig-gray-400), 0.38);
  border-color: hsla(var(--ig-gray-400), var(--ig-gray-a));
}`;
