// igniteui-webcomponents/components/input/themes/light/input.material.css.js
import { css } from 'lit';
export const styles = css `/* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */
input:-moz-placeholder-shown + [part=notch] [part=label] {
  --label-position: max(var(--is-large, 1) * max(26px, -1 * 26px), var(--is-medium, 1) * max(22px, -1 * 22px), var(--is-small, 1) * max(18px, -1 * 18px));
  transform: translateY(calc(var(--label-position) * -1));
  font-size: 0.75rem;
}
input:-ms-input-placeholder + [part=notch] [part=label] {
  --label-position: max(var(--is-large, 1) * max(26px, -1 * 26px), var(--is-medium, 1) * max(22px, -1 * 22px), var(--is-small, 1) * max(18px, -1 * 18px));
  -ms-transform: translateY(calc(var(--label-position) * -1));
      transform: translateY(calc(var(--label-position) * -1));
  font-size: 0.75rem;
}
:host(:focus-within) [part=label], input:placeholder-shown + [part=notch] [part=label],
[part~=filled] + [part=notch] [part=label] {
  --label-position: max(var(--is-large, 1) * max(26px, -1 * 26px), var(--is-medium, 1) * max(22px, -1 * 22px), var(--is-small, 1) * max(18px, -1 * 18px));
  -webkit-transform: translateY(calc(var(--label-position) * -1));
      -ms-transform: translateY(calc(var(--label-position) * -1));
          transform: translateY(calc(var(--label-position) * -1));
  font-size: 0.75rem;
}

:host input:-moz-placeholder-shown + [part=notch] {
  border-top: 0.0625rem solid transparent;
}

:host input:-ms-input-placeholder + [part=notch] {
  border-top: 0.0625rem solid transparent;
}

:host([disabled]) [part~=filled] + [part=notch], :host([invalid]) [part~=filled] + [part=notch],
:host([invalid]:focus-within) [part~=filled] + [part=notch], :host input:placeholder-shown + [part=notch],
:host [part~=filled] + [part=notch] {
  border-top: 0.0625rem solid transparent;
}

[part=end] > [part=suffix] ::slotted(*), [part=start] > [part=prefix] ::slotted(*) {
  display: -webkit-inline-box;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  height: 100%;
  padding-inline: max(var(--is-large, 1) * max(1rem, -1 * 1rem) * var(--ig-spacing-inline-large, var(--ig-spacing-inline, --ig-spacing)), var(--is-medium, 1) * max(0.875rem, -1 * 0.875rem) * var(--ig-spacing-inline-medium, var(--ig-spacing-inline, --ig-spacing)), var(--is-small, 1) * max(0.75rem, -1 * 0.75rem) * var(--ig-spacing-inline-small, var(--ig-spacing-inline, --ig-spacing)));
}

:host input::-webkit-input-placeholder {
  color: hsla(var(--ig-gray-700), var(--ig-gray-a));
}

:host input::-moz-placeholder {
  color: hsla(var(--ig-gray-700), var(--ig-gray-a));
}

:host input:-ms-input-placeholder {
  color: hsla(var(--ig-gray-700), var(--ig-gray-a));
}

:host input::-ms-input-placeholder {
  color: hsla(var(--ig-gray-700), var(--ig-gray-a));
}

:host input::placeholder {
  color: hsla(var(--ig-gray-700), var(--ig-gray-a));
}

[part~=input] {
  background: transparent;
  padding: 0 0.25rem;
  font-size: 1rem;
  grid-area: 1/2/span 1/span 2;
}

[part=label] {
  color: hsla(var(--ig-gray-700), var(--ig-gray-a));
  -webkit-transition: color 150ms cubic-bezier(0.4, 0, 0.2, 1), font-size 150ms cubic-bezier(0.4, 0, 0.2, 1), -webkit-transform 150ms cubic-bezier(0.4, 0, 0.2, 1);
  transition: color 150ms cubic-bezier(0.4, 0, 0.2, 1), font-size 150ms cubic-bezier(0.4, 0, 0.2, 1), -webkit-transform 150ms cubic-bezier(0.4, 0, 0.2, 1);
  -o-transition: transform 150ms cubic-bezier(0.4, 0, 0.2, 1), color 150ms cubic-bezier(0.4, 0, 0.2, 1), font-size 150ms cubic-bezier(0.4, 0, 0.2, 1);
  transition: transform 150ms cubic-bezier(0.4, 0, 0.2, 1), color 150ms cubic-bezier(0.4, 0, 0.2, 1), font-size 150ms cubic-bezier(0.4, 0, 0.2, 1);
  transition: transform 150ms cubic-bezier(0.4, 0, 0.2, 1), color 150ms cubic-bezier(0.4, 0, 0.2, 1), font-size 150ms cubic-bezier(0.4, 0, 0.2, 1), -webkit-transform 150ms cubic-bezier(0.4, 0, 0.2, 1);
  line-height: 1;
  -o-text-overflow: ellipsis;
     text-overflow: ellipsis;
  overflow: hidden;
  will-change: font-size, color, transform;
}

:host(:focus-within) [part=start] {
  border-color: hsla(var(--ig-primary-500), var(--ig-primary-a));
  border-inline-start-width: 0.125rem;
  border-block-start-width: 0.125rem;
  border-block-end-width: 0.125rem;
}
:host(:focus-within) [part=notch] {
  border-width: 0.125rem;
  border-color: hsla(var(--ig-primary-500), var(--ig-primary-a));
  border-top: 0.0625rem solid transparent;
}
:host(:focus-within) [part=filler] {
  border-width: 0.125rem;
  border-color: hsla(var(--ig-primary-500), var(--ig-primary-a));
}
:host(:focus-within) [part=end] {
  border-color: hsla(var(--ig-primary-500), var(--ig-primary-a));
  border-inline-end-width: 0.125rem;
  border-block-start-width: 0.125rem;
  border-block-end-width: 0.125rem;
}
:host(:focus-within) [part=label] {
  color: hsla(var(--ig-primary-500), var(--ig-primary-a));
}
:host(:focus-within) [part~=prefix],
:host(:focus-within) [part~=suffix] {
  color: hsla(var(--ig-gray-900), var(--ig-gray-a));
}

[part~=filled] [part~=prefix],
[part~=filled] [part~=suffix] {
  color: hsla(var(--ig-gray-900), var(--ig-gray-a));
}

[part^=container] {
  grid-template-columns: auto auto 1fr auto;
}

[part=start] {
  -webkit-box-pack: start;
      -ms-flex-pack: start;
          justify-content: flex-start;
  grid-area: 1/1;
  border-style: solid;
  border-color: hsla(var(--ig-gray-500), var(--ig-gray-a));
  border-inline-start-width: 0.0625rem;
  border-inline-end-width: 0;
  border-block-start-width: 0.0625rem;
  border-block-end-width: 0.0625rem;
  border-start-start-radius: 0.25rem;
  border-end-start-radius: 0.25rem;
}
[part=notch] {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  width: auto;
  min-width: 0;
  height: 100%;
  grid-area: 1/2;
  padding: 0 0.25rem;
  border-width: 0.0625rem;
  border-style: solid;
  border-color: hsla(var(--ig-gray-500), var(--ig-gray-a));
  border-left: none;
  border-right: none;
  overflow: visible;
}
[part=notch]:empty {
  display: none;
}

[part=filler] {
  grid-area: 1/3;
  border-width: 0.0625rem;
  border-style: solid;
  border-color: hsla(var(--ig-gray-500), var(--ig-gray-a));
  border-left: none;
  border-right: none;
}

[part=end] {
  -webkit-box-pack: end;
      -ms-flex-pack: end;
          justify-content: flex-end;
  grid-area: 1/4;
  border-style: solid;
  border-color: hsla(var(--ig-gray-500), var(--ig-gray-a));
  border-inline-start-width: 0;
  border-inline-end-width: 0.0625rem;
  border-block-start-width: 0.0625rem;
  border-block-end-width: 0.0625rem;
  border-start-end-radius: 0.25rem;
  border-end-end-radius: 0.25rem;
}
[part=start],
[part=end] {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  min-width: max(var(--is-large, 1) * max(0.875rem, -1 * 0.875rem), var(--is-medium, 1) * max(0.75rem, -1 * 0.75rem), var(--is-small, 1) * max(0.625rem, -1 * 0.625rem));
  height: var(--size);
  overflow: hidden;
}

[part=helper-text] {
  font-family: var(--ig-caption-font-family, var(--ig-font-family));
  font-size: var(--ig-caption-font-size);
  font-weight: var(--ig-caption-font-weight);
  font-style: var(--ig-caption-font-style);
  line-height: var(--ig-caption-line-height);
  letter-spacing: var(--ig-caption-letter-spacing);
  text-transform: var(--ig-caption-text-transform);
  margin-top: var(--ig-caption-margin-top);
  margin-bottom: var(--ig-caption-margin-bottom);
  padding-inline: max(var(--is-large, 1) * max(1.125rem, -1 * 1.125rem) * var(--ig-spacing-inline-large, var(--ig-spacing-inline, --ig-spacing)), var(--is-medium, 1) * max(1rem, -1 * 1rem) * var(--ig-spacing-inline-medium, var(--ig-spacing-inline, --ig-spacing)), var(--is-small, 1) * max(0.875rem, -1 * 0.875rem) * var(--ig-spacing-inline-small, var(--ig-spacing-inline, --ig-spacing)));
  margin-top: 0.3125rem;
  color: hsla(var(--ig-gray-700), var(--ig-gray-a));
}

:host([invalid]) [part=start],
:host([invalid]) [part=notch],
:host([invalid]) [part=filler],
:host([invalid]) [part=end],
:host([invalid]:focus-within) [part=start],
:host([invalid]:focus-within) [part=notch],
:host([invalid]:focus-within) [part=filler],
:host([invalid]:focus-within) [part=end] {
  border-color: hsla(var(--ig-error-500), var(--ig-error-a));
}
:host([invalid]) [part=label],
:host([invalid]:focus-within) [part=label] {
  color: hsla(var(--ig-error-500), var(--ig-error-a));
}
:host([invalid]:focus-within) [part=notch] {
  border-top: 0.0625rem solid transparent;
}

:host([disabled]) {
  pointer-events: none;
}
:host([disabled]) [part~=input],
:host([disabled]) [part=label],
:host([disabled]) [part=prefix],
:host([disabled]) [part=suffix],
:host([disabled]) [part=helper-text] {
  color: hsla(var(--ig-gray-500), var(--ig-gray-a));
}
:host([disabled]) [part=start],
:host([disabled]) [part=filler],
:host([disabled]) [part=notch],
:host([disabled]) [part=end] {
  color: hsla(var(--ig-gray-500), var(--ig-gray-a));
  border-color: hsla(var(--ig-gray-500), var(--ig-gray-a));
}
:host([disabled]) input::-webkit-input-placeholder {
  color: hsla(var(--ig-gray-500), var(--ig-gray-a));
}
:host([disabled]) input::-moz-placeholder {
  color: hsla(var(--ig-gray-500), var(--ig-gray-a));
}
:host([disabled]) input:-ms-input-placeholder {
  color: hsla(var(--ig-gray-500), var(--ig-gray-a));
}
:host([disabled]) input::-ms-input-placeholder {
  color: hsla(var(--ig-gray-500), var(--ig-gray-a));
}
:host([disabled]) input::placeholder {
  color: hsla(var(--ig-gray-500), var(--ig-gray-a));
}
:host([outlined]:active) [part=suffix],
:host([outlined]:focus) [part=suffix],
:host([outlined]:focus-within) [part=suffix] {
  -webkit-margin-end: -0.0625rem;
          margin-inline-end: -0.0625rem;
}
:host([outlined]:active) [part=prefix],
:host([outlined]:focus) [part=prefix],
:host([outlined]:focus-within) [part=prefix] {
  -webkit-margin-start: -0.0625rem;
          margin-inline-start: -0.0625rem;
}
:host([outlined]:active) [part=suffix] ::slotted(*),
:host([outlined]:active) [part=prefix] ::slotted(*),
:host([outlined]:focus) [part=suffix] ::slotted(*),
:host([outlined]:focus) [part=prefix] ::slotted(*),
:host([outlined]:focus-within) [part=suffix] ::slotted(*),
:host([outlined]:focus-within) [part=prefix] ::slotted(*) {
  padding-block: max(var(--is-large, 1) * max(14px, -1 * 14px) * var(--ig-spacing-block-large, var(--ig-spacing-block, --ig-spacing)), var(--is-medium, 1) * max(12px, -1 * 12px) * var(--ig-spacing-block-medium, var(--ig-spacing-block, --ig-spacing)), var(--is-small, 1) * max(10px, -1 * 10px) * var(--ig-spacing-block-small, var(--ig-spacing-block, --ig-spacing)));
}

:host(:not([outlined])) [part=start],
:host(:not([outlined])) [part=end] {
  border-color: transparent;
  border-width: 0.0625rem;
}
:host(:not([outlined])) [part=start] {
  border-end-start-radius: 0;
}
:host(:not([outlined])) [part=end] {
  border-end-end-radius: 0;
}
:host(:not([outlined])) [part=notch],
:host(:not([outlined])) [part=filler] {
  border-top: 1px solid transparent;
  border-bottom: transparent;
  border-bottom-width: 0.0625rem;
}
:host(:not([outlined])) [part=label] {
  inset-inline-start: 0.125rem;
}
:host(:not([outlined])) [part~=container] {
  background: hsla(var(--ig-gray-200), var(--ig-gray-a));
  border-bottom: 1px solid hsla(var(--ig-gray-500), var(--ig-gray-a));
  border-start-end-radius: 0.25rem;
  border-start-start-radius: 0.25rem;
  -webkit-transition: border-bottom-width 150ms cubic-bezier(0.4, 0, 0.2, 1);
  -o-transition: border-bottom-width 150ms cubic-bezier(0.4, 0, 0.2, 1);
  transition: border-bottom-width 150ms cubic-bezier(0.4, 0, 0.2, 1);
}
:host(:not([outlined])) [part~=container]::after {
  position: absolute;
  content: "";
  width: 100%;
  height: 0.125rem;
  left: 0;
  right: 0;
  bottom: -1px;
  background: hsla(var(--ig-primary-500), var(--ig-primary-a));
  -webkit-transform: scaleX(0);
      -ms-transform: scaleX(0);
          transform: scaleX(0);
  -webkit-transition: opacity 180ms cubic-bezier(0.4, 0, 0.2, 1), -webkit-transform 180ms cubic-bezier(0.4, 0, 0.2, 1);
  transition: opacity 180ms cubic-bezier(0.4, 0, 0.2, 1), -webkit-transform 180ms cubic-bezier(0.4, 0, 0.2, 1);
  -o-transition: transform 180ms cubic-bezier(0.4, 0, 0.2, 1), opacity 180ms cubic-bezier(0.4, 0, 0.2, 1);
  transition: transform 180ms cubic-bezier(0.4, 0, 0.2, 1), opacity 180ms cubic-bezier(0.4, 0, 0.2, 1);
  transition: transform 180ms cubic-bezier(0.4, 0, 0.2, 1), opacity 180ms cubic-bezier(0.4, 0, 0.2, 1), -webkit-transform 180ms cubic-bezier(0.4, 0, 0.2, 1);
}
:host(:not([outlined])) [part~=labelled] [part~=input] {
  padding-top: 1.25rem;
  padding-bottom: 0.375rem;
}
:host(:not([outlined])) input:-moz-placeholder-shown + [part=notch] [part=label] {
  transform: translateY(-106%);
}
:host(:not([outlined])) input:-ms-input-placeholder + [part=notch] [part=label] {
  -ms-transform: translateY(-106%);
      transform: translateY(-106%);
}
:host(:not([outlined])) input:placeholder-shown + [part=notch] [part=label],
:host(:not([outlined])) [part~=filled] + [part=notch] [part=label] {
  -webkit-transform: translateY(-106%);
      -ms-transform: translateY(-106%);
          transform: translateY(-106%);
}

:host(:not([outlined]):hover) [part~=container] {
  background: hsla(var(--ig-gray-200), var(--ig-gray-a));
  border-bottom: 1px solid hsla(var(--ig-gray-800), var(--ig-gray-a));
}

:host(:not([outlined]):focus-within) [part~=container] {
  background: var(--focus-background, hsla(var(--ig-gray-300), var(--ig-gray-a)));
  border-bottom: 1px solid hsla(var(--ig-primary-500), var(--ig-primary-a));
}
:host(:not([outlined]):focus-within) [part~=container]::after {
  -webkit-transform: scaleX(1);
      -ms-transform: scaleX(1);
          transform: scaleX(1);
  opacity: 1;
}
:host(:not([outlined]):focus-within) [part=notch] [part=label] {
  -webkit-transform: translateY(-106%);
      -ms-transform: translateY(-106%);
          transform: translateY(-106%);
}

:host(:not([outlined])[invalid]) [part=start],
:host(:not([outlined])[invalid]) [part=notch],
:host(:not([outlined])[invalid]) [part=filler],
:host(:not([outlined])[invalid]) [part=end],
:host(:not([outlined])[invalid]:focus-within) [part=start],
:host(:not([outlined])[invalid]:focus-within) [part=notch],
:host(:not([outlined])[invalid]:focus-within) [part=filler],
:host(:not([outlined])[invalid]:focus-within) [part=end] {
  border-color: transparent;
}
:host(:not([outlined])[invalid]) [part~=container],
:host(:not([outlined])[invalid]:focus-within) [part~=container] {
  border-color: hsla(var(--ig-error-500), var(--ig-error-a));
}
:host(:not([outlined])[invalid]) [part~=container]::after,
:host(:not([outlined])[invalid]:focus-within) [part~=container]::after {
  background: hsla(var(--ig-error-500), var(--ig-error-a));
}

:host(:not([outlined])[disabled]) [part~=input],
:host(:not([outlined])[disabled]) [part=label],
:host(:not([outlined])[disabled]) [part=prefix],
:host(:not([outlined])[disabled]) [part=suffix],
:host(:not([outlined])[disabled]) [part=helper-text] {
  color: hsla(var(--ig-gray-500), var(--ig-gray-a));
}
:host(:not([outlined])[disabled]) [part~=container] {
  background: hsla(var(--ig-gray-100), var(--ig-gray-a));
  border-color: hsla(var(--ig-gray-500), var(--ig-gray-a));
  border-bottom-style: dashed;
}`;
