// igniteui-webcomponents/components/input/themes/light/input.indigo.css.js
import { css } from 'lit';
export const styles = css `/* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */
:host [part=suffix] ::slotted(*), :host [part=prefix] ::slotted(*) {
  display: -webkit-inline-box;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  height: calc(100% - 0.125rem);
  padding-inline: max(var(--is-large, 1) * max(16px, -1 * 16px) * var(--ig-spacing-inline-large, var(--ig-spacing-inline, --ig-spacing)), var(--is-medium, 1) * max(12px, -1 * 12px) * var(--ig-spacing-inline-medium, var(--ig-spacing-inline, --ig-spacing)), var(--is-small, 1) * max(8px, -1 * 8px) * var(--ig-spacing-inline-small, var(--ig-spacing-inline, --ig-spacing)));
}

:host [part=prefix],
:host [part=suffix] {
  color: hsla(var(--ig-gray-600), var(--ig-gray-a));
}
:host [part=prefix] {
  grid-area: 1/1;
}
:host [part=suffix] {
  grid-area: 1/3;
}
:host [part~=label] {
  color: hsla(var(--ig-gray-900), var(--ig-gray-a));
  font-size: 0.875rem;
  font-weight: 600;
  line-height: 1;
}
:host [part^=container] {
  background: transparent;
  border-bottom: 1px solid hsla(var(--ig-gray-500), var(--ig-gray-a));
  -webkit-transition: background 0.25s ease-out;
  -o-transition: background 0.25s ease-out;
  transition: background 0.25s ease-out;
  grid-template-columns: auto 1fr auto;
}
:host [part^=container]::after {
  position: absolute;
  content: "";
  width: 100%;
  height: 0.125rem;
  left: 0;
  right: 0;
  bottom: -1px;
  background: hsla(var(--ig-secondary-500), var(--ig-secondary-a));
  -webkit-transform: scaleY(0);
      -ms-transform: scaleY(0);
          transform: scaleY(0);
  -webkit-transition: opacity 180ms cubic-bezier(0.4, 0, 0.2, 1), -webkit-transform 180ms cubic-bezier(0.4, 0, 0.2, 1);
  transition: opacity 180ms cubic-bezier(0.4, 0, 0.2, 1), -webkit-transform 180ms cubic-bezier(0.4, 0, 0.2, 1);
  -o-transition: transform 180ms cubic-bezier(0.4, 0, 0.2, 1), opacity 180ms cubic-bezier(0.4, 0, 0.2, 1);
  transition: transform 180ms cubic-bezier(0.4, 0, 0.2, 1), opacity 180ms cubic-bezier(0.4, 0, 0.2, 1);
  transition: transform 180ms cubic-bezier(0.4, 0, 0.2, 1), opacity 180ms cubic-bezier(0.4, 0, 0.2, 1), -webkit-transform 180ms cubic-bezier(0.4, 0, 0.2, 1);
  -webkit-transform-origin: bottom center;
      -ms-transform-origin: bottom center;
          transform-origin: bottom center;
}
:host [part~=input] {
  font-size: 1rem;
  background: initial;
  padding-block: 0.5rem;
  padding-inline: max(var(--is-large, 1) * max(1rem, -1 * 1rem) * var(--ig-spacing-inline-large, var(--ig-spacing-inline, --ig-spacing)), var(--is-medium, 1) * max(0.75rem, -1 * 0.75rem) * var(--ig-spacing-inline-medium, var(--ig-spacing-inline, --ig-spacing)), var(--is-small, 1) * max(0.5rem, -1 * 0.5rem) * var(--ig-spacing-inline-small, var(--ig-spacing-inline, --ig-spacing)));
  border: none;
  grid-area: 1/2;
}
:host [part*="input prefixed"] {
  -webkit-padding-start: 0;
          padding-inline-start: 0;
}
:host [part*="input suffixed"] {
  -webkit-padding-end: 0;
          padding-inline-end: 0;
}
:host [part=helper-text] {
  margin-top: 0.375rem;
  padding-inline: max(var(--is-large, 1) * max(1rem, -1 * 1rem) * var(--ig-spacing-inline-large, var(--ig-spacing-inline, --ig-spacing)), var(--is-medium, 1) * max(0.75rem, -1 * 0.75rem) * var(--ig-spacing-inline-medium, var(--ig-spacing-inline, --ig-spacing)), var(--is-small, 1) * max(0.5rem, -1 * 0.5rem) * var(--ig-spacing-inline-small, var(--ig-spacing-inline, --ig-spacing)));
  color: hsla(var(--ig-gray-600), var(--ig-gray-a));
}

:host(:hover) [part^=container],
:host([outlined]:hover) [part^=container] {
  background: hsla(var(--ig-gray-100), var(--ig-gray-a));
}

:host(:focus-within) [part^=container],
:host([outlined]:focus-within) [part^=container] {
  background: hsla(var(--ig-gray-100), var(--ig-gray-a));
}
:host(:focus-within) [part^=container]::after,
:host([outlined]:focus-within) [part^=container]::after {
  -webkit-transform: scaleY(1);
      -ms-transform: scaleY(1);
          transform: scaleY(1);
}
:host(:focus-within) [part=label],
:host([outlined]:focus-within) [part=label] {
  color: var(--label-focus-color, hsla(var(--ig-primary-500), var(--ig-primary-a)));
}

:host([invalid]) [part^=container] {
  border-color: hsla(var(--ig-error-500), var(--ig-error-a));
}
:host([invalid]) [part^=container]::after {
  background: hsla(var(--ig-error-500), var(--ig-error-a));
}
:host([invalid]) [part=label] {
  color: hsla(var(--ig-error-500), var(--ig-error-a));
}

:host([disabled]) {
  pointer-events: none;
}
:host([disabled]) [part^=container] {
  border-color: hsla(var(--ig-gray-300), var(--ig-gray-a));
  color: hsla(var(--ig-gray-300), var(--ig-gray-a));
}
:host([disabled]) [part~=input]::-webkit-input-placeholder {
  color: hsla(var(--ig-gray-300), var(--ig-gray-a));
}
:host([disabled]) [part~=input]::-moz-placeholder {
  color: hsla(var(--ig-gray-300), var(--ig-gray-a));
}
:host([disabled]) [part~=input]:-ms-input-placeholder {
  color: hsla(var(--ig-gray-300), var(--ig-gray-a));
}
:host([disabled]) [part~=input]::-ms-input-placeholder {
  color: hsla(var(--ig-gray-300), var(--ig-gray-a));
}
:host([disabled]) [part=prefix],
:host([disabled]) [part=suffix],
:host([disabled]) [part=label],
:host([disabled]) [part=helper-text],
:host([disabled]) [part~=input]::placeholder {
  color: hsla(var(--ig-gray-300), var(--ig-gray-a));
}`;
