// igniteui-webcomponents/components/chip/themes/chip.base.css.js
import { css } from 'lit';
export const styles = css `:host {
  --is-large: clamp(0, (var(--component-size, 1) + 1) - var(--ig-size-large, 3), 1);
  --is-medium:
      min(
          clamp(0, (var(--component-size, 1) + 1) - var(--ig-size-medium, 2), 1),
          clamp(0, var(--ig-size-large, 3) - var(--component-size, 1), 1)
      );
  --is-small: clamp(0, var(--ig-size-medium) - var(--component-size, 1), 1);
  position: relative;
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
  scrollbar-width: var(--ig-scrollbar-size);
  scrollbar-color: var(--ig-scrollbar-thumb-background) var(--ig-scrollbar-track-background);
}
:host *,
:host *::before,
:host *::after {
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
}
:host ::-webkit-scrollbar {
  width: var(--ig-scrollbar-size);
  height: var(--ig-scrollbar-size);
  background: var(--ig-scrollbar-track-background);
}
:host ::-webkit-scrollbar-thumb {
  background: var(--ig-scrollbar-thumb-background);
}

[hidden] {
  display: none !important;
}

/* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */
:host {
  --size: max(var(--is-large, 1) * max(2rem, -1 * 2rem), var(--is-medium, 1) * max(1.5rem, -1 * 1.5rem), var(--is-small, 1) * max(1.25rem, -1 * 1.25rem));
  display: -webkit-inline-box;
  display: -ms-inline-flexbox;
  display: inline-flex;
  vertical-align: top;
}

:host([size=small]) {
  --component-size: var(--ig-size, var(--ig-size-small));
}
:host([size=small]) [part=prefix] {
  -webkit-margin-end: 0.1875rem;
          margin-inline-end: 0.1875rem;
}
:host([size=small]) [part=suffix] {
  -webkit-margin-start: 0.1875rem;
          margin-inline-start: 0.1875rem;
}

:host([size=medium]) {
  --component-size: var(--ig-size, var(--ig-size-medium));
}
:host([size=medium]) [part=prefix] {
  -webkit-margin-end: 0.375rem;
          margin-inline-end: 0.375rem;
}
:host([size=medium]) [part=suffix] {
  -webkit-margin-start: 0.375rem;
          margin-inline-start: 0.375rem;
}

:host([size=large]) {
  --component-size: var(--ig-size, var(--ig-size-large));
}
:host([size=large]) [part=prefix] {
  -webkit-margin-end: 0.5rem;
          margin-inline-end: 0.5rem;
}
:host([size=large]) [part=suffix] {
  -webkit-margin-start: 0.5rem;
          margin-inline-start: 0.5rem;
}
:host([size=large]) igc-icon[size=large] {
  --size: 1.25rem;
}

:host button {
  font-family: var(--ig-body-2-font-family, var(--ig-font-family));
  font-size: var(--ig-body-2-font-size);
  font-weight: var(--ig-body-2-font-weight);
  font-style: var(--ig-body-2-font-style);
  line-height: var(--ig-body-2-line-height);
  letter-spacing: var(--ig-body-2-letter-spacing);
  text-transform: var(--ig-body-2-text-transform);
  margin-top: var(--ig-body-2-margin-top);
  margin-bottom: var(--ig-body-2-margin-bottom);
  border-radius: clamp(0rem, calc(var(--ig-radius-factor) * 1rem), 1rem);
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  font-family: var(--ig-font-family);
  border-style: none;
  -webkit-box-shadow: none;
          box-shadow: none;
  cursor: pointer;
  height: var(--size);
  padding: max(var(--is-large, 1) * max(12px, -1 * 12px) * var(--ig-spacing-large, var(--ig-spacing, --ig-spacing)), var(--is-medium, 1) * max(6px, -1 * 6px) * var(--ig-spacing-medium, var(--ig-spacing, --ig-spacing)), var(--is-small, 1) * max(2px, -1 * 2px) * var(--ig-spacing-small, var(--ig-spacing, --ig-spacing)));
}
:host button[disabled] {
  pointer-events: none;
  color: hsla(var(--ig-gray-500), var(--ig-gray-a));
}

:host([variant=primary]) button,
:host([selected][variant=primary]) button:not([disabled]) {
  background: hsla(var(--ig-primary-500), var(--ig-primary-a));
  color: var(--ig-primary-500-contrast);
}
:host([variant=primary]) button:hover,
:host([selected][variant=primary]) button:not([disabled]):hover {
  background: hsla(var(--ig-primary-600), var(--ig-primary-a));
  color: var(--ig-primary-600-contrast);
}
:host([variant=primary]) button:focus,
:host([selected][variant=primary]) button:not([disabled]):focus {
  background: hsla(var(--ig-primary-800), var(--ig-primary-a));
  color: var(--ig-primary-800-contrast);
}

:host([variant=info]) button,
:host([selected][variant=info]) button:not([disabled]) {
  background: hsla(var(--ig-info-500), var(--ig-info-a));
  color: var(--ig-info-500-contrast);
}
:host([variant=info]) button:hover,
:host([selected][variant=info]) button:not([disabled]):hover {
  background: hsla(var(--ig-info-600), var(--ig-info-a));
  color: var(--ig-info-600-contrast);
}
:host([variant=info]) button:focus,
:host([selected][variant=info]) button:not([disabled]):focus {
  background: hsla(var(--ig-info-800), var(--ig-info-a));
  color: var(--ig-info-800-contrast);
}

:host([variant=success]) button,
:host([selected][variant=success]) button:not([disabled]) {
  background: hsla(var(--ig-success-500), var(--ig-success-a));
  color: var(--ig-success-500-contrast);
}
:host([variant=success]) button:hover,
:host([selected][variant=success]) button:not([disabled]):hover {
  background: hsla(var(--ig-success-600), var(--ig-success-a));
  color: var(--ig-success-600-contrast);
}
:host([variant=success]) button:focus,
:host([selected][variant=success]) button:not([disabled]):focus {
  background: hsla(var(--ig-success-800), var(--ig-success-a));
  color: var(--ig-success-800-contrast);
}

:host([variant=warning]) button,
:host([selected][variant=warning]) button:not([disabled]) {
  background: hsla(var(--ig-warn-500), var(--ig-warn-a));
  color: var(--ig-warn-500-contrast);
}
:host([variant=warning]) button:hover,
:host([selected][variant=warning]) button:not([disabled]):hover {
  background: hsla(var(--ig-warn-600), var(--ig-warn-a));
  color: var(--ig-warn-600-contrast);
}
:host([variant=warning]) button:focus,
:host([selected][variant=warning]) button:not([disabled]):focus {
  background: hsla(var(--ig-warn-800), var(--ig-warn-a));
  color: var(--ig-warn-800-contrast);
}

:host([variant=danger]) button,
:host([selected][variant=danger]) button:not([disabled]) {
  background: hsla(var(--ig-error-500), var(--ig-error-a));
  color: var(--ig-error-500-contrast);
}
:host([variant=danger]) button:hover,
:host([selected][variant=danger]) button:not([disabled]):hover {
  background: hsla(var(--ig-error-600), var(--ig-error-a));
  color: var(--ig-error-600-contrast);
}
:host([variant=danger]) button:focus,
:host([selected][variant=danger]) button:not([disabled]):focus {
  background: hsla(var(--ig-error-800), var(--ig-error-a));
  color: var(--ig-error-800-contrast);
}

[part=prefix],
[part=suffix] {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
}

::slotted(*) {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
}

:host([variant=primary]) button[disabled],
:host([variant=info]) button[disabled],
:host([variant=success]) button[disabled],
:host([variant=warning]) button[disabled],
:host([variant=danger]) button[disabled] {
  color: hsla(var(--ig-gray-500), var(--ig-gray-a));
}`;
