// igniteui-webcomponents/components/dialog/themes/light/dialog.base.css.js
import { css } from 'lit';
export const styles = css `:host {
  --is-large: clamp(0, (var(--component-size, 1) + 1) - var(--ig-size-large, 3), 1);
  --is-medium:
      min(
          clamp(0, (var(--component-size, 1) + 1) - var(--ig-size-medium, 2), 1),
          clamp(0, var(--ig-size-large, 3) - var(--component-size, 1), 1)
      );
  --is-small: clamp(0, var(--ig-size-medium) - var(--component-size, 1), 1);
  position: relative;
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
  scrollbar-width: var(--ig-scrollbar-size);
  scrollbar-color: var(--ig-scrollbar-thumb-background) var(--ig-scrollbar-track-background);
}
:host *,
:host *::before,
:host *::after {
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
}
:host ::-webkit-scrollbar {
  width: var(--ig-scrollbar-size);
  height: var(--ig-scrollbar-size);
  background: var(--ig-scrollbar-track-background);
}
:host ::-webkit-scrollbar-thumb {
  background: var(--ig-scrollbar-thumb-background);
}

[hidden] {
  display: none !important;
}

/* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */
:host {
  border-radius: clamp(0rem, calc(var(--ig-radius-factor) * 2px), 2px);
  background: hsla(var(--ig-surface-500), var(--ig-surface-a));
  color: var(--ig-surface-500-contrast);
  border-style: none;
  padding: unset;
}

[part=base] {
  background: inherit;
  color: inherit;
  border-style: inherit;
  padding: inherit;
  border: inherit;
  border-radius: inherit;
  min-width: inherit;
  -webkit-box-shadow: var(--ig-elevation-24);
          box-shadow: var(--ig-elevation-24);
}

[part=content] {
  color: var(--content-color, hsla(var(--ig-gray-700), var(--ig-gray-a)));
  font-family: var(--ig-body-1-font-family, var(--ig-font-family));
  font-size: var(--ig-body-1-font-size);
  font-weight: var(--ig-body-1-font-weight);
  font-style: var(--ig-body-1-font-style);
  line-height: var(--ig-body-1-line-height);
  letter-spacing: var(--ig-body-1-letter-spacing);
  text-transform: var(--ig-body-1-text-transform);
  margin-top: var(--ig-body-1-margin-top);
  margin-bottom: var(--ig-body-1-margin-bottom);
  margin: 0;
}

[part=title],
[part=footer] {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  color: hsla(var(--ig-gray-900), var(--ig-gray-a));
  font-family: var(--ig-h6-font-family, var(--ig-font-family));
  font-size: var(--ig-h6-font-size);
  font-weight: var(--ig-h6-font-weight);
  font-style: var(--ig-h6-font-style);
  line-height: var(--ig-h6-line-height);
  letter-spacing: var(--ig-h6-letter-spacing);
  text-transform: var(--ig-h6-text-transform);
  margin-top: var(--ig-h6-margin-top);
  margin-bottom: var(--ig-h6-margin-bottom);
  margin: 0;
}

[part=footer] {
  -ms-flex-wrap: nowrap;
      flex-wrap: nowrap;
  -webkit-box-pack: end;
      -ms-flex-pack: end;
          justify-content: flex-end;
}

::slotted(*) {
  font: inherit;
  letter-spacing: inherit;
  line-height: inherit;
  text-transform: inherit;
  margin: 0;
}

slot[name=title]::slotted(:not(:last-child)),
slot[name=footer]::slotted(:not(:last-child)) {
  -webkit-margin-end: 0.5rem;
          margin-inline-end: 0.5rem;
}

dialog::-webkit-backdrop {
  background: none;
}

dialog::backdrop {
  background: none;
}

[part=backdrop] {
  position: fixed;
  inset: 0;
  background: hsla(var(--ig-gray-500), 0.54);
}`;
