// igniteui-webcomponents/components/stepper/themes/step/light/step.base.css.js
import { css } from 'lit';
export const styles = css `:host {
  --is-large: clamp(0, (var(--component-size, 1) + 1) - var(--ig-size-large, 3), 1);
  --is-medium:
      min(
          clamp(0, (var(--component-size, 1) + 1) - var(--ig-size-medium, 2), 1),
          clamp(0, var(--ig-size-large, 3) - var(--component-size, 1), 1)
      );
  --is-small: clamp(0, var(--ig-size-medium) - var(--component-size, 1), 1);
  position: relative;
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
  scrollbar-width: var(--ig-scrollbar-size);
  scrollbar-color: var(--ig-scrollbar-thumb-background) var(--ig-scrollbar-track-background);
}
:host *,
:host *::before,
:host *::after {
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
}
:host ::-webkit-scrollbar {
  width: var(--ig-scrollbar-size);
  height: var(--ig-scrollbar-size);
  background: var(--ig-scrollbar-track-background);
}
:host ::-webkit-scrollbar-thumb {
  background: var(--ig-scrollbar-thumb-background);
}

[hidden] {
  display: none !important;
}

/* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */
:host {
  --title-color: var(--ig-gray-50-contrast);
  --subtitle-color: hsla(var(--ig-gray-700), var(--ig-gray-a));
  --separator-color: hsla(var(--ig-gray-300), var(--ig-gray-a));
  --separator-color--completed: hsla(var(--ig-gray-900), var(--ig-gray-a));
  --header-color: var(--ig-gray-50-contrast);
  --header-background--hover: hsla(var(--ig-gray-200), var(--ig-gray-a));
  --header-background--focus: hsla(var(--ig-gray-100), var(--ig-gray-a));
  --header-shadow: hsla(var(--ig-gray-50), 0);
  --indicator-color: hsla(var(--ig-gray-900), var(--ig-gray-a));
  --indicator-background: hsla(var(--ig-gray-400), var(--ig-gray-a));
  --indicator-shadow: var(--indicator-background);
  --optional-indicator-color: var(--indicator-color);
  --optional-indicator-background: var(--indicator-background);
  --optional-indicator-shadow: var(--indicator-shadow);
  --completed-indicator-color: var(--indicator-color);
  --completed-indicator-background: var(--indicator-background);
  --completed-indicator-shadow: var(--indicator-shadow);
  --active-indicator-color: var(--ig-primary-800-contrast);
  --active-indicator-background: hsla(var(--ig-primary-500), var(--ig-primary-a));
  --active-indicator-shadow: var(--active-indicator-background);
  --active-title-color: var(--title-color);
  --active-subtitle-color: var(--subtitle-color);
  --error-color: var(--ig-error-800-contrast);
  --error-background: hsla(var(--ig-error-500), var(--ig-error-a));
  --error-shadow: var(--error-background);
  --disabled-color: hsla(var(--ig-gray-500), var(--ig-gray-a));
  --disabled-background: hsla(var(--ig-gray-100), var(--ig-gray-a));
  --disabled-shadow: var(--disabled-background);
  -webkit-box-align: start;
      -ms-flex-align: start;
          align-items: flex-start;
}

[part~=header-container],
[part~=header] {
  width: var(--header-width-vertical, auto);
}
[part~=header-container]::before,
[part~=header]::before {
  content: "";
  width: auto;
  -webkit-box-flex: 1;
      -ms-flex: 1;
          flex: 1;
  -webkit-border-before: var(--separator-size) var(--separator-type) var(--separator-color);
          border-block-start: var(--separator-size) var(--separator-type) var(--separator-color);
  height: var(--separator-size);
  position: relative;
}

[part~=header-container] {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  pointer-events: none;
  position: relative;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
  z-index: 0;
}
[part~=header-container] [part~=header] {
  pointer-events: all;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
}
[part~=header-container] [part=indicator],
[part~=header-container] [part=title],
[part~=header-container] [part=subtitle] {
  pointer-events: none;
}
[part~=header-container]::before {
  min-width: var(--separator-min-width);
  display: var(--hide-horizontal-separator, var(--horizontal-separator-display--first-of-type));
}

[part~=header] {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-align: var(--align-items, flex-start);
      -ms-flex-align: var(--align-items, flex-start);
          align-items: var(--align-items, flex-start);
  grid-row: var(--header-bottom, 1);
  position: relative;
  overflow: hidden;
  padding: var(--header-padding);
  -webkit-box-pack: start;
      -ms-flex-pack: start;
          justify-content: flex-start;
  gap: var(--header-gap);
  color: var(--header-color);
  z-index: var(--vertical-header-z-index);
}
[part~=header]::before, [part~=header]::after {
  content: "";
  -webkit-box-flex: 1;
      -ms-flex: 1;
          flex: 1;
  -webkit-border-before: var(--separator-size) var(--separator-type) var(--separator-color);
          border-block-start: var(--separator-size) var(--separator-type) var(--separator-color);
  height: var(--separator-size);
  min-width: var(--separator-min-width--header);
  width: calc(50% - var(--indicator-size) / 2 - var(--header-padding));
  position: absolute;
  z-index: -1;
}
[part~=header]::before {
  inset-inline-start: 0;
  /* stylelint-disable */
  display: var(--hide-horizontal-separator, var(--separator-disply-not-full, var(--horizontal-separator-display--first-of-type)));
  /* stylelint-enable */
}
[part~=header]::after {
  inset-inline-end: 0;
  /* stylelint-disable */
  display: var(--hide-horizontal-separator, var(--separator-disply-not-full, var(--horizontal-separator-display--last-of-type)));
  /* stylelint-enable */
}
[part~=header]:focus, [part~=header]:hover {
  cursor: pointer;
  outline: none;
}
[part~=header]:hover {
  background: var(--header-background--hover);
}
[part~=header]:hover::before, [part~=header]:hover::after {
  visibility: var(--header-separator-visibility--hover);
}
[part~=header]:focus, [part~=header]:focus-within, [part~=header]:active {
  background: var(--header-background--focus);
  -webkit-box-shadow: inset 0 0 0 0.0625rem var(--header-shadow);
          box-shadow: inset 0 0 0 0.0625rem var(--header-shadow);
}
[part~=header]:focus::before, [part~=header]:focus::after, [part~=header]:focus-within::before, [part~=header]:focus-within::after, [part~=header]:active::before, [part~=header]:active::after {
  visibility: var(--header-separator-visibility--hover);
}

[part~=body] {
  display: var(--vertical-body-disply--last-of-type, var(--vertical-body-disply, none));
  grid-row-start: var(--body-top, 2);
  grid-column: span var(--steps-count);
  -webkit-margin-start: calc(var(--body-indent--vertical) / 2);
          margin-inline-start: calc(var(--body-indent--vertical) / 2);
  -webkit-padding-start: var(--horizontal-body-padding, calc(var(--body-indent--vertical) / 2));
          padding-inline-start: var(--horizontal-body-padding, calc(var(--body-indent--vertical) / 2));
  -webkit-padding-end: var(--horizontal-body-padding, var(--vertical-body-padding));
          padding-inline-end: var(--horizontal-body-padding, var(--vertical-body-padding));
  -webkit-padding-before: var(--horizontal-body-padding, var(--vertical-body-padding));
          padding-block-start: var(--horizontal-body-padding, var(--vertical-body-padding));
  -webkit-padding-after: var(--horizontal-body-padding, var(--vertical-body-padding));
          padding-block-end: var(--horizontal-body-padding, var(--vertical-body-padding));
  z-index: 1;
  position: relative;
  color: hsla(var(--ig-gray-900), var(--ig-gray-a));
  width: 0;
  min-width: 100%;
}
[part~=body]::before {
  display: var(--hide-last-separator, var(--hide-first-separator, var(--vertical-separator-visibility)));
  content: "";
  position: absolute;
  inset-inline-start: calc(var(--separator-size) / 2 * -1);
  height: 100%;
  inset-block-start: 0 !important;
  -webkit-border-start: var(--separator-size) var(--separator-type) var(--separator-color);
          border-inline-start: var(--separator-size) var(--separator-type) var(--separator-color);
}

[part=content] {
  display: var(--horizontal-content-disply, none);
}

[part~=top] {
  -webkit-box-align: var(--step-not-full-header-aligmnet, flex-end);
      -ms-flex-align: var(--step-not-full-header-aligmnet, flex-end);
          align-items: var(--step-not-full-header-aligmnet, flex-end);
}
[part~=top]::before, [part~=top]::after {
  inset-block-end: var(--step-separator-position);
}
[part~=top] [part~=header] {
  -webkit-box-align: var(--align-items, flex-start);
      -ms-flex-align: var(--align-items, flex-start);
          align-items: var(--align-items, flex-start);
  -webkit-box-orient: vertical;
  -webkit-box-direction: reverse;
      -ms-flex-direction: column-reverse;
          flex-direction: column-reverse;
}
[part~=top] [part~=header]::before, [part~=top] [part~=header]::after {
  inset-block-end: var(--step-separator-position);
}

[part~=bottom] {
  -webkit-box-align: var(--step-not-full-header-aligmnet, start);
      -ms-flex-align: var(--step-not-full-header-aligmnet, start);
          align-items: var(--step-not-full-header-aligmnet, start);
}
[part~=bottom]::before, [part~=bottom]::after {
  inset-block-start: var(--step-separator-position);
}
[part~=bottom] [part~=header] {
  -webkit-box-align: var(--align-items, flex-start);
      -ms-flex-align: var(--align-items, flex-start);
          align-items: var(--align-items, flex-start);
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
}
[part~=bottom] [part~=header]::before, [part~=bottom] [part~=header]::after {
  inset-block-start: var(--step-separator-position);
}

[part~=top] [part~=text],
[part~=bottom] [part~=text] {
  width: 100%;
}

[part~=start] [part=text] {
  -webkit-box-ordinal-group: 0;
      -ms-flex-order: -1;
          order: -1;
}

[part~=start],
[part~=end] {
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
}
[part~=start] [part~=text],
[part~=end] [part~=text] {
  overflow: hidden;
}
[part~=start] [part~=header]::before, [part~=start] [part~=header]::after,
[part~=end] [part~=header]::before,
[part~=end] [part~=header]::after {
  display: none;
}

[part~=indicator] {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
  width: var(--indicator-size);
  min-width: var(--indicator-size);
  height: var(--indicator-size);
  position: relative;
}
[part~=indicator] span {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
  width: var(--indicator-size);
  height: var(--indicator-size);
  border-radius: var(--indicator-roundness);
  background: var(--indicator-background);
  color: var(--indicator-color);
  -webkit-box-shadow: inset 0 0 0 var(--indicator-box-shadow-size) var(--indicator-shadow);
          box-shadow: inset 0 0 0 var(--indicator-box-shadow-size) var(--indicator-shadow);
  font-family: var(--ig-caption-font-family, var(--ig-font-family));
  font-size: var(--ig-caption-font-size);
  font-weight: var(--ig-caption-font-weight);
  font-style: var(--ig-caption-font-style);
  line-height: var(--ig-caption-line-height);
  letter-spacing: var(--ig-caption-letter-spacing);
  text-transform: var(--ig-caption-text-transform);
  margin-top: var(--ig-caption-margin-top);
  margin-bottom: var(--ig-caption-margin-bottom);
}

[part~=text] {
  text-align: var(--align-text, initial);
  min-width: var(--indicator-size);
}

[part~=subtitle],
[part~=title] {
  overflow: hidden;
  -o-text-overflow: ellipsis;
     text-overflow: ellipsis;
  white-space: nowrap;
}

[part~=title] {
  font-family: var(--ig-body-2-font-family, var(--ig-font-family));
  font-size: var(--ig-body-2-font-size);
  font-weight: var(--ig-body-2-font-weight);
  font-style: var(--ig-body-2-font-style);
  line-height: var(--ig-body-2-line-height);
  letter-spacing: var(--ig-body-2-letter-spacing);
  text-transform: var(--ig-body-2-text-transform);
  margin-top: var(--ig-body-2-margin-top);
  margin-bottom: var(--ig-body-2-margin-bottom);
  color: var(--title-color);
}

[part~=subtitle] {
  font-family: var(--ig-caption-font-family, var(--ig-font-family));
  font-size: var(--ig-caption-font-size);
  font-weight: var(--ig-caption-font-weight);
  font-style: var(--ig-caption-font-style);
  line-height: var(--ig-caption-line-height);
  letter-spacing: var(--ig-caption-letter-spacing);
  text-transform: var(--ig-caption-text-transform);
  margin-top: var(--ig-caption-margin-top);
  margin-bottom: var(--ig-caption-margin-bottom);
  color: var(--subtitle-color);
}

[part~=empty] {
  display: none;
}

[part~=optional] [part~=indicator] span {
  color: var(--optional-indicator-color);
  background: var(--optional-indicator-background);
  -webkit-box-shadow: inset 0 0 0 var(--indicator-box-shadow-size) var(--optional-indicator-shadow);
          box-shadow: inset 0 0 0 var(--indicator-box-shadow-size) var(--optional-indicator-shadow);
}

[part~=complete-start]::after {
  border-block-start-style: var(--separator-type--completed);
  border-block-start-color: var(--separator-color--completed);
}
[part~=complete-start] [part~=header]::after {
  border-block-start-style: var(--separator-type--completed);
  border-block-start-color: var(--separator-color--completed);
}
[part~=complete-start] [part~=indicator] span {
  background: var(--completed-indicator-background);
  color: var(--completed-indicator-color);
  -webkit-box-shadow: inset 0 0 0 var(--indicator-box-shadow-size) var(--completed-indicator-shadow);
          box-shadow: inset 0 0 0 var(--indicator-box-shadow-size) var(--completed-indicator-shadow);
}

:host([complete]) [part~=body]::before {
  border-inline-start-style: var(--separator-type--completed);
  border-inline-start-color: var(--separator-color--completed);
}

[part~=complete-end]::before {
  border-block-start-style: var(--separator-type--completed);
  border-block-start-color: var(--separator-color--completed);
}
[part~=complete-end] [part~=header]::before {
  border-block-start-style: var(--separator-type--completed);
  border-block-start-color: var(--separator-color--completed);
}

[part~=invalid] [part~=indicator] span {
  color: var(--error-color);
  background: var(--error-background);
  -webkit-box-shadow: inset 0 0 0 var(--indicator-box-shadow-size) var(--error-shadow);
          box-shadow: inset 0 0 0 var(--indicator-box-shadow-size) var(--error-shadow);
}
[part~=invalid] [part~=title],
[part~=invalid] [part~=subtitle] {
  color: var(--error-background);
}

:host([active]) [part~=body] {
  display: var(--vertical-body-disply, block);
}
:host([active]) [part~="body bottom"]::before {
  inset-block-start: 0;
}
:host([active]) [part=content] {
  display: var(--horizontal-content-disply, block);
  pointer-events: all;
}
:host([active]) [part~=indicator] span {
  background: var(--active-indicator-background);
  color: var(--active-indicator-color);
  -webkit-box-shadow: inset 0 0 0 var(--indicator-box-shadow-size) var(--active-indicator-shadow);
          box-shadow: inset 0 0 0 var(--indicator-box-shadow-size) var(--active-indicator-shadow);
}
:host([active]) [part~=title] {
  color: var(--active-title-color);
  font-weight: 600;
}
:host([active]) [part~=subtitle] {
  color: var(--active-subtitle-color);
}

[part~=disabled] [part=header] {
  background: hsla(var(--ig-gray-50), 0);
}
[part~=disabled]:focus [part=header], [part~=disabled]:focus-visible [part=header], [part~=disabled]:focus-within [part=header], [part~=disabled]:active [part=header], [part~=disabled]:hover [part=header] {
  background: hsla(var(--ig-gray-50), 0);
  cursor: default;
  -webkit-box-shadow: inset 0 0 0 0.0625rem hsla(var(--ig-gray-50), 0);
          box-shadow: inset 0 0 0 0.0625rem hsla(var(--ig-gray-50), 0);
}
[part~=disabled] [part~=indicator] span {
  color: var(--disabled-color);
  background: var(--disabled-background);
  -webkit-box-shadow: inset 0 0 0 var(--indicator-box-shadow-size) var(--disabled-shadow);
          box-shadow: inset 0 0 0 var(--indicator-box-shadow-size) var(--disabled-shadow);
}
[part~=disabled] [part~=title],
[part~=disabled] [part~=subtitle] {
  color: var(--disabled-color);
}`;
