// igniteui-webcomponents/components/checkbox/checkbox-base.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
import { LitElement } from 'lit';
import { property, query, queryAssignedNodes, state } from 'lit/decorators.js';
import { alternateName } from '../common/decorators/alternateName.js';
import { blazorDeepImport } from '../common/decorators/blazorDeepImport.js';
import { blazorTwoWayBind } from '../common/decorators/blazorTwoWayBind.js';
import { EventEmitterMixin } from '../common/mixins/event-emitter.js';
let IgcCheckboxBaseComponent = class IgcCheckboxBaseComponent extends EventEmitterMixin(LitElement) {
    constructor() {
        super(...arguments);
        this.focused = false;
        this.hideLabel = false;
        this.disabled = false;
        this.checked = false;
        this.required = false;
        this.invalid = false;
        this.labelPosition = 'after';
    }
    click() {
        this.input.click();
    }
    focus(options) {
        this.input.focus(options);
    }
    blur() {
        this.input.blur();
    }
    reportValidity() {
        return this.input.reportValidity();
    }
    checkValidity() {
        return this.input.checkValidity();
    }
    setCustomValidity(message) {
        this.input.setCustomValidity(message);
        this.invalid = !this.input.checkValidity();
    }
    handleBlur() {
        this.emitEvent('igcBlur');
        this.focused = false;
    }
    handleFocus() {
        this.emitEvent('igcFocus');
    }
    handleMouseDown(event) {
        event.preventDefault();
        this.input.focus();
        this.focused = false;
    }
    connectedCallback() {
        super.connectedCallback();
        this.addEventListener('keyup', this.handleKeyUp);
    }
    disconnectedCallback() {
        this.removeEventListener('keyup', this.handleKeyUp);
        super.disconnectedCallback();
    }
    handleKeyUp() {
        if (!this.focused) {
            this.focused = true;
        }
    }
    async firstUpdated() {
        if (this.label.length === 0) {
            this.hideLabel = true;
        }
        await this.updateComplete;
    }
};
__decorate([
    query('input[type="checkbox"]', true)
], IgcCheckboxBaseComponent.prototype, "input", void 0);
__decorate([
    queryAssignedNodes({ flatten: true })
], IgcCheckboxBaseComponent.prototype, "label", void 0);
__decorate([
    state()
], IgcCheckboxBaseComponent.prototype, "focused", void 0);
__decorate([
    state()
], IgcCheckboxBaseComponent.prototype, "hideLabel", void 0);
__decorate([
    property()
], IgcCheckboxBaseComponent.prototype, "name", void 0);
__decorate([
    property()
], IgcCheckboxBaseComponent.prototype, "value", void 0);
__decorate([
    property({ type: Boolean, reflect: true })
], IgcCheckboxBaseComponent.prototype, "disabled", void 0);
__decorate([
    property({ type: Boolean }),
    blazorTwoWayBind('igcChange', 'detail')
], IgcCheckboxBaseComponent.prototype, "checked", void 0);
__decorate([
    property({ type: Boolean, reflect: true })
], IgcCheckboxBaseComponent.prototype, "required", void 0);
__decorate([
    property({ type: Boolean, reflect: true })
], IgcCheckboxBaseComponent.prototype, "invalid", void 0);
__decorate([
    property({ reflect: true, attribute: 'label-position' })
], IgcCheckboxBaseComponent.prototype, "labelPosition", void 0);
__decorate([
    property({ reflect: true, attribute: 'aria-labelledby' })
], IgcCheckboxBaseComponent.prototype, "ariaLabelledby", void 0);
__decorate([
    alternateName('focusComponent')
], IgcCheckboxBaseComponent.prototype, "focus", null);
__decorate([
    alternateName('blurComponent')
], IgcCheckboxBaseComponent.prototype, "blur", null);
IgcCheckboxBaseComponent = __decorate([
    blazorDeepImport
], IgcCheckboxBaseComponent);
export { IgcCheckboxBaseComponent };
