// igniteui-webcomponents/components/tree/themes/light/tree-item.base.css.js
import { css } from 'lit';
export const styles = css `:host {
  --is-large: clamp(0, (var(--component-size, 1) + 1) - var(--ig-size-large, 3), 1);
  --is-medium:
      min(
          clamp(0, (var(--component-size, 1) + 1) - var(--ig-size-medium, 2), 1),
          clamp(0, var(--ig-size-large, 3) - var(--component-size, 1), 1)
      );
  --is-small: clamp(0, var(--ig-size-medium) - var(--component-size, 1), 1);
  position: relative;
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
  scrollbar-width: var(--ig-scrollbar-size);
  scrollbar-color: var(--ig-scrollbar-thumb-background) var(--ig-scrollbar-track-background);
}
:host *,
:host *::before,
:host *::after {
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
}
:host ::-webkit-scrollbar {
  width: var(--ig-scrollbar-size);
  height: var(--ig-scrollbar-size);
  background: var(--ig-scrollbar-track-background);
}
:host ::-webkit-scrollbar-thumb {
  background: var(--ig-scrollbar-thumb-background);
}

[hidden] {
  display: none !important;
}

/* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */
:host {
  --size: max(var(--is-large, 1) * max(3.125rem, -1 * 3.125rem), var(--is-medium, 1) * max(2.5rem, -1 * 2.5rem), var(--is-small, 1) * max(2rem, -1 * 2rem));
  --igc-tree-indentation-size: 24px;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
  font-family: inherit;
}

:host(:focus-within) {
  outline: none;
}

:host([disabled]) {
  color: hsla(var(--ig-gray-500), var(--ig-gray-a));
  background: hsla(var(--ig-surface-500), var(--ig-surface-a));
}

[part~=wrapper] {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  position: relative;
  background: hsla(var(--ig-surface-500), var(--ig-surface-a));
  color: var(--ig-surface-500-contrast);
  height: var(--size);
  padding: 0 max(var(--is-large, 1) * max(24px, -1 * 24px) * var(--ig-spacing-large, var(--ig-spacing, --ig-spacing)), var(--is-medium, 1) * max(16px, -1 * 16px) * var(--ig-spacing-medium, var(--ig-spacing, --ig-spacing)), var(--is-small, 1) * max(12px, -1 * 12px) * var(--ig-spacing-small, var(--ig-spacing, --ig-spacing)));
}
[part~=wrapper]::after {
  content: "";
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: 0;
  background: transparent;
}
[part~=wrapper]:hover::after {
  background: var(--hover-color, hsla(var(--ig-gray-900), 0.1));
}

[part~=active] {
  background: hsla(var(--ig-gray-200), var(--ig-gray-a));
  color: var(--ig-gray-200-contrast);
}

[part~=selected] {
  background: hsla(var(--ig-secondary-200), var(--ig-secondary-a));
  color: var(--ig-secondary-200-contrast);
}

[part~=selected][part~=focused][part~=active],
[part~=selected][part~=focused],
[part~=selected][part~=active] {
  background: hsla(var(--ig-secondary-300), var(--ig-secondary-a));
  color: var(--ig-secondary-300-contrast);
}

[part~=focused] {
  -webkit-box-shadow: inset 0 0 0 1px hsla(var(--ig-secondary-500), var(--ig-secondary-a));
          box-shadow: inset 0 0 0 1px hsla(var(--ig-secondary-500), var(--ig-secondary-a));
}

[part~=indicator] {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  z-index: 1;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
  cursor: pointer;
  -webkit-user-select: none;
     -moz-user-select: none;
      -ms-user-select: none;
          user-select: none;
  -webkit-margin-end: 0.25rem;
          margin-inline-end: 0.25rem;
  min-width: 1.5rem;
}
[part~=indicator] igc-icon {
  width: 1.5rem;
  height: 1.5rem;
  font-size: 1.5rem;
}

[part~=label] {
  display: block;
  font-family: var(--ig-body-2-font-family, var(--ig-font-family));
  font-size: var(--ig-body-2-font-size);
  font-weight: var(--ig-body-2-font-weight);
  font-style: var(--ig-body-2-font-style);
  line-height: var(--ig-body-2-line-height);
  letter-spacing: var(--ig-body-2-letter-spacing);
  text-transform: var(--ig-body-2-text-transform);
  margin-top: var(--ig-body-2-margin-top);
  margin-bottom: var(--ig-body-2-margin-bottom);
  margin: 0;
}

[part~=label],
[part~=text] {
  z-index: 1;
  white-space: nowrap;
  -o-text-overflow: ellipsis;
     text-overflow: ellipsis;
  overflow: hidden;
}

[part~=select] {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  z-index: 1;
  -webkit-margin-end: 0.25rem;
          margin-inline-end: 0.25rem;
}
[part~=select] igc-checkbox::part(label) {
  display: none;
}

[part~=large] igc-circular-progress {
  --diameter: 1.5rem;
}

[part~=medium] igc-circular-progress {
  --diameter: 1.25rem;
}

[part~=small] igc-circular-progress {
  --diameter: 1.125rem;
}`;
