// igniteui-webcomponents/components/button/themes/button/light/button.bootstrap.css.js
import { css } from 'lit';
export const styles = css `/* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */
:host {
  --size: max(var(--is-large, 1) * max(3rem, -1 * 3rem), var(--is-medium, 1) * max(2.375rem, -1 * 2.375rem), var(--is-small, 1) * max(1.9375rem, -1 * 1.9375rem));
}

:host [part=base] {
  -webkit-transition: all 0.15s ease-out;
  -o-transition: all 0.15s ease-out;
  transition: all 0.15s ease-out;
}

:host(:not([disabled])[variant=flat]) [part=base] {
  border-radius: clamp(0rem, calc(var(--ig-radius-factor) * 0.25rem), 0.25rem);
  color: hsla(var(--ig-primary-500), var(--ig-primary-a));
}
:host(:not([disabled])[variant=flat]) [part=base]:hover {
  color: hsla(var(--ig-primary-800), var(--ig-primary-a));
  background: transparent;
}
:host(:not([disabled])[variant=flat]) [part=base]:focus, :host(:not([disabled])[variant=flat]) [part=base]:active {
  color: hsla(var(--ig-primary-600), var(--ig-primary-a));
  -webkit-box-shadow: 0 0 0 0.25rem hsla(var(--ig-primary-200), 0.5);
          box-shadow: 0 0 0 0.25rem hsla(var(--ig-primary-200), 0.5);
  background: transparent;
}

:host([disabled][variant=flat]) [part=base] {
  background: transparent;
  color: hsla(var(--ig-primary-100), var(--ig-primary-a));
}

:host([disabled][variant=fab]) [part=base],
:host([disabled][variant=contained]) [part=base] {
  background: hsla(var(--ig-primary-200), var(--ig-primary-a));
  color: var(--ig-primary-600-contrast);
}

:host(:not([disabled])[variant=contained]) [part=base] {
  color: var(--ig-primary-600-contrast);
  -webkit-box-shadow: var(--ig-elevation-0);
          box-shadow: var(--ig-elevation-0);
  background: hsla(var(--ig-primary-500), var(--ig-primary-a));
  opacity: 1;
}
:host(:not([disabled])[variant=contained]) [part=base]:hover {
  color: var(--ig-primary-600-contrast);
}
:host(:not([disabled])[variant=contained]) [part=base]:focus, :host(:not([disabled])[variant=contained]) [part=base]:active {
  color: var(--ig-primary-600-contrast);
  -webkit-box-shadow: 0 0 0 0.25rem hsla(var(--ig-primary-200), 0.5);
          box-shadow: 0 0 0 0.25rem hsla(var(--ig-primary-200), 0.5);
}
:host(:not([disabled])[variant=contained]) [part=base]:hover, :host(:not([disabled])[variant=contained]) [part=base]:focus, :host(:not([disabled])[variant=contained]) [part=base]:active {
  background: hsla(var(--ig-primary-600), var(--ig-primary-a));
}

:host(:not([disabled])[variant=outlined]) [part=base] {
  color: hsla(var(--ig-primary-500), var(--ig-primary-a));
  border: 0.0625rem solid hsla(var(--ig-primary-500), var(--ig-primary-a));
  -webkit-box-shadow: none;
          box-shadow: none;
  overflow: visible;
}
:host(:not([disabled])[variant=outlined]) [part=base]:focus {
  background: transparent;
  -webkit-box-shadow: 0 0 0 0.25rem hsla(var(--ig-primary-200), 0.5);
          box-shadow: 0 0 0 0.25rem hsla(var(--ig-primary-200), 0.5);
}
:host(:not([disabled])[variant=outlined]) [part=base]:hover, :host(:not([disabled])[variant=outlined]) [part=base]:active {
  background: hsla(var(--ig-primary-500), var(--ig-primary-a));
  color: var(--ig-primary-600-contrast);
}

:host([disabled][variant=outlined]) [part=base] {
  -webkit-box-shadow: none;
          box-shadow: none;
  color: hsla(var(--ig-primary-200), var(--ig-primary-a));
  border: 0.0625rem solid hsla(var(--ig-primary-200), var(--ig-primary-a));
  background: transparent;
}

:host([variant=fab][size=large]) [part=base] {
  border-radius: clamp(0rem, calc(var(--ig-radius-factor) * 1rem), 1rem);
}

:host([variant=fab][size=medium]) [part=base] {
  border-radius: clamp(0rem, calc(var(--ig-radius-factor) * 0.75rem), 0.75rem);
}

:host([variant=fab][size=small]) [part=base] {
  border-radius: clamp(0rem, calc(var(--ig-radius-factor) * 0.5rem), 0.5rem);
}

:host(:not([disabled])[variant=fab]) [part=base] {
  color: var(--ig-primary-600-contrast);
  -webkit-box-shadow: var(--ig-elevation-0);
          box-shadow: var(--ig-elevation-0);
  background: hsla(var(--ig-primary-500), var(--ig-primary-a));
}
:host(:not([disabled])[variant=fab]) [part=base]:hover {
  color: var(--ig-primary-600-contrast);
}
:host(:not([disabled])[variant=fab]) [part=base]:focus, :host(:not([disabled])[variant=fab]) [part=base]:active {
  color: var(--ig-primary-600-contrast);
  -webkit-box-shadow: 0 0 0 0.25rem hsla(var(--ig-primary-200), 0.5);
          box-shadow: 0 0 0 0.25rem hsla(var(--ig-primary-200), 0.5);
}
:host(:not([disabled])[variant=fab]) [part=base]:hover, :host(:not([disabled])[variant=fab]) [part=base]:focus, :host(:not([disabled])[variant=fab]) [part=base]:active {
  background: hsla(var(--ig-primary-600), var(--ig-primary-a));
}`;
