// igniteui-webcomponents/components/button/themes/button/light/button.material.css.js
import { css } from 'lit';
export const styles = css `:host {
  --is-large: clamp(0, (var(--component-size, 1) + 1) - var(--ig-size-large, 3), 1);
  --is-medium:
      min(
          clamp(0, (var(--component-size, 1) + 1) - var(--ig-size-medium, 2), 1),
          clamp(0, var(--ig-size-large, 3) - var(--component-size, 1), 1)
      );
  --is-small: clamp(0, var(--ig-size-medium) - var(--component-size, 1), 1);
  position: relative;
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
  scrollbar-width: var(--ig-scrollbar-size);
  scrollbar-color: var(--ig-scrollbar-thumb-background) var(--ig-scrollbar-track-background);
}
:host *,
:host *::before,
:host *::after {
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
}
:host ::-webkit-scrollbar {
  width: var(--ig-scrollbar-size);
  height: var(--ig-scrollbar-size);
  background: var(--ig-scrollbar-track-background);
}
:host ::-webkit-scrollbar-thumb {
  background: var(--ig-scrollbar-thumb-background);
}

[hidden] {
  display: none !important;
}

/* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */
:host {
  --size: max(var(--is-large, 1) * max(2.25rem, -1 * 2.25rem), var(--is-medium, 1) * max(1.875rem, -1 * 1.875rem), var(--is-small, 1) * max(1.5rem, -1 * 1.5rem));
  display: -webkit-inline-box;
  display: -ms-inline-flexbox;
  display: inline-flex;
  vertical-align: middle;
  height: var(--size);
}
:host ::slotted(*) {
  display: -webkit-inline-box;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  font-size: inherit !important;
}

:host([size=small]) {
  --component-size: var(--ig-size, var(--ig-size-small));
}

:host([size=medium]) {
  --component-size: var(--ig-size, var(--ig-size-medium));
}

:host([size=large]) {
  --component-size: var(--ig-size, var(--ig-size-large));
}

[part=base] {
  border-radius: clamp(0rem, calc(var(--ig-radius-factor) * 0.25rem), 0.25rem);
  font-family: var(--ig-button-font-family, var(--ig-font-family));
  font-size: var(--ig-button-font-size);
  font-weight: var(--ig-button-font-weight);
  font-style: var(--ig-button-font-style);
  line-height: var(--ig-button-line-height);
  letter-spacing: var(--ig-button-letter-spacing);
  text-transform: var(--ig-button-text-transform);
  margin-top: var(--ig-button-margin-top);
  margin-bottom: var(--ig-button-margin-bottom);
  position: relative;
  display: -webkit-inline-box;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
  -webkit-box-flex: 1;
      -ms-flex: 1;
          flex: 1;
  border: none;
  -webkit-user-select: none;
     -moz-user-select: none;
      -ms-user-select: none;
          user-select: none;
  outline-style: none;
  -webkit-tap-highlight-color: transparent;
  overflow: hidden;
  white-space: nowrap;
  text-decoration: none;
  z-index: 0;
  height: var(--size);
  padding: 0 max(var(--is-large, 1) * max(16px, -1 * 16px) * var(--ig-spacing-large, var(--ig-spacing, --ig-spacing)), var(--is-medium, 1) * max(12px, -1 * 12px) * var(--ig-spacing-medium, var(--ig-spacing, --ig-spacing)), var(--is-small, 1) * max(8px, -1 * 8px) * var(--ig-spacing-small, var(--ig-spacing, --ig-spacing)));
  gap: 0.5rem;
}

:host(:not([disabled])) [part=base]:hover {
  cursor: pointer;
}

:host([disabled]) {
  pointer-events: none;
}
:host([disabled]) [part=base] {
  background: hsla(var(--ig-gray-300), var(--ig-gray-a));
  color: hsla(var(--ig-gray-500), var(--ig-gray-a));
  -webkit-box-shadow: none;
          box-shadow: none;
}

:host(:not([disabled])[variant=outlined]) [part=base], :host(:not([disabled])[variant=flat]) [part=base] {
  color: hsla(var(--ig-secondary-500), var(--ig-secondary-a));
  background: transparent;
  -webkit-transition: background-color 0.1s ease-out 0s;
  -o-transition: background-color 0.1s ease-out 0s;
  transition: background-color 0.1s ease-out 0s;
}
:host(:not([disabled])[variant=outlined]) [part=base]:hover, :host(:not([disabled])[variant=flat]) [part=base]:hover {
  background: hsla(var(--ig-secondary-500), 0.12);
}
:host(:not([disabled])[variant=outlined]) [part=base]:focus, :host(:not([disabled])[variant=flat]) [part=base]:focus, :host(:not([disabled])[variant=outlined]) [part=base]:active, :host(:not([disabled])[variant=flat]) [part=base]:active {
  background: hsla(var(--ig-secondary-500), 0.24);
}

:host([disabled][variant=flat]) [part=base] {
  background: transparent;
}

:host(:not([disabled])[variant=outlined]) [part=base] {
  -webkit-box-shadow: inset 0 0 0 0.0625rem hsla(var(--ig-secondary-500), var(--ig-secondary-a));
          box-shadow: inset 0 0 0 0.0625rem hsla(var(--ig-secondary-500), var(--ig-secondary-a));
}

:host([disabled][variant=outlined]) [part=base] {
  background: transparent;
  -webkit-box-shadow: inset 0 0 0 0.0625rem hsla(var(--ig-gray-300), var(--ig-gray-a));
          box-shadow: inset 0 0 0 0.0625rem hsla(var(--ig-gray-300), var(--ig-gray-a));
}

:host(:not([disabled])[variant=contained]) [part=base], :host(:not([disabled])[variant=fab]) [part=base] {
  background: hsla(var(--ig-secondary-500), var(--ig-secondary-a));
  color: var(--ig-secondary-500-contrast);
}

:host([variant=fab]) {
  --size: max(var(--is-large, 1) * max(3rem, -1 * 3rem), var(--is-medium, 1) * max(2.5rem, -1 * 2.5rem), var(--is-small, 1) * max(2rem, -1 * 2rem));
}

:host([variant=fab]) [part=base] {
  border-radius: clamp(0rem, calc(var(--ig-radius-factor) * calc(var(--size) / 2)), calc(var(--size) / 2));
  min-width: var(--size);
  min-height: var(--size);
}

:host(:not([disabled])[variant=fab]) [part=base], :host(:not([disabled])[variant=contained]) [part=base] {
  -webkit-transition: -webkit-box-shadow 0.1s ease-out;
  transition: -webkit-box-shadow 0.1s ease-out;
  -o-transition: box-shadow 0.1s ease-out;
  transition: box-shadow 0.1s ease-out;
  transition: box-shadow 0.1s ease-out, -webkit-box-shadow 0.1s ease-out;
}
:host(:not([disabled])[variant=fab]) [part=base]::before, :host(:not([disabled])[variant=contained]) [part=base]::before {
  content: "";
  position: absolute;
  width: 100%;
  height: 100%;
  inset-inline-start: 0;
  inset-block-start: 0;
  opacity: 0;
}
:host(:not([disabled])[variant=fab]) [part=base]:hover, :host(:not([disabled])[variant=contained]) [part=base]:hover {
  -webkit-box-shadow: var(--ig-elevation-4);
          box-shadow: var(--ig-elevation-4);
}
:host(:not([disabled])[variant=fab]) [part=base]:hover::before, :host(:not([disabled])[variant=contained]) [part=base]:hover::before {
  background: #fff;
  opacity: 0.12;
}
:host(:not([disabled])[variant=fab]) [part=base]:focus, :host(:not([disabled])[variant=contained]) [part=base]:focus, :host(:not([disabled])[variant=fab]) [part=base]:active, :host(:not([disabled])[variant=contained]) [part=base]:active {
  -webkit-box-shadow: var(--ig-elevation-8);
          box-shadow: var(--ig-elevation-8);
}
:host(:not([disabled])[variant=fab]) [part=base]:focus::before, :host(:not([disabled])[variant=contained]) [part=base]:focus::before, :host(:not([disabled])[variant=fab]) [part=base]:active::before, :host(:not([disabled])[variant=contained]) [part=base]:active::before {
  background: #fff;
  opacity: 0.24;
}

:host(:not([disabled])[variant=contained]) [part=base] {
  -webkit-box-shadow: var(--ig-elevation-2);
          box-shadow: var(--ig-elevation-2);
}

:host(:not([disabled])[variant=fab]) [part=base] {
  -webkit-box-shadow: var(--ig-elevation-6);
          box-shadow: var(--ig-elevation-6);
}
:host(:not([disabled])[variant=fab]) [part=base]:hover, :host(:not([disabled])[variant=fab]) [part=base]:focus, :host(:not([disabled])[variant=fab]) [part=base]:active {
  -webkit-box-shadow: var(--ig-elevation-12);
          box-shadow: var(--ig-elevation-12);
}`;
