// igniteui-webcomponents/components/select/themes/light/select.base.css.js
import { css } from 'lit';
export const styles = css `:host {
  --is-large: clamp(0, (var(--component-size, 1) + 1) - var(--ig-size-large, 3), 1);
  --is-medium:
      min(
          clamp(0, (var(--component-size, 1) + 1) - var(--ig-size-medium, 2), 1),
          clamp(0, var(--ig-size-large, 3) - var(--component-size, 1), 1)
      );
  --is-small: clamp(0, var(--ig-size-medium) - var(--component-size, 1), 1);
  position: relative;
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
  scrollbar-width: var(--ig-scrollbar-size);
  scrollbar-color: var(--ig-scrollbar-thumb-background) var(--ig-scrollbar-track-background);
}
:host *,
:host *::before,
:host *::after {
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
}
:host ::-webkit-scrollbar {
  width: var(--ig-scrollbar-size);
  height: var(--ig-scrollbar-size);
  background: var(--ig-scrollbar-track-background);
}
:host ::-webkit-scrollbar-thumb {
  background: var(--ig-scrollbar-thumb-background);
}

[hidden] {
  display: none !important;
}

/* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */
:host {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  font-family: var(--ig-font-family);
}

[part=base] {
  border-radius: clamp(0rem, calc(var(--ig-radius-factor) * 0.25rem), 0.25rem);
  overflow-y: auto;
  overflow-x: hidden;
  background: hsla(var(--ig-surface-500), var(--ig-surface-a));
  -webkit-box-shadow: var(--ig-elevation-8);
          box-shadow: var(--ig-elevation-8);
  min-width: 7rem;
  padding: 0.5rem 0;
  z-index: 10005;
}

:host([size=small]) {
  --component-size: var(--ig-size, var(--ig-size-small));
}

:host([size=medium]) {
  --component-size: var(--ig-size, var(--ig-size-medium));
}

:host([size=large]) {
  --component-size: var(--ig-size, var(--ig-size-large));
}

:host {
  display: block;
}

[role=combobox] {
  outline-style: none;
}

:host([invalid]) ::slotted([slot=helper-text]) {
  color: hsla(var(--ig-error-500), var(--ig-error-a));
}

:host([disabled]) [role=combobox] {
  pointer-events: none;
}

igc-input {
  cursor: pointer;
}

igc-input::part(input) {
  cursor: pointer;
}
igc-input::part(input)::-moz-selection {
  background: transparent;
}
igc-input::part(input)::selection {
  background: transparent;
}

[part=helper-text] {
  line-height: 1;
}
[part=helper-text] ::slotted([slot=helper-text]) {
  font-family: var(--ig-caption-font-family, var(--ig-font-family));
  font-size: var(--ig-caption-font-size);
  font-weight: var(--ig-caption-font-weight);
  font-style: var(--ig-caption-font-style);
  line-height: var(--ig-caption-line-height);
  letter-spacing: var(--ig-caption-letter-spacing);
  text-transform: var(--ig-caption-text-transform);
  margin-top: var(--ig-caption-margin-top);
  margin-bottom: var(--ig-caption-margin-bottom);
  line-height: inherit;
}

[name=prefix]::slotted(*),
[name=suffix]::slotted(*) {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
}`;
